---
title: "GO_results_SCRI_ROOT_RNAseq"
author: "Houston Saxe"
date: "1/24/2022"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load libraries

```{r}
pacman::p_load(dplyr, 
               tibble, 
               readr, 
               stringr, 
               data.table, 
               ggplot2, 
               sjPlot, 
               ggpubr, 
               tidyr, 
               jsonlite,
               openxlsx)
```

# Get GO Chromosome annotation for J. regia genome

```{r}
CHr_annotation = fread('DGEresults/annotation_combined.csv') 

peaks = CHr_annotation %>% 
  filter(start >= 26359154 & end <= 26669075 & chromosome == '4D')
```

```{r}
CHr_annotation = CHr_annotation %>% 
  select(GeneID, chromosome)

CHr_annotation[CHr_annotation == ''] <- 'Unmapped'
```


# CG Get logFCs from limma results for J. regia and J. microcarpa and merge with GO results

```{r}
impCG = fread('DGEresults/Limma_results_table_CG.csv') %>% 
  distinct(GeneID, 
           name, 
           logFC,
           `Jr-GeneID`,
           Parent_haplotype)
```

## Merge DEG data with Chromosome annotation

```{r}
impCG_CHr = impCG %>% 
  left_join(CHr_annotation) %>% 
  distinct()

length(unique(impCG_CHr$logFC))
```

## Make list for KS test results

```{r}
print_res = list()
```

# Panther uses Kolmogorov-Smirnov (KS) test to test that the distribution of logFC values for a particular Chromosome are different from a null distribution. In this case, I'm using the enitre set of DEG logFCs as the null distribution.

## CG Chromosome KS test

```{r}
ref_CG = impCG_CHr %>% 
  distinct(GeneID, logFC) %>% 
  pull(logFC)

ks_res_CG = impCG_CHr %>% 
  group_by(chromosome) %>% 
  mutate(p.val = ks.test(logFC, ref_CG)$p.value,
         Avg_logFC = mean(logFC),
         Number_of_genes = n()) %>% 
  ungroup() %>% 
  mutate(fdr = p.adjust(p.val, method = 'fdr'))

ks_res_less_CG = ks_res_CG %>% 
  select(!c(GeneID,
            logFC,
            name,
            Parent_haplotype)) %>% 
  distinct() %>% 
  mutate(p.val = ifelse(p.val == 0, 0.000000000000001, p.val),
           fdr = p.adjust(p.val, method = 'fdr')) 

print_res$CG_No_filter = print(paste('Total of',
            ks_res_CG %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_CG %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
            
print_res$CG_Filter = print(paste('and with fdr <= 0.05', 
            ks_res_CG %>% filter(fdr < 0.05) %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_CG %>% filter(fdr < 0.05) %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
```

## Summary of KS results

```{r}
print_res
```

## CG Make plotting data

```{r fig.height=8, fig.width=10}
one = ks_res_less_CG

# Merge DGE results with GO results
plot_dat = one %>% 
  select(chromosome, 
         fdr,
         `Number_of_genes`,
         Avg_logFC) %>% 
  distinct() %>%
  drop_na() %>% 
  filter(fdr <= 0.05)
  

# Make new table for combining DEG results with GO results and writing out
impCG_CHr_Enrichment_out = plot_dat %>% 
  mutate(Trait = 'CG')

labs_normal = plot_dat %>% 
  # Make labels for terms containing these strings and fdr less than or equal to 0.05
  filter(fdr <= 0.05) %>% 
  select(chromosome,
         fdr,
         Avg_logFC,
         `Number_of_genes`) %>% 
  distinct()

counts = labs_normal %>% 
  count(chromosome)

# Labels for top 5 terms by fdr
top = plot_dat %>% 
  slice_min(order_by = fdr, n = 5) %>% 
  rbind(labs_normal) %>% 
  distinct() %>% 
  mutate(Gene_type = ifelse(Avg_logFC >= 0, 'Susceptibility', 'Resistance'))

```

## CG Make plot2

```{r fig.height=6, fig.width=9, warning=FALSE, message=FALSE}

CG = ggplot(top, aes(x = Gene_type, y = reorder(chromosome, Avg_logFC), color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1, color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1), shape = 1, stroke = 0.25, color = 'black')+
  labs(title = 'Chromosome Kolmogorov-Smirnov Test A. tumefaciens',
       x = 'Gene Type',
       y = 'Chromosome',
       color = 'Mean log2\nExpression\nFold Change',
       size = 'log10(fdr)*-1')+
  theme_grey(base_size = 12)+
  theme(plot.title = element_text(hjust = 0.9))+
  # geom_hline(yintercept = log10(0.05)*-1, linetype = 'dashed', color = 'red')+
  # geom_text(aes(min(Avg_logFC), log10(0.05)*-1),
  #           label = 'FDR 0.05',
  #           vjust = 1.5,
  #           hjust = 0.5, 
  #           color = 'black',
  #           size = 3)+
  # geom_vline(xintercept = 0, linetype = 'dashed', color = 'black')+
  # geom_hline(yintercept = 0, linetype = 'dashed', color = 'black')+
  # lims(x = c(sqrt(max(plot_dat$logFC^2))*-1, sqrt(max(plot_dat$logFC^2))),
  #      y = c(-1, max(log10(plot_dat$fdr)*-1)))+
  scale_color_gradient2(low = 'blue', high = 'red')

  # scale_size_continuous(breaks = c(5, 20, 50, 200, 600),
  #                       limits = c(0, 600))+
    # facet_wrap(~term.level, ncol = 2, scales = 'free')

save_plot('Chromosome_KS_results/CHr_Enrich_testing_CG.png', 
          CG,
          width = 18, 
          height = 15, 
          dpi = 300)
```

# PHY Get logFCs from limma results for J. regia and J. microcarpa and merge with GO results

```{r}
impPHY = fread('DGEresults/Limma_results_table_PHY.csv') %>% 
  distinct(GeneID, name, logFC, Parent_haplotype)

```

## Merge DEG data with Chromosome annotation

```{r}
impPHY_CHr = impPHY %>% 
  left_join(CHr_annotation) %>% 
  distinct()

length(unique(impPHY_CHr$logFC))
```

# Panther uses Kolmogorov-Smirnov (KS) test to test that the distribution of logFC values for a particular Chromosome are different from a null distribution. In this case, I'm using the enitre set of DEG logFCs as the null distribution.

## PHY Chromosome KS test

```{r}
ref_PHY = impPHY_CHr %>% 
  distinct(GeneID, logFC) %>% 
  pull(logFC)

ks_res_PHY = impPHY_CHr %>% 
  group_by(chromosome) %>% 
  mutate(p.val = ks.test(logFC, ref_PHY)$p.value,
         Avg_logFC = mean(logFC),
         Number_of_genes = n()) %>% 
  ungroup() %>% 
  mutate(fdr = p.adjust(p.val, method = 'fdr'))

ks_res_less_PHY = ks_res_PHY %>% 
  select(!c(GeneID,
            logFC,
            name,
            Parent_haplotype)) %>% 
  distinct() %>% 
  mutate(p.val = ifelse(p.val == 0, 0.000000000000001, p.val),
           fdr = p.adjust(p.val, method = 'fdr')) 

print_res$PHY_No_filter = print(paste('Total of',
            ks_res_PHY %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_PHY %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
            
print_res$PHY_Filter = print(paste('and with fdr <= 0.05', 
            ks_res_PHY %>% filter(fdr < 0.05) %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_PHY %>% filter(fdr < 0.05) %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
```

## Summary of KS results

```{r}
print_res
```

## PHY Make plotting data

```{r fig.height=8, fig.width=10}
one = ks_res_less_PHY 

# Merge DGE results with GO results
plot_dat = one %>% 
  select(chromosome, 
         fdr,
         `Number_of_genes`,
         Avg_logFC) %>% 
  distinct() %>%
  drop_na() %>% 
  filter(fdr <= 0.05)
  

# Make new table for combining DEG results with GO results and writing out
impPHY_CHr_Enrichment_out = plot_dat %>% 
  mutate(Trait = 'PHY')

labs_normal = plot_dat %>% 
  # Make labels for terms containing these strings and fdr less than or equal to 0.05
  filter(fdr <= 0.05) %>% 
  select(chromosome,
         fdr,
         Avg_logFC,
         `Number_of_genes`) %>% 
  distinct()

counts = labs_normal %>% 
  count(chromosome)

# Labels for top 5 terms by fdr
top = plot_dat %>% 
  slice_min(order_by = fdr, n = 5) %>% 
  rbind(labs_normal) %>% 
  distinct() %>% 
  mutate(Gene_type = ifelse(Avg_logFC >= 0, 'Susceptibility', 'Resistance'))

```

## PHY Make plot2

```{r fig.height=6, fig.width=9, warning=FALSE, message=FALSE}

PHY = ggplot(top, aes(x = Gene_type, y = reorder(chromosome, Avg_logFC), color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1, color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1), shape = 1, stroke = 0.25, color = 'black')+
  labs(title = 'Chromosome Kolmogorov-Smirnov Test Phytophthora',
       x = 'Gene Type',
       y = 'Chromosome',
       color = 'Mean log2\nExpression\nFold Change',
       size = 'log10(fdr)*-1')+
  theme_grey(base_size = 12)+
  theme(plot.title = element_text(hjust = 0.9))+
  # geom_hline(yintercept = log10(0.05)*-1, linetype = 'dashed', color = 'red')+
  # geom_text(aes(min(Avg_logFC), log10(0.05)*-1),
  #           label = 'FDR 0.05',
  #           vjust = 1.5,
  #           hjust = 0.5, 
  #           color = 'black',
  #           size = 3)+
  # geom_vline(xintercept = 0, linetype = 'dashed', color = 'black')+
  # geom_hline(yintercept = 0, linetype = 'dashed', color = 'black')+
  # lims(x = c(sqrt(max(plot_dat$logFC^2))*-1, sqrt(max(plot_dat$logFC^2))),
  #      y = c(-1, max(log10(plot_dat$fdr)*-1)))+
  scale_color_gradient2(low = 'blue', high = 'red')

  # scale_size_continuous(breaks = c(5, 20, 50, 200, 600),
  #                       limits = c(0, 600))+
    # facet_wrap(~term.level, ncol = 2, scales = 'free')

save_plot('Chromosome_KS_results/CHr_Enrich_testing_PHY.png', 
          PHY,
          width = 18, 
          height = 15, 
          dpi = 300)
```

# NEM_3Y Get logFCs from limma results for J. regia and J. microcarpa and merge with GO results

```{r}
impNEM_3Y = fread('DGEresults/Limma_results_table_NEM_3Y.csv') %>% 
  distinct(GeneID, name, logFC, Parent_haplotype)

```

## Merge DEG data with Chromosome annotation

```{r}
impNEM_3Y_CHr = impNEM_3Y %>% 
  left_join(CHr_annotation) %>% 
  distinct()

length(unique(impNEM_3Y_CHr$logFC))
```

# Panther uses Kolmogorov-Smirnov (KS) test to test that the distribution of logFC values for a particular Chromosome are different from a null distribution. In this case, I'm using the enitre set of DEG logFCs as the null distribution.

## NEM_3Y Chromosome KS test

```{r}
ref_NEM_3Y = impNEM_3Y_CHr %>% 
  distinct(GeneID, logFC) %>% 
  pull(logFC)

ks_res_NEM_3Y = impNEM_3Y_CHr %>% 
  group_by(chromosome) %>% 
  mutate(p.val = ks.test(logFC, ref_NEM_3Y)$p.value,
         Avg_logFC = mean(logFC),
         Number_of_genes = n()) %>% 
  ungroup() %>% 
  mutate(fdr = p.adjust(p.val, method = 'fdr'))

ks_res_less_NEM_3Y = ks_res_NEM_3Y %>% 
  select(!c(GeneID,
            logFC,
            name,
            Parent_haplotype)) %>% 
  distinct() %>% 
  mutate(p.val = ifelse(p.val == 0, 0.000000000000001, p.val),
           fdr = p.adjust(p.val, method = 'fdr')) 

print_res$NEM_3Y_No_filter = print(paste('Total of',
            ks_res_NEM_3Y %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_NEM_3Y %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
            
print_res$NEM_3Y_Filter = print(paste('and with fdr <= 0.05', 
            ks_res_NEM_3Y %>% filter(fdr < 0.05) %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_NEM_3Y %>% filter(fdr < 0.05) %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
```

## Summary of KS results

```{r}
print_res
```

## NEM_3Y Make plotting data

```{r fig.height=8, fig.width=10}
one = ks_res_less_NEM_3Y

# Merge DGE results with GO results
plot_dat = one %>% 
  select(chromosome, 
         fdr,
         `Number_of_genes`,
         Avg_logFC) %>% 
  distinct() %>%
  drop_na() %>% 
  filter(fdr <= 0.05)
  

# Make new table for combining DEG results with GO results and writing out
impNEM_3Y_CHr_Enrichment_out = plot_dat %>% 
  mutate(Trait = 'NEM_3Y')

labs_normal = plot_dat %>% 
  # Make labels for terms containing these strings and fdr less than or equal to 0.05
  filter(fdr <= 0.05) %>% 
  select(chromosome,
         fdr,
         Avg_logFC,
         `Number_of_genes`) %>% 
  distinct()

counts = labs_normal %>% 
  count(chromosome)

# Labels for top 5 terms by fdr
top = plot_dat %>% 
  slice_min(order_by = fdr, n = 5) %>% 
  rbind(labs_normal) %>% 
  distinct() %>% 
  mutate(Gene_type = ifelse(Avg_logFC >= 0, 'Susceptibility', 'Resistance'))

```

## NEM_3Y Make plot2

```{r fig.height=6, fig.width=9, warning=FALSE, message=FALSE}

NEM_3Y = ggplot(top, aes(x = Gene_type, y = reorder(chromosome, Avg_logFC), color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1, color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1), shape = 1, stroke = 0.25, color = 'black')+
  labs(title = 'Chromosome Kolmogorov-Smirnov Test P. vulnus 3Y',
       x = 'Gene Type',
       y = 'Chromosome',
       color = 'Mean log2\nExpression\nFold Change',
       size = 'log10(fdr)*-1')+
  theme_grey(base_size = 12)+
  theme(plot.title = element_text(hjust = 0.9))+
  # geom_hline(yintercept = log10(0.05)*-1, linetype = 'dashed', color = 'red')+
  # geom_text(aes(min(Avg_logFC), log10(0.05)*-1),
  #           label = 'FDR 0.05',
  #           vjust = 1.5,
  #           hjust = 0.5, 
  #           color = 'black',
  #           size = 3)+
  # geom_vline(xintercept = 0, linetype = 'dashed', color = 'black')+
  # geom_hline(yintercept = 0, linetype = 'dashed', color = 'black')+
  # lims(x = c(sqrt(max(plot_dat$logFC^2))*-1, sqrt(max(plot_dat$logFC^2))),
  #      y = c(-1, max(log10(plot_dat$fdr)*-1)))+
  scale_color_gradient2(low = 'blue', high = 'red')

  # scale_size_continuous(breaks = c(5, 20, 50, 200, 600),
  #                       limits = c(0, 600))+
    # facet_wrap(~term.level, ncol = 2, scales = 'free')

save_plot('Chromosome_KS_results/CHr_Enrich_testing_NEM_3Y.png', 
          NEM_3Y,
          width = 18, 
          height = 15, 
          dpi = 300)
```

# Length_3Y Get logFCs from limma results for J. regia and J. microcarpa and merge with GO results

```{r}
impLength_3Y = fread('DGEresults/Limma_results_table_Length_3Y.csv') %>% 
  distinct(GeneID, name, logFC, Parent_haplotype)

```

## Merge DEG data with Chromosome annotation

```{r}
impLength_3Y_CHr = impLength_3Y %>% 
  left_join(CHr_annotation) %>% 
  distinct()

length(unique(impLength_3Y_CHr$logFC))
```

# Panther uses Kolmogorov-Smirnov (KS) test to test that the distribution of logFC values for a particular Chromosome are different from a null distribution. In this case, I'm using the enitre set of DEG logFCs as the null distribution.

## Length_3Y Chromosome KS test

```{r}
ref_Length_3Y = impLength_3Y_CHr %>% 
  distinct(GeneID, logFC) %>% 
  pull(logFC)

ks_res_Length_3Y = impLength_3Y_CHr %>% 
  group_by(chromosome) %>% 
  mutate(p.val = ks.test(logFC, ref_Length_3Y)$p.value,
         Avg_logFC = mean(logFC),
         Number_of_genes = n()) %>% 
  ungroup() %>% 
  mutate(fdr = p.adjust(p.val, method = 'fdr'))

ks_res_less_Length_3Y = ks_res_Length_3Y %>% 
  select(!c(GeneID,
            logFC,
            name,
            Parent_haplotype)) %>% 
  distinct() %>% 
  mutate(p.val = ifelse(p.val == 0, 0.000000000000001, p.val),
           fdr = p.adjust(p.val, method = 'fdr')) 

print_res$Length_3Y_No_filter = print(paste('Total of',
            ks_res_Length_3Y %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_Length_3Y %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
            
print_res$Length_3Y_Filter = print(paste('and with fdr <= 0.05', 
            ks_res_Length_3Y %>% filter(fdr < 0.05) %>% pull(logFC) %>% unique() %>% length(),
            'genes mapped to', 
            ks_res_Length_3Y %>% filter(fdr < 0.05) %>% pull(chromosome) %>% unique() %>% length(),
            'Chromosome '))
```

## Summary of KS results

```{r}
print_res
```

## Length_3Y Make plotting data

```{r fig.height=8, fig.width=10}
one = ks_res_less_Length_3Y 

# Merge DGE results with GO results
plot_dat = one %>% 
  select(chromosome, 
         fdr,
         `Number_of_genes`,
         Avg_logFC) %>% 
  distinct() %>%
  drop_na() %>% 
  filter(fdr <= 0.05)
  

# Make new table for combining DEG results with GO results and writing out
impLength_3Y_CHr_Enrichment_out = plot_dat %>% 
  mutate(Trait = 'Length_3Y')

labs_normal = plot_dat %>% 
  # Make labels for terms containing these strings and fdr less than or equal to 0.05
  filter(fdr <= 0.05) %>% 
  select(chromosome,
         fdr,
         Avg_logFC,
         `Number_of_genes`) %>% 
  distinct()

counts = labs_normal %>% 
  count(chromosome)

# Labels for top 5 terms by fdr
top = plot_dat %>% 
  slice_min(order_by = fdr, n = 5) %>% 
  rbind(labs_normal) %>% 
  distinct() %>% 
  mutate(Gene_type = ifelse(Avg_logFC >= 0, 'Susceptibility', 'Resistance'))

```

## Length_3Y Make plot2

```{r fig.height=6, fig.width=9, warning=FALSE, message=FALSE}

Length_3Y = ggplot(top, aes(x = Gene_type, y = reorder(chromosome, Avg_logFC), color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1, color = Avg_logFC))+
  geom_point(aes(size = log10(fdr)*-1), shape = 1, stroke = 0.25, color = 'black')+
  labs(title = 'Chromosome Kolmogorov-Smirnov Test Tree Length 3Y',
       x = 'Gene Type',
       y = 'Chromosome',
       color = 'Mean log2\nExpression\nFold Change',
       size = 'log10(fdr)*-1')+
  theme_grey(base_size = 12)+
  theme(plot.title = element_text(hjust = 0.9))+
  # geom_hline(yintercept = log10(0.05)*-1, linetype = 'dashed', color = 'red')+
  # geom_text(aes(min(Avg_logFC), log10(0.05)*-1),
  #           label = 'FDR 0.05',
  #           vjust = 1.5,
  #           hjust = 0.5, 
  #           color = 'black',
  #           size = 3)+
  # geom_vline(xintercept = 0, linetype = 'dashed', color = 'black')+
  # geom_hline(yintercept = 0, linetype = 'dashed', color = 'black')+
  # lims(x = c(sqrt(max(plot_dat$logFC^2))*-1, sqrt(max(plot_dat$logFC^2))),
  #      y = c(-1, max(log10(plot_dat$fdr)*-1)))+
  scale_color_gradient2(low = 'blue', high = 'red')

  # scale_size_continuous(breaks = c(5, 20, 50, 200, 600),
  #                       limits = c(0, 600))+
    # facet_wrap(~term.level, ncol = 2, scales = 'free')

save_plot('Chromosome_KS_results/CHr_Enrich_testing_Length_3Y.png', 
          Length_3Y,
          width = 18, 
          height = 15, 
          dpi = 300)
```



# Write out full results for supplementary data

```{r}
total_results = impCG_CHr_Enrichment_out %>% 
  rbind(impPHY_CHr_Enrichment_out) %>% 
  rbind(impNEM_3Y_CHr_Enrichment_out) %>% 
  rbind(impLength_3Y_CHr_Enrichment_out)

fwrite(total_results, 'Biological_Process_KS_results/Full_supplementary_results_KS_CHr.csv')
```

