Jm3101\_v1.0\_and\_Walnut\_2.0
================
Houston Saxe
1/28/2022

# Load neccessary libraries

``` r
pacman::p_load(rlang, data.table, ggplot2, ggfortify, stringr, dplyr, statmod, tibble, ggpubr, sjPlot, tidyr, tidytext, OmicsAnalyst, tidytable, gridExtra)
```

# Read in and prepare expression data

``` r
dat = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/SCRI_ROOT_RNAseq_counts_combined_genomes.txt")

dat$GeneID = gsub("LOC", "", dat$GeneID)


head(dat)
```

    ##       GeneID 29JM-11-1 29JM-11-2 29JM-11-3 29JM-11-4 29JM-11-5 29JM-11-6
    ## 1: 121242473         6         5         8        10        16        20
    ## 2: 121268866         0         0         0         0         0         0
    ## 3: 121241406       210        70       274       445       564       409
    ## 4: 121250192        38        12        40       111       109        76
    ## 5: 121241144         8         6        17        13        33        13
    ## 6: 121255391        76        28       146       196       206       102
    ##    JMS-12-1 JMS-12-2 JMS-12-3 JMS-12-4 JMS-12-5 JMS-12-6 MS1-122-1 MS1-122-2
    ## 1:       11        6       11       14       16       15        23        29
    ## 2:        0        0        0        0        0        0         0         0
    ## 3:      373      410      415      261      439      609       622       674
    ## 4:       88       81      106       79       92      108        80       123
    ## 5:       11        4        5        7        9       16        13        17
    ## 6:       23      102      110      106       85      275       181       181
    ##    MS1-122-3 MS1-122-4 MS1-122-5 MS1-122-6 MS1-36-1 MS1-36-2 MS1-36-3 MS1-36-4
    ## 1:        50        26        54        23       13       11       15       13
    ## 2:         0         0         0         0        0        0        0        0
    ## 3:       618       233       986       368      787      342     1020      531
    ## 4:        91        52       176        70      153       91      228      148
    ## 5:        20        13        32        11        3       15       20        3
    ## 6:       253        69       259       162      331       76      386      180
    ##    MS1-36-5 MS1-36-6 MS1-41-1 MS1-41-2 MS1-41-3 MS1-41-4 MS1-41-5 MS1-41-6
    ## 1:       12       33        9       14        9        7       17        7
    ## 2:        0        0        0        0        0        0        0        0
    ## 3:      491      714      535      231      454      355      618      331
    ## 4:      129      173      101       54       82       73      107       75
    ## 5:        1       17       10        2        7        9        2        6
    ## 6:       14      126      115       88      181      122      235      162
    ##    MS1-56-1 MS1-56-2 MS1-56-3 MS1-56-4 MS1-56-5 MS1-56-6 STJM-4-1 STJM-4-2
    ## 1:       35       21       25       10        0       30        0        6
    ## 2:        0        0        0        0        0        0        0        0
    ## 3:     1122     1017      846      828      155     1116       18       50
    ## 4:      196      242      156      152       46      247       17       33
    ## 5:        1        1        1        2        1        0        0        0
    ## 6:       43       20       77       26       11       15        0        0
    ##    STJM-4-3 STJM-4-4 STJM-4-5 STJM-4-6
    ## 1:        3        9        3        4
    ## 2:        0        0        0        0
    ## 3:       52      184       83       69
    ## 4:       19       58       18       23
    ## 5:        1        6        0        0
    ## 6:        5        7        2        3

# Read in metadata

``` r
metadata = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/R/Phenotyping/SCRI/LongList3_2Y.csv", stringsAsFactors = T)

head(metadata)
```

    ##    CAL:_Wip_ID...1 CG_Avg. CG_Dec. Cinn.PCLR Cinn.PRLR Pini.PCLR Pini.PRLR
    ## 1:            I8-1    2.40       D         7        30        81        25
    ## 2:            I9-2    2.30       D        22        21        63        14
    ## 3:           Vlach    3.40       K        72        79        59        39
    ## 4:           VX211    3.10       K        57        65        64        44
    ## 5:          29JM-4    3.10       D        24        17        68        31
    ## 6:          29JM-8    2.75       D        55        50        35        34
    ##                  PHY_Dec. TwoY_length TwoY_RLN NEM_Dec. Unity
    ## 1:                      D      124.50      5.0        D D-D-D
    ## 2:                      D      139.75     58.5        D D-D-D
    ## 3:                   SAVE      170.57    112.2        K K-K-K
    ## 4:                   SAVE      133.30     87.4        K K-K-K
    ## 5:                      D       70.10     14.8        D D-D-D
    ## 6: save due to in trials?       77.50     18.7        K D-K-K

# Modify metadata for this analysis

``` r
metadata = data.frame(Sample = colnames(dat)[colnames(dat) != 'GeneID']) %>%
  mutate(Hybrid = as.factor(gsub("\\-\\d$", "", Sample))) %>%
  left_join(metadata, by = c('Hybrid' = 'CAL:_Wip_ID...1'))
  
row.names(metadata) = metadata$Sample

metadata = metadata %>% 
  mutate(PHY_Avg = rowMeans(select(., matches('PCLR|PRLR'))))

head(metadata)
```

    ##              Sample  Hybrid CG_Avg. CG_Dec. Cinn.PCLR Cinn.PRLR Pini.PCLR
    ## 29JM-11-1 29JM-11-1 29JM-11     2.8       D        36        34        45
    ## 29JM-11-2 29JM-11-2 29JM-11     2.8       D        36        34        45
    ## 29JM-11-3 29JM-11-3 29JM-11     2.8       D        36        34        45
    ## 29JM-11-4 29JM-11-4 29JM-11     2.8       D        36        34        45
    ## 29JM-11-5 29JM-11-5 29JM-11     2.8       D        36        34        45
    ## 29JM-11-6 29JM-11-6 29JM-11     2.8       D        36        34        45
    ##           Pini.PRLR PHY_Dec. TwoY_length TwoY_RLN NEM_Dec. Unity PHY_Avg
    ## 29JM-11-1        15        D        96.3     68.9        K D-D-K    32.5
    ## 29JM-11-2        15        D        96.3     68.9        K D-D-K    32.5
    ## 29JM-11-3        15        D        96.3     68.9        K D-D-K    32.5
    ## 29JM-11-4        15        D        96.3     68.9        K D-D-K    32.5
    ## 29JM-11-5        15        D        96.3     68.9        K D-D-K    32.5
    ## 29JM-11-6        15        D        96.3     68.9        K D-D-K    32.5

``` r
fwrite(metadata, 'SCRI_ROOT_RNAseq_metadata.csv')
```

# Plotting metadata

``` r
metaLong = metadata %>% 
  select(!Sample) %>% 
  select(Hybrid, CG_Avg., PHY_Avg, TwoY_RLN) %>% 
  distinct() %>% 
  pivot_longer(where(is.numeric), names_to = 'Trait')

p =ggplot(metaLong, aes(reorder_within(Hybrid, value, Trait), value, fill = Hybrid))+
  geom_col()+
  scale_x_reordered()+
  theme(axis.text.x = element_text(angle = 30))+
  facet_wrap(~Trait, scales = 'free', ncol = 4)
p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-5-1.png)<!-- -->

``` r
save_plot('DGEresults/metadata_plot.png', p, height = 10, width = 17)
```

    ## png 
    ##   2

# Center and scale predictors to make comparisons more interpretable in results

``` r
metadata_S = metadata %>%
  mutate_if(is.numeric, scale)
```

# Limma needs GeneIDs as rownames in expression data. Also, colnames of expression data need to match rownames of metadata

``` r
dat1 = dat %>%
  column_to_rownames(var =  "GeneID") %>%
  as.matrix()

## Do colnames in data match rownames in metadata? If they don't, use match(x,y) produces the order of y required to match the order of x

all(colnames(dat1) == rownames(metadata_S))
```

    ## [1] TRUE

``` r
# Names match

# If they didn't match, use below code
## What order do rows of metadat need to be in to match colnames of dat1?
# match(colnames(dat1), rownames(metadata))

## Reset rownames
# metadata = metadata[match(colnames(dat1), rownames(metadata)),]

# all(colnames(dat1) == rownames(metadata))
# now they match
```

# Microcarpa annotation

``` r
annotation_Jm = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jm_x_Jr/Jm_x_Jr_Genomic_annotation.csv")

## Extract everything but class mRNA and other isoforms. This reduces duplication in the data
annotation_Jm = annotation_Jm %>%
  filter(feature != "mRNA", !grepl('\\sX[2-9]$|\\sX1[0-9]$', name)) %>%
  mutate(GeneID = as.character(GeneID)) %>% 
  mutate(Parent_haplotype = "J.microcarpa")

head(annotation_Jm)
```

    ##    feature        class chromosome  start    end
    ## 1:   ncRNA       lncRNA         1D      7   1181
    ## 2:   ncRNA       lncRNA         1D  14257  15054
    ## 3:     CDS with_protein         1D  27395  59712
    ## 4:     CDS with_protein         1D  85492 108529
    ## 5:     CDS with_protein         1D  85492 108529
    ## 6:     CDS with_protein         1D 137771 144108
    ##                                                         name    GeneID
    ## 1:                              uncharacterized LOC121242473 121242473
    ## 2:                              uncharacterized LOC121268866 121268866
    ## 3: LOW QUALITY PROTEIN: uncharacterized protein LOC121241406 121241406
    ## 4:          membrane-bound O-acyltransferase gup1 isoform X1 121250192
    ## 5:          membrane-bound O-acyltransferase gup1 isoform X1 121250192
    ## 6:                  pectin acetylesterase 12-like isoform X1 121241144
    ##    product_accession Jr-GeneID Jr-chromosome         Subcell_loc Score
    ## 1:    XR_005935900.1        NA                                      NA
    ## 2:    XR_005941271.1        NA                                      NA
    ## 3:    XP_040995100.1 108981175             7             nucleus  1.00
    ## 4:    XP_041005135.1 108980963             7 endomembrane system  0.92
    ## 5:    XP_041005135.1 108980963             7 endomembrane system  0.92
    ## 6:    XP_040994724.1 108980844             7 extracellular space  0.77
    ##            ID                                           Name           Category
    ## 1:                                                                             
    ## 2:                                                                             
    ## 3:                                                                             
    ## 4:    PC00042                                acyltransferase                   
    ## 5: GO:0016746 transferase activity, transferring acyl groups                   
    ## 6: GO:0009505                           plant-type cell wall cellular_component
    ##    Parent_haplotype
    ## 1:     J.microcarpa
    ## 2:     J.microcarpa
    ## 3:     J.microcarpa
    ## 4:     J.microcarpa
    ## 5:     J.microcarpa
    ## 6:     J.microcarpa

# Regia annotation

``` r
annotation_Jr = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jr/Jr_Genomic_annotation.csv")

## Extract everything but class mRNA and other isoforms. This reduces duplication in the data
annotation_Jr = annotation_Jr %>%
  filter(feature != "mRNA", !grepl('\\sX[2-9]$|\\sX1[0-9]$', name)) %>%
  mutate(GeneID = as.character(GeneID)) %>% 
  mutate(Parent_haplotype = "J.regia")

head(annotation_Jr)
```

    ##    feature        class chromosome start
    ## 1:     CDS with_protein          1     6
    ## 2:     CDS with_protein          1  7068
    ## 3:     CDS with_protein          1 18754
    ## 4:     CDS with_protein          1 18754
    ## 5:     CDS with_protein          1 18754
    ## 6:     CDS with_protein          1 18754
    ##                                                 name    GeneID
    ## 1:             auxin efflux carrier component 2-like 118344036
    ## 2:             probable WRKY transcription factor 17 118348327
    ## 3: KH domain-containing protein HEN4-like isoform X1 108990460
    ## 4: KH domain-containing protein HEN4-like isoform X1 108990460
    ## 5: KH domain-containing protein HEN4-like isoform X1 108990460
    ## 6: KH domain-containing protein HEN4-like isoform X1 108990460
    ##    product_accession         Subcell_loc Score         ID
    ## 1:    XP_035539669.1 endomembrane system  0.78           
    ## 2:    XP_035545773.1             nucleus  1.00           
    ## 3:    XP_018819971.1             nucleus  1.00    PC00031
    ## 4:    XP_018819971.1             nucleus  1.00 GO:0005634
    ## 5:    XP_018819971.1             nucleus  1.00 GO:0010468
    ## 6:    XP_018819971.1             nucleus  1.00 GO:0010467
    ##                             Name           Category Parent_haplotype
    ## 1:                                                           J.regia
    ## 2:                                                           J.regia
    ## 3:        RNA metabolism protein                             J.regia
    ## 4:                       nucleus cellular_component          J.regia
    ## 5: regulation of gene expression biological_process          J.regia
    ## 6:               gene expression biological_process          J.regia

# Combine Jm and Jr annotations

``` r
annotation_combined = annotation_Jm %>%
  rbind(annotation_Jr, fill = T)

fwrite(annotation_combined, 'DGEresults/annotation_combined.csv')

BP_anno = annotation_Jm %>% 
  distinct(`Jr-GeneID`) %>% 
  rbind(distinct(annotation_Jr, GeneID), use.names = F) %>% 
  distinct(`Jr-GeneID`) %>% 
  mutate(`Jr-GeneID` = paste0('LOC', `Jr-GeneID`))

fwrite(BP_anno, 'GeneIDs_for_annotation_with_GO.csv')
```

# Create DGEList object

``` r
library(edgeR)
```

    ## Loading required package: limma

``` r
dds = DGEList(dat1)

dim(dds$counts)
```

    ## [1] 67924    42

``` r
## Calculate library normalization factors (does not do anything to data)
dds = calcNormFactors(dds)

## These are the size factors (normalization factors) for each sample
# dds$samples
```

# Filter by max CPM of 75 (This is used for DGE analysis)

``` r
d = expression_filter(dds, DGEList = T, FilterFUN = max, FilterThreshold = 75)
```

    ## Make sure no numeric identifiers are in data as this will negatively impact filtering

    ## [1] "Data has no missing values"
    ## [1] "Removed 60193 features based on max threshold of 75, 7731 remaining"

``` r
## CPM normalized counts of all data
cpm = cpm(dds, prior.count = 2, log = F) 

cpm = cpm %>% 
  data.frame() %>% 
  rownames_to_column(var = 'GeneID') %>%  
  rename_with(~ gsub('X', '', gsub('\\.', '-', .x)))

fwrite(cpm, 'DGEresults/cpm_plotting_data.csv')

## CPM normalized counts of filtered data
cpmd = cpm(d)
```

# Less conservative threshold for looking at expressed genes

``` r
# d2 = expression_filter(dds, DGEList = T, FilterFUN = mean, FilterThreshold = 20)
# 
# fwrite(as.data.frame(d2$counts), 'DGEresults/CPM_20_filtered.csv', row.names = T)
```

## Cluster visualization

``` r
## MDS visualization
# plotMDS(d, col = as.numeric(metadata$Hybrid), cex=1)
```

# Plot PCA to look in R

``` r
pca = plot_pca(cpmd, metadata,
               join_by_name = 'Sample',
               plotting_factors_in = 'col_names',
               plotting_factors_name = Hybrid, 
               x = 'PC1',
               y = 'PC2',
               scale = T, 
               center = T, 
               color = 'Hybrid',
               fill = 'Hybrid',
               plot_type = '2D')
```

    ## Make sure no numeric identifiers are in data as this will drastically impact PCA

    ## Joining, by = c("Sample", "Hybrid")

``` r
pca$plot
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-16-1.png)<!-- -->

# Plot PCA to save

``` r
# a = plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'CG_Avg.', fill = 'CG_Avg.', plot_type = '2D', group_by = 'Hybrid')
# 
# b = plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'Cinn.PRLR', fill = 'Cinn.PRLR', plot_type = '2D', group_by = 'Hybrid')
#           
# c = plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'TwoY_RLN', fill = 'TwoY_RLN', plot_type = '2D', group_by = 'Hybrid')
# 
# 
# save_plot('PCA_Hybrids.png', ggarrange(a, b, c, labels = c('A', 'B', 'C'), ncol = 3), base_height = 3, base_width = 12)
```

# PCA boxplot

``` r
pca = plot_pca(cpmd, metadata,
               join_by_name = 'Sample',
               plotting_factors_in = 'col_names',
               plotting_factors_name = Hybrid, 
               x = 'Hybrid',
               y = 'PC1',
               scale = T, 
               center = T, 
               color = 'Hybrid',
               fill = 'PHY_Avg',
               plot_type = 'boxplot')
```

    ## Make sure no numeric identifiers are in data as this will drastically impact PCA

    ## Joining, by = c("Sample", "Hybrid")

``` r
pca$plot
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-18-1.png)<!-- -->

# PCA scatterplot with fitted line and correlation coefficient

``` r
## Available plotting data: 
# [1] "Sample"      "Hybrid"      "CG_Avg."     "CG_Dec."     "Cinn.PCLR"   "Cinn.PRLR"  
#  [7] "Pini.PCLR"   "Pini.PRLR"   "PHY_Dec."    "TwoY_length" "TwoY_RLN"    "NEM_Dec."   # [13] "Unity"       "PHY_Avg" 

a = plot_pca(cpmd, metadata,
             join_by_name = 'Sample',
             plotting_factors_in = 'col_names',
             plotting_factors_name = Hybrid,
             x = 'PC2', y = 'CG_Avg.',
             scale = T,
             center = T,
             color = 'Hybrid',
             plot_type = 'scatter',
             summarise_for_scatter = T)
```

    ## Make sure no numeric identifiers are in data as this will drastically impact PCA

    ## Joining, by = c("Sample", "Hybrid")

``` r
b = plot_pca(cpmd, metadata,
             join_by_name = 'Sample',
             plotting_factors_in = 'col_names',
             plotting_factors_name = Hybrid,
             x = 'PC5',
             y = 'PHY_Avg',
             scale = T,
             center = T,
             color = 'Hybrid',
             plot_type = 'scatter',
             summarise_for_scatter = T)
```

    ## Make sure no numeric identifiers are in data as this will drastically impact PCA
    ## Joining, by = c("Sample", "Hybrid")

``` r
c = plot_pca(cpmd, metadata,
             join_by_name = 'Sample',
             plotting_factors_in = 'col_names',
             plotting_factors_name = Hybrid, 
             x = 'PC1',
             y = 'TwoY_RLN',
             scale = T, 
             center = T, 
             color = 'Hybrid',
             plot_type = 'scatter',
             summarise_for_scatter = T)
```

    ## Make sure no numeric identifiers are in data as this will drastically impact PCA
    ## Joining, by = c("Sample", "Hybrid")

``` r
arr = ggarrange(a$plot + ggtitle('Trait: Crown Gall Disease'),
                b$plot + ggtitle('Trait: Phytophthora Root Rot'),
                c$plot + ggtitle('Trait: Nematode Count'), 
                labels = c('A)', 'B)', 'C)'), ncol = 1)
```

    ## `geom_smooth()` using formula 'y ~ x'
    ## `geom_smooth()` using formula 'y ~ x'
    ## `geom_smooth()` using formula 'y ~ x'
    ## `geom_smooth()` using formula 'y ~ x'
    ## `geom_smooth()` using formula 'y ~ x'
    ## `geom_smooth()` using formula 'y ~ x'

``` r
arr
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-19-1.png)<!-- -->

``` r
sjPlot::save_plot('DGEresults/PCA_scatter_CG_PH_NEM.png', arr, height = 35, width = 18)
```

    ## png 
    ##   2

# Heatmap

``` r
# library(pheatmap)
# 
# cor = cor(cpmd)
# 
# pheatmap(cor, annotation = select(metadata,Hybrid))
```

## Differential expression analysis

# Fit model for genes associated with CG score

``` r
mm = model.matrix(~CG_Avg., data = metadata_S)

head(mm)
```

    ##           (Intercept)   CG_Avg.
    ## 29JM-11-1           1 0.3775968
    ## 29JM-11-2           1 0.3775968
    ## 29JM-11-3           1 0.3775968
    ## 29JM-11-4           1 0.3775968
    ## 29JM-11-5           1 0.3775968
    ## 29JM-11-6           1 0.3775968

## What is voom?

## 1. Counts are transformed to log2 counts per million reads (CPM), where “per million reads” is defined based on the normalization factors we calculated earlier.

## 2. A linear model is fitted to the log2 CPM for each gene, and the residuals are calculated.

## 3. A smoothed curve is fitted to the sqrt(residual standard deviation) by average expression.

## (see red line in plot below)

## 4. The smoothed curve is used to obtain weights for each gene and sample that are passed into limma along with the log2 CPMs.

## More details at “[voom: precision weights unlock linear model analysis tools for RNA-seq read counts](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29)”

# Filtered mean-variance trend CG

``` r
y <- voom(d, mm, plot = T)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-22-1.png)<!-- -->

# Vs Unfiltered mean-variance trend CG

``` r
# tmp <- voom(dds, mm, plot = T)
```

# Fitting linear models in limma with random effects CG

``` r
## Need to tell limma where the within class correlation is coming from
dupcor_CG = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.CG = dupcor_CG$consensus.correlation

consensus.corr.CG
```

    ## [1] 0.5638856

``` r
# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.CG) 
```

## The variance characteristics of low expressed genes are different from high expressed genes, if treated the same, the effect is to over represent low expressed genes in the DE list. This is corrected for by the log transformation and voom. However, some genes will have increased or decreased variance that is not a result of low expression, but due to other random factors. We are going to run empirical Bayes to adjust the variance of these genes.

## Empirical Bayes smoothing of standard errors (shifts standard errors that are much larger or smaller than those from other genes towards the average standard error) (see “[Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments](https://www.degruyter.com/doi/10.2202/1544-6115.1027)”

``` r
BlockFit_CG = eBayes(fit)
```

# Limma results CG

``` r
res_summaries_CG = BlockFit_CG %>% decideTests() %>% summary()

res_summaries_CG
```

    ##        (Intercept) CG_Avg.
    ## Down             7    1839
    ## NotSig         151    5081
    ## Up            7573     811

# Table of results CG

``` r
impCG = topTable(BlockFit_CG, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)
```

    ## Removing intercept from test coefficients

``` r
dim(impCG)
```

    ## [1] 2650    7

``` r
## adding Hybrid as a blocking variable reduced DEGs by several thousand
```

# PCA of CG DEGs

``` r
ids = impCG %>% 
  rownames_to_column(var = 'GeneID') %>% 
  select(GeneID)

PCA_CG = cpmd %>% 
  as.data.frame() %>% 
  rownames_to_column(var = 'GeneID') %>% 
  right_join(ids) %>% 
  column_to_rownames(var = 'GeneID')
```

    ## Joining, by = "GeneID"

``` r
pca = plot_pca(PCA_CG, metadata,
               join_by_name = 'Sample',
               plotting_factors_in = 'col_names',
               plotting_factors_name = Group, 
               x = 'PC1',
               y = 'PC2',
               scale = T, 
               center = T, 
               color = 'CG_Avg.',
               fill = 'CG_Avg.',
               plot_type = '2D')
```

    ## Make sure no numeric identifiers are in data as this will drastically impact PCA
    ## Joining, by = "Sample"

``` r
pca$plot
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-28-1.png)<!-- -->

# Merge annotation with results CG

``` r
impCG = impCG %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impCG)
```

    ##      GeneID     logFC   AveExpr         t      P.Value    adj.P.Val         B
    ## 1 109017709 -4.882749  1.749451 -8.402591 1.046345e-10 8.089293e-07 12.965646
    ## 2 109016198 -3.897781  3.937337 -5.316091 3.345511e-06 1.657958e-04  4.310482
    ## 3 118344489 -3.647541  3.370420 -7.097001 8.099787e-09 5.218287e-06  9.666944
    ## 4 109018703  3.608900  2.962288  2.781222 7.933791e-03 2.813584e-02 -2.483054
    ## 5 121249183 -3.561753  4.682594 -5.740942 8.016327e-07 7.291085e-05  5.638601
    ## 6 118345731 -3.537504 30.981442 -6.007780 3.249562e-07 3.987676e-05  6.584201
    ##           R feature        class chromosome    start    end
    ## 1 0.7989668     CDS with_protein         15  5713036     NA
    ## 2 0.6434379     CDS with_protein          8 25240305     NA
    ## 3 0.7465672     CDS with_protein         14 14438923     NA
    ## 4 0.4025467     CDS with_protein          4 25672386     NA
    ## 5 0.6721179     CDS with_protein         2D   325909 326637
    ## 6 0.6887165     CDS with_protein               31740     NA
    ##                                               name product_accession Jr-GeneID
    ## 1             uncharacterized protein LOC109017709    XP_035541988.1        NA
    ## 2                               dehydrin DHN1-like    XP_035549466.1        NA
    ## 3           cell division cycle protein 48 homolog    XP_035541057.1        NA
    ## 4 desmethylxanthohumol 6'-O-methyltransferase-like    XP_018856397.1        NA
    ## 5         light-regulated protein 1, chloroplastic    XP_041003832.1 109001297
    ## 6                             endochitinase 2-like    XP_035542912.1        NA
    ##   Jr-chromosome         Subcell_loc Score      ID              Name Category
    ## 1          <NA>             nucleus  1.00                                   
    ## 2          <NA>             nucleus  1.00                                   
    ## 3          <NA>           cytoplasm  0.70                                   
    ## 4          <NA>  organelle membrane  0.82 PC00155 methyltransferase         
    ## 5             1         chloroplast  1.00                                   
    ## 6          <NA> extracellular space  1.00                                   
    ##   Parent_haplotype
    ## 1          J.regia
    ## 2          J.regia
    ## 3          J.regia
    ## 4          J.regia
    ## 5     J.microcarpa
    ## 6          J.regia

``` r
length(unique(impCG$GeneID))
```

    ## [1] 2650

``` r
fwrite(impCG, 'DGEresults/Limma_results_table_CG.csv')
```

# Read in CG results

``` r
impCG = fread('DGEresults/Limma_results_table_CG.csv')
```

# Positive CG DEGs for GO

``` r
CG_GO_pos_Jm = impCG %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(CG_GO_pos_Jm$GeneID))
```

    ## [1] 352

``` r
CG_GO_pos_both = impCG %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(CG_GO_pos_Jm)

length(unique(CG_GO_pos_both$GeneID))
```

    ## [1] 625

``` r
head(CG_GO_pos_both)
```

    ##          GeneID
    ## 1: LOC109018703
    ## 2: LOC108994959
    ## 3: LOC109001241
    ## 4: LOC118349856
    ## 5: LOC109017407
    ## 6: LOC109005579

``` r
fwrite(CG_GO_pos_both, "DGEresults/CG_DEGs_pos_GO.csv")
```

# Expression from alleles is causing duplication in GO results which means data loss as GO removes duplicates. However, these are not true duplicates.

# Maybe analyze separately and try to recombine?

``` r
# CG_GO_pos_Jm = impCG %>%
#   filter(logFC > 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`) %>%
#   mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
#   rename(GeneID = `Jr-GeneID`)
# 
# 
# 
# CG_GO_pos_Jr= impCG %>% 
#   filter(logFC > 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID) %>%
#   mutate(GeneID = paste0("LOC", GeneID))
# 
# length(unique(CG_GO_pos_Jr$GeneID)) + length(unique(CG_GO_pos_Jm$GeneID))
# 
# fwrite(CG_GO_pos_Jm, 'GO_test_CG_pos_Jm.csv')
# 
# 
# fwrite(CG_GO_pos_Jr, 'GO_test_CG_pos_Jr.csv')
```

# Negative CG DEGs for GO

``` r
CG_GO_neg_Jm = impCG %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(CG_GO_neg_Jm$GeneID))
```

    ## [1] 933

``` r
CG_GO_neg_both = impCG %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(CG_GO_neg_Jm)

length(unique(CG_GO_neg_both$GeneID))
```

    ## [1] 1369

``` r
head(CG_GO_neg_both)
```

    ##          GeneID
    ## 1: LOC109017709
    ## 2: LOC109016198
    ## 3: LOC118344489
    ## 4: LOC118345731
    ## 5: LOC109020534
    ## 6: LOC108998592

``` r
fwrite(CG_GO_neg_both, "DGEresults/CG_DEGs_neg_GO.csv")
```

# Expression from alleles is causing duplication in GO results which means data loss as GO removes duplicates. However, these are not true duplicates.

``` r
# dups_neg_Jm = impCG %>%
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`, Parent_haplotype) %>%
#   rename(GeneID = `Jr-GeneID`)
# 
# dups_neg_both = impCG %>% 
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID, Parent_haplotype) %>% 
#   rbind(dups_neg_Jm)
```

# Maybe analyze separately and try to recombine?

``` r
# CG_GO_neg_Jm = impCG %>%
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`) %>%
#   mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
#   rename(GeneID = `Jr-GeneID`)
# 
# 
# 
# CG_GO_neg_Jr= impCG %>% 
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID) %>%
#   mutate(GeneID = paste0("LOC", GeneID))
# 
# length(unique(CG_GO_neg_Jr$GeneID)) + length(unique(CG_GO_neg_Jm$GeneID))
# 
# fwrite(CG_GO_neg_Jm, 'GO_test_CG_neg_Jm.csv')
# 
# 
# fwrite(CG_GO_neg_Jr, 'GO_test_CG_neg_Jr.csv')
```

# Fit model for genes associated with PHY score

``` r
mm = model.matrix(~PHY_Avg, data = metadata_S)

head(mm)
```

    ##           (Intercept)   PHY_Avg
    ## 29JM-11-1           1 0.4675931
    ## 29JM-11-2           1 0.4675931
    ## 29JM-11-3           1 0.4675931
    ## 29JM-11-4           1 0.4675931
    ## 29JM-11-5           1 0.4675931
    ## 29JM-11-6           1 0.4675931

# Filtered mean-variance trend PHY

``` r
y <- voom(d, mm, plot = T)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-37-1.png)<!-- -->

# Fitting linear models in limma with random effects PHY

``` r
## Need to tell limma where the within class correlation is coming from
dupcor_PHY = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.PHY = dupcor_PHY$consensus.correlation

consensus.corr.PHY
```

    ## [1] 0.6408328

``` r
# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.PHY) 

# Ebayes
BlockFit_PHY = eBayes(fit)
```

# Limma results PHY

``` r
res_summaries_PHY = BlockFit_PHY %>% decideTests() %>% summary()

res_summaries_PHY
```

    ##        (Intercept) PHY_Avg
    ## Down             6     746
    ## NotSig         189    6806
    ## Up            7536     179

# Table of results PHY

``` r
impPHY = topTable(BlockFit_PHY, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)
```

    ## Removing intercept from test coefficients

``` r
dim(impPHY)
```

    ## [1] 925   7

``` r
## adding Hybrid as a blocking variable reduced DEGs by several thousand

head(impPHY)
```

    ##               logFC   AveExpr         t      P.Value    adj.P.Val          B
    ## 108996798 -3.928851  1.167056 -7.494921 2.098807e-09 1.622588e-05  7.9914384
    ## 109001297 -3.808179 12.793231 -4.099776 1.742248e-04 1.247160e-02  0.7269986
    ## 121247190 -3.769789 19.294528 -3.474925 1.156409e-03 2.504257e-02 -0.8876449
    ## 108983669 -3.439929  9.412055 -4.624935 3.263099e-05 6.468466e-03  2.1129823
    ## 118345731 -3.176899 30.981442 -4.527675 4.470694e-05 7.053661e-03  1.9140768
    ## 108984807 -2.939146  2.620315 -4.789052 1.910416e-05 4.615444e-03  2.0844077
    ##                   R
    ## 108996798 0.7642555
    ## 109001297 0.5439447
    ## 121247190 0.4815377
    ## 108983669 0.5902782
    ## 118345731 0.5821007
    ## 108984807 0.6036753

# Merge annotation with results PHY

``` r
impPHY = impPHY %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impPHY)
```

    ##      GeneID     logFC  AveExpr         t      P.Value    adj.P.Val        B
    ## 1 108996798 -3.928851 1.167056 -7.494921 2.098807e-09 1.622588e-05 7.991438
    ## 2 108996798 -3.928851 1.167056 -7.494921 2.098807e-09 1.622588e-05 7.991438
    ## 3 108996798 -3.928851 1.167056 -7.494921 2.098807e-09 1.622588e-05 7.991438
    ## 4 108996798 -3.928851 1.167056 -7.494921 2.098807e-09 1.622588e-05 7.991438
    ## 5 108996798 -3.928851 1.167056 -7.494921 2.098807e-09 1.622588e-05 7.991438
    ## 6 108996798 -3.928851 1.167056 -7.494921 2.098807e-09 1.622588e-05 7.991438
    ##           R feature        class chromosome   start end
    ## 1 0.7642555     CDS with_protein          6 2953518  NA
    ## 2 0.7642555     CDS with_protein          6 2953518  NA
    ## 3 0.7642555     CDS with_protein          6 2953518  NA
    ## 4 0.7642555     CDS with_protein          6 2953518  NA
    ## 5 0.7642555     CDS with_protein          6 2953518  NA
    ## 6 0.7642555     CDS with_protein          6 2953518  NA
    ##                                     name product_accession Jr-GeneID
    ## 1 wall-associated receptor kinase 2-like    XP_018828352.1        NA
    ## 2 wall-associated receptor kinase 2-like    XP_018828352.1        NA
    ## 3 wall-associated receptor kinase 2-like    XP_018828352.1        NA
    ## 4 wall-associated receptor kinase 2-like    XP_018828352.1        NA
    ## 5 wall-associated receptor kinase 2-like    XP_018828352.1        NA
    ## 6 wall-associated receptor kinase 2-like    XP_018828354.1        NA
    ##   Jr-chromosome     Subcell_loc Score         ID
    ## 1          <NA> plasma membrane  0.94    PC00197
    ## 2          <NA> plasma membrane  0.94 GO:0004674
    ## 3          <NA> plasma membrane  0.94 GO:0006468
    ## 4          <NA> plasma membrane  0.94 GO:0005886
    ## 5          <NA> plasma membrane  0.94 GO:0007166
    ## 6          <NA> plasma membrane  0.94    PC00197
    ##                                       Name           Category Parent_haplotype
    ## 1            transmembrane signal receptor                             J.regia
    ## 2 protein serine/threonine kinase activity                             J.regia
    ## 3                  protein phosphorylation biological_process          J.regia
    ## 4                          plasma membrane cellular_component          J.regia
    ## 5  cell surface receptor signaling pathway biological_process          J.regia
    ## 6            transmembrane signal receptor                             J.regia

``` r
fwrite(impPHY, 'DGEresults/Limma_results_table_PHY.csv')

impPHY = fread('DGEresults/Limma_results_table_PHY.csv')
```

# Read in PHY results

``` r
impPHY = fread('DGEresults/Limma_results_table_PHY.csv')
```

# Positive PHY DEGs for GO

``` r
PHY_GO_pos_Jm = impPHY %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(PHY_GO_pos_Jm$GeneID))
```

    ## [1] 74

``` r
PHY_GO_pos_both = impPHY %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(PHY_GO_pos_Jm)

length(unique(PHY_GO_pos_both$GeneID))
```

    ## [1] 146

``` r
head(PHY_GO_pos_both)
```

    ##          GeneID
    ## 1: LOC109002793
    ## 2: LOC109011620
    ## 3: LOC108993193
    ## 4: LOC108987644
    ## 5: LOC108991653
    ## 6: LOC108984526

``` r
fwrite(PHY_GO_pos_both, "DGEresults/PHY_DEGs_pos_GO.csv")
```

# Negative PHY DEGs for GO

``` r
PHY_GO_neg_Jm = impPHY %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(PHY_GO_neg_Jm$GeneID))
```

    ## [1] 376

``` r
PHY_GO_neg_both = impPHY %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(PHY_GO_neg_Jm)

length(unique(PHY_GO_neg_both$GeneID))
```

    ## [1] 602

``` r
head(PHY_GO_neg_both)
```

    ##          GeneID
    ## 1: LOC108996798
    ## 2: LOC109001297
    ## 3: LOC108983669
    ## 4: LOC118345731
    ## 5: LOC108984807
    ## 6: LOC109021657

``` r
fwrite(PHY_GO_neg_both, "DGEresults/PHY_DEGs_neg_GO.csv")
```

# Fit model for genes associated with NEM score

``` r
mm = model.matrix(~TwoY_RLN, data = metadata_S)

head(mm)
```

    ##           (Intercept) TwoY_RLN
    ## 29JM-11-1           1 1.156418
    ## 29JM-11-2           1 1.156418
    ## 29JM-11-3           1 1.156418
    ## 29JM-11-4           1 1.156418
    ## 29JM-11-5           1 1.156418
    ## 29JM-11-6           1 1.156418

# Filtered mean-variance trend NEM

``` r
y <- voom(d, mm, plot = T)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-46-1.png)<!-- -->

# Fitting linear models in limma with random effects NEM

``` r
## Need to tell limma where the within class correlation is coming from
dupcor_NEM = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.NEM = dupcor_NEM$consensus.correlation

consensus.corr.NEM
```

    ## [1] 0.61497

``` r
# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.NEM) 

# Ebayes
BlockFit_NEM = eBayes(fit)
```

## Limma results NEM

``` r
res_summaries_NEM = BlockFit_NEM %>% decideTests() %>% summary()

res_summaries_NEM
```

    ##        (Intercept) TwoY_RLN
    ## Down             6     1318
    ## NotSig         198     6087
    ## Up            7527      326

## Table of results NEM

``` r
impNEM = topTable(BlockFit_NEM, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)
```

    ## Removing intercept from test coefficients

``` r
dim(impNEM)
```

    ## [1] 1644    7

``` r
## adding Hybrid as a blocking variable reduced DEGs by several thousand

head(impNEM)
```

    ##               logFC   AveExpr         t      P.Value    adj.P.Val          B
    ## 109016198 -6.399087  3.937337 -5.520731 1.689387e-06 0.0014277057  4.3861890
    ## 109017709 -5.762641  1.749451 -6.617844 4.121929e-08 0.0001062221  6.7327958
    ## 121249183 -4.437534  4.682594 -3.920248 3.048988e-04 0.0125547489  0.1998001
    ## 109022106 -4.292110  1.259263 -4.173055 1.389859e-04 0.0101938824  0.3976808
    ## 109004718 -3.810274 11.686749 -2.992665 4.517800e-03 0.0316082491 -2.0369181
    ## 118343653 -3.574582  3.486222 -4.118100 1.651166e-04 0.0105800998  0.6514344
    ##                   R
    ## 109016198 0.6576099
    ## 109017709 0.7229450
    ## 121249183 0.5268449
    ## 109022106 0.5507363
    ## 109004718 0.4277154
    ## 118343653 0.5456531

## Merge annotation with results NEM

``` r
impNEM = impNEM %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impNEM)
```

    ##      GeneID     logFC   AveExpr         t      P.Value    adj.P.Val          B
    ## 1 109016198 -6.399087  3.937337 -5.520731 1.689387e-06 0.0014277057  4.3861890
    ## 2 109017709 -5.762641  1.749451 -6.617844 4.121929e-08 0.0001062221  6.7327958
    ## 3 121249183 -4.437534  4.682594 -3.920248 3.048988e-04 0.0125547489  0.1998001
    ## 4 109022106 -4.292110  1.259263 -4.173055 1.389859e-04 0.0101938824  0.3976808
    ## 5 109004718 -3.810274 11.686749 -2.992665 4.517800e-03 0.0316082491 -2.0369181
    ## 6 118343653 -3.574582  3.486222 -4.118100 1.651166e-04 0.0105800998  0.6514344
    ##           R feature        class chromosome    start    end
    ## 1 0.6576099     CDS with_protein          8 25240305     NA
    ## 2 0.7229450     CDS with_protein         15  5713036     NA
    ## 3 0.5268449     CDS with_protein         2D   325909 326637
    ## 4 0.5507363     CDS with_protein         10  8877504     NA
    ## 5 0.4277154     CDS with_protein          7 17915184     NA
    ## 6 0.5456531     CDS with_protein          8 25218108     NA
    ##                                       name product_accession Jr-GeneID
    ## 1                       dehydrin DHN1-like    XP_035549466.1        NA
    ## 2     uncharacterized protein LOC109017709    XP_035541988.1        NA
    ## 3 light-regulated protein 1, chloroplastic    XP_041003832.1 109001297
    ## 4                cytochrome P450 78A7-like    XP_018860457.1        NA
    ## 5 late embryogenesis abundant protein D-29    XP_018838898.2        NA
    ## 6                       dehydrin DHN1-like    XP_035549467.1        NA
    ##   Jr-chromosome         Subcell_loc Score ID Name Category Parent_haplotype
    ## 1          <NA>             nucleus  1.00                           J.regia
    ## 2          <NA>             nucleus  1.00                           J.regia
    ## 3             1         chloroplast  1.00                      J.microcarpa
    ## 4          <NA> endomembrane system  0.78                           J.regia
    ## 5          <NA> extracellular space  0.82                           J.regia
    ## 6          <NA>             nucleus  1.00                           J.regia

``` r
fwrite(impNEM, 'DGEresults/Limma_results_table_NEM.csv')

impNEM = fread('DGEresults/Limma_results_table_NEM.csv')
```

# Read in NEM results

``` r
impNEM = fread('DGEresults/Limma_results_table_NEM.csv')
```

# Positive NEM DEGs for GO

``` r
NEM_GO_pos_Jm = impNEM %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(NEM_GO_pos_Jm$GeneID))
```

    ## [1] 137

``` r
NEM_GO_pos_both = impNEM %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(NEM_GO_pos_Jm)

length(unique(NEM_GO_pos_both$GeneID))
```

    ## [1] 262

``` r
head(NEM_GO_pos_both)
```

    ##          GeneID
    ## 1: LOC108994617
    ## 2: LOC109002793
    ## 3: LOC109011537
    ## 4: LOC108989061
    ## 5: LOC109004222
    ## 6: LOC109018242

``` r
fwrite(NEM_GO_pos_both, "DGEresults/NEM_DEGs_pos_GO.csv")
```

# Negative NEM DEGs for GO

``` r
NEM_GO_neg_Jm = impNEM %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(NEM_GO_neg_Jm$GeneID))
```

    ## [1] 641

``` r
NEM_GO_neg_both = impNEM %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(NEM_GO_neg_Jm)

length(unique(NEM_GO_neg_both$GeneID))
```

    ## [1] 1028

``` r
head(NEM_GO_neg_both)
```

    ##          GeneID
    ## 1: LOC109016198
    ## 2: LOC109017709
    ## 3: LOC109022106
    ## 4: LOC109004718
    ## 5: LOC118343653
    ## 6: LOC118349596

``` r
fwrite(NEM_GO_neg_both, "DGEresults/NEM_DEGs_neg_GO.csv")
```

# Any common DEGs between all pathosystems?

``` r
CG = impCG %>%
  select(GeneID, name, logFC, `Jr-GeneID`, Parent_haplotype) %>%
  rename_with(~paste0(., '_CG'))

PHY = impPHY %>%
  select(GeneID, name, logFC, Parent_haplotype) %>%
  rename_with(~paste0(., '_PHY'))

NEM = impNEM %>%
  select(GeneID, name, logFC, Parent_haplotype) %>%
  rename_with(~paste0(., '_NEM'))

impUNITY = CG %>%
  inner_join(PHY, by = c('GeneID_CG' = 'GeneID_PHY')) %>%
  inner_join(NEM, by = c('GeneID_CG' = 'GeneID_NEM'))
```

# Common positive genes across pathosystems for GO

``` r
UNITY_GO_pos_Jm = impUNITY %>%
  filter(logFC_CG > 0 & logFC_PHY > 0 & logFC_NEM > 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG, `Jr-GeneID_CG`) %>%
  filter(Parent_haplotype_CG == "J.microcarpa") %>%
  select(`Jr-GeneID_CG`) %>%
  mutate(`Jr-GeneID_CG` = paste0("LOC", `Jr-GeneID_CG`)) %>% 
  rename(GeneID = `Jr-GeneID_CG`)

length(unique(UNITY_GO_pos_Jm$GeneID))
```

    ## [1] 41

``` r
UNITY_GO_pos_both = impUNITY %>% 
  filter(logFC_CG > 0 & logFC_PHY > 0 & logFC_NEM > 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG,) %>%
  filter(Parent_haplotype_CG == "J.regia") %>% 
  mutate(GeneID_CG = paste0("LOC", GeneID_CG)) %>% 
  select(GeneID_CG) %>%
  rename(GeneID = GeneID_CG)
  rbind(UNITY_GO_pos_Jm, fill = TRUE)
```

    ##           GeneID
    ##  1: LOC108994822
    ##  2: LOC108980693
    ##  3: LOC108985933
    ##  4: LOC109003103
    ##  5: LOC108991181
    ##  6: LOC108997218
    ##  7: LOC108993723
    ##  8: LOC109001162
    ##  9: LOC109010746
    ## 10: LOC108983147
    ## 11: LOC108989203
    ## 12: LOC108997940
    ## 13: LOC109012461
    ## 14: LOC109001180
    ## 15: LOC108994209
    ## 16:        LOCNA
    ## 17: LOC109002963
    ## 18: LOC109010898
    ## 19: LOC109011698
    ## 20: LOC108990002
    ## 21: LOC108993851
    ## 22: LOC108981100
    ## 23: LOC109006717
    ## 24: LOC109011954
    ## 25: LOC108999916
    ## 26: LOC108997367
    ## 27: LOC109022042
    ## 28: LOC109020139
    ## 29: LOC109008795
    ## 30: LOC109007613
    ## 31: LOC109005021
    ## 32: LOC108996558
    ## 33: LOC109004756
    ## 34: LOC108993767
    ## 35: LOC108994691
    ## 36: LOC109008438
    ## 37: LOC108990908
    ## 38: LOC109002071
    ## 39: LOC108999468
    ## 40: LOC109005051
    ## 41: LOC108983614
    ##           GeneID

``` r
length(unique(UNITY_GO_pos_both$GeneID))
```

    ## [1] 55

``` r
head(UNITY_GO_pos_both)
```

    ##          GeneID
    ## 1: LOC109008746
    ## 2: LOC108993193
    ## 3: LOC108981170
    ## 4: LOC108983147
    ## 5: LOC109001162
    ## 6: LOC108991181

``` r
fwrite(UNITY_GO_pos_both, "DGEresults/UNITY_DEGs_pos_GO.csv")
```

# Common negative genes across pathosystems for GO

``` r
UNITY_GO_neg_Jm = impUNITY %>%
  filter(logFC_CG < 0 & logFC_PHY < 0 & logFC_NEM < 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG, `Jr-GeneID_CG`) %>%
  filter(Parent_haplotype_CG == "J.microcarpa") %>%
  select(`Jr-GeneID_CG`) %>%
  mutate(`Jr-GeneID_CG` = paste0("LOC", `Jr-GeneID_CG`)) %>% 
  rename(GeneID = `Jr-GeneID_CG`)

length(unique(UNITY_GO_neg_Jm$GeneID))
```

    ## [1] 287

``` r
UNITY_GO_neg_both = impUNITY %>% 
  filter(logFC_CG < 0 & logFC_PHY < 0 & logFC_NEM < 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG,) %>%
  filter(Parent_haplotype_CG == "J.regia") %>% 
  mutate(GeneID_CG = paste0("LOC", GeneID_CG)) %>% 
  select(GeneID_CG) %>%
  rename(GeneID = GeneID_CG)
  rbind(UNITY_GO_neg_Jm, fill = TRUE)
```

    ##            GeneID
    ##   1: LOC108989475
    ##   2: LOC108981955
    ##   3: LOC108982143
    ##   4: LOC109012906
    ##   5: LOC109012041
    ##  ---             
    ## 293: LOC108986719
    ## 294: LOC108987681
    ## 295: LOC108988014
    ## 296: LOC109009593
    ## 297: LOC109006335

``` r
length(unique(UNITY_GO_neg_both$GeneID))
```

    ## [1] 280

``` r
head(UNITY_GO_neg_both)
```

    ##          GeneID
    ## 1: LOC118345731
    ## 2: LOC109021657
    ## 3: LOC108998169
    ## 4: LOC108984104
    ## 5: LOC109011509
    ## 6: LOC108996798

``` r
fwrite(UNITY_GO_neg_both, "DGEresults/UNITY_DEGs_neg_GO.csv")
```

# Summary of combined unique DEGs

``` r
c(length(unique(UNITY_GO_pos_both)), length(unique(UNITY_GO_neg_both$Gen)))
```

    ## [1]   1 280

# Summaries of each analysis

``` r
summary = cbind(res_summaries_CG,res_summaries_PHY, res_summaries_NEM) %>% 
  as.data.frame() %>% 
  select(!`(Intercept)`) %>% 
  rename(PHY_Avg. = PHY_Avg) %>% 
  rownames_to_column(var = 'Association') %>% 
  mutate(Association = gsub('Down', 'Negative', gsub('Up', 'Positive', Association))) %>% 
  rename_with(~ str_extract(.x, 'Association|CG|PHY|RLN')) %>% 
  rename_with(~ gsub('RLN', 'NEM', .x))

head(summary)
```

    ##   Association   CG  PHY  NEM
    ## 1    Negative 1839  746 1318
    ## 2      NotSig 5081 6806 6087
    ## 3    Positive  811  179  326

``` r
write.csv(summary, 'DGEresults/Results_summaries.csv')
```

# Venn Diagram positive genes

``` r
CG_pos = impCG %>%
  filter(logFC >= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)
  

PHY_pos = impPHY %>%
  filter(logFC >= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)

NEM_pos = impNEM %>%
  filter(logFC >= 0) %>% 
  distinct(GeneID)  %>% 
  pull(GeneID)

venn_pos = list(CG_pos = CG_pos, PHY_pos = PHY_pos, NEM_pos = NEM_pos)
```

# Venn Diagram negative genes

``` r
CG_neg = impCG %>%
  filter(logFC <= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)
  

PHY_neg = impPHY %>%
  filter(logFC <= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)

NEM_neg = impNEM %>%
  filter(logFC <= 0) %>% 
  distinct(GeneID)  %>% 
  pull(GeneID)

venn_neg = list(CG_neg = CG_neg, PHY_neg = PHY_neg, NEM_neg = NEM_neg)
```

# Plot and save venn diagram figure with results summaries

``` r
library(ggvenn)
```

    ## Loading required package: grid

``` r
a = ggvenn(venn_pos)

b = ggvenn(venn_neg)

p = tableGrob(summary)

c = grid.arrange(p)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-61-1.png)<!-- -->

``` r
ggarrange(a, b, c,  labels = c('A)', 'B)', 'C)'), label.y = 0.75)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-61-2.png)<!-- -->

``` r
save_plot('DGEresults/Venn_Pos_Neg.png',
          ggarrange(a, b, c,  labels = c('A)', 'B)', 'C)'), label.y = 0.75, heights = c(4,1)),
          width = 25, height = 23)
```

    ## png 
    ##   2

# Trait data have some collinearity

``` r
library(ggcorrplot)

# Get matrix for correlation
cordat =  metadata %>%
  select(CG_Avg., Cinn.PRLR, TwoY_RLN, TwoY_length) %>%
  scale() %>%
  as.matrix()

# Calculate p-values
corpmat1 = cor_pmat(cordat)

# Calculate correlation coefficients
cormat1 =cor(cordat)

# Plot. All relationships are significant
p = ggcorrplot(cormat1, type = 'lower', hc.order = T, lab = T, p.mat = corpmat1)

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_files/figure-gfm/unnamed-chunk-62-1.png)<!-- -->

``` r
save_plot('DGEresults/Pheno_corrplot.png', p)
```

    ## png 
    ##   2
