---
title: "Jm3101_v1.0_and_Walnut_2.0"
author: "Houston Saxe"
date: "1/28/2022"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load neccessary libraries
```{r echo=TRUE}
pacman::p_load(rlang, data.table, ggplot2, ggfortify, stringr, dplyr, statmod, tibble, ggpubr, sjPlot, tidyr, tidytext, OmicsAnalyst, tidytable, gridExtra)
```

# Read in and prepare expression data
```{r}
dat = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/SCRI_ROOT_RNAseq_counts_combined_genomes.txt")

dat$GeneID = gsub("LOC", "", dat$GeneID)


head(dat)
```

# Read in metadata
```{r}
metadata = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/R/Phenotyping/SCRI/LongList3_2Y.csv", stringsAsFactors = T)

head(metadata)
```

# Modify metadata for this analysis
```{r}
metadata = data.frame(Sample = colnames(dat)[colnames(dat) != 'GeneID']) %>%
  mutate(Hybrid = as.factor(gsub("\\-\\d$", "", Sample))) %>%
  left_join(metadata, by = c('Hybrid' = 'CAL:_Wip_ID...1'))
  
row.names(metadata) = metadata$Sample

metadata = metadata %>% 
  mutate(PHY_Avg = rowMeans(select(., matches('PCLR|PRLR'))))

head(metadata)

fwrite(metadata, 'SCRI_ROOT_RNAseq_metadata.csv')
```

# Plotting metadata
```{r}
metaLong = metadata %>% 
  select(!Sample) %>% 
  select(Hybrid, CG_Avg., PHY_Avg, TwoY_RLN) %>% 
  distinct() %>% 
  pivot_longer(where(is.numeric), names_to = 'Trait')

p =ggplot(metaLong, aes(reorder_within(Hybrid, value, Trait), value, fill = Hybrid))+
  geom_col()+
  scale_x_reordered()+
  theme(axis.text.x = element_text(angle = 30))+
  facet_wrap(~Trait, scales = 'free', ncol = 4)

print(p)
```
```{r}
# save_plot('DGEresults/metadata_plot.png', p, height = 10, width = 17)
```


# Center and scale predictors to make comparisons more interpretable in results
```{r}
metadata_S = metadata %>%
  mutate_if(is.numeric, scale)
```

# Limma needs GeneIDs as rownames in expression data. Also, colnames of expression data need to match rownames of metadata
```{r}
dat1 = dat %>%
  column_to_rownames(var =  "GeneID") %>%
  as.matrix()

## Do colnames in data match rownames in metadata? If they don't, use match(x,y) produces the order of y required to match the order of x

all(colnames(dat1) == rownames(metadata_S))
# Names match

# If they didn't match, use below code
## What order do rows of metadat need to be in to match colnames of dat1?
# match(colnames(dat1), rownames(metadata))

## Reset rownames
# metadata = metadata[match(colnames(dat1), rownames(metadata)),]

# all(colnames(dat1) == rownames(metadata))
# now they match
```


# Microcarpa annotation
```{r}
annotation_Jm = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jm_x_Jr/Jm_x_Jr_Genomic_annotation.csv")

## Extract everything but class mRNA and other isoforms. This reduces duplication in the data
annotation_Jm = annotation_Jm %>%
  filter(feature != "mRNA", !grepl('\\sX[2-9]$|\\sX1[0-9]$', name)) %>%
  mutate(GeneID = as.character(GeneID)) %>% 
  mutate(Parent_haplotype = "J.microcarpa")

head(annotation_Jm)

```

# Regia annotation
```{r}
annotation_Jr = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jr/Jr_Genomic_annotation.csv")

## Extract everything but class mRNA and other isoforms. This reduces duplication in the data
annotation_Jr = annotation_Jr %>%
  filter(feature != "mRNA", !grepl('\\sX[2-9]$|\\sX1[0-9]$', name)) %>%
  mutate(GeneID = as.character(GeneID)) %>% 
  mutate(Parent_haplotype = "J.regia")

head(annotation_Jr)

```

# Combine Jm and Jr annotations
```{r}
annotation_combined = annotation_Jm %>%
  rbind(annotation_Jr, fill = T)

fwrite(annotation_combined, 'DGEresults/annotation_combined.csv')

BP_anno = annotation_Jm %>% 
  distinct(`Jr-GeneID`) %>% 
  rbind(distinct(annotation_Jr, GeneID), use.names = F) %>% 
  distinct(`Jr-GeneID`) %>% 
  mutate(`Jr-GeneID` = paste0('LOC', `Jr-GeneID`))

fwrite(BP_anno, 'GeneIDs_for_annotation_with_GO.csv')
```

# Create DGEList object
```{r}
library(edgeR)

dds = DGEList(dat1)

dim(dds$counts)

## Calculate library normalization factors (does not do anything to data)
dds = calcNormFactors(dds)

## These are the size factors (normalization factors) for each sample
# dds$samples
```


# Filter by max CPM of 75 (This is used for DGE analysis)
```{r}
d = expression_filter(dds, DGEList = T, FilterFUN = max, FilterThreshold = 75)

## CPM normalized counts of all data
cpm = cpm(dds, prior.count = 2, log = F) 

cpm = cpm %>% 
  data.frame() %>% 
  rownames_to_column(var = 'GeneID') %>%  
  rename_with(~ gsub('X', '', gsub('\\.', '-', .x)))

fwrite(cpm, 'DGEresults/cpm_plotting_data.csv')

## CPM normalized counts of filtered data
cpmd = cpm(d)
```

# Less conservative threshold for looking at expressed genes
```{r}
# d2 = expression_filter(dds, DGEList = T, FilterFUN = mean, FilterThreshold = 20)
# 
# fwrite(as.data.frame(d2$counts), 'DGEresults/CPM_20_filtered.csv', row.names = T)
```

## Cluster visualization
```{r}
## MDS visualization
# plotMDS(d, col = as.numeric(metadata$Hybrid), cex=1)
```

# Plot PCA to look in R
```{r}
pca = plot_pca(cpmd, metadata,
               join_by_name = 'Sample',
               plotting_factors_in = 'col_names',
               plotting_factors_name = Hybrid, 
               x = 'PC1',
               y = 'PC2',
               scale = T, 
               center = T, 
               color = 'Hybrid',
               fill = 'Hybrid',
               plot_type = '2D')
```
```{r}
pca$plot
```


# Plot PCA to save
```{r}
# a = plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'CG_Avg.', fill = 'CG_Avg.', plot_type = '2D', group_by = 'Hybrid')
# 
# b = plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'Cinn.PRLR', fill = 'Cinn.PRLR', plot_type = '2D', group_by = 'Hybrid')
#           
# c = plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'TwoY_RLN', fill = 'TwoY_RLN', plot_type = '2D', group_by = 'Hybrid')
# 
# 
# save_plot('PCA_Hybrids.png', ggarrange(a, b, c, labels = c('A', 'B', 'C'), ncol = 3), base_height = 3, base_width = 12)

```

# PCA boxplot
```{r}
pca = plot_pca(cpmd, metadata,
               join_by_name = 'Sample',
               plotting_factors_in = 'col_names',
               plotting_factors_name = Hybrid, 
               x = 'Hybrid',
               y = 'PC1',
               scale = T, 
               center = T, 
               color = 'Hybrid',
               fill = 'PHY_Avg',
               plot_type = 'boxplot')

pca$plot_dat
```


# PCA scatterplot with fitted line and correlation coefficient
```{r fig.height=10, fig.width=7}
## Available plotting data: 
# [1] "Sample"      "Hybrid"      "CG_Avg."     "CG_Dec."     "Cinn.PCLR"   "Cinn.PRLR"  
#  [7] "Pini.PCLR"   "Pini.PRLR"   "PHY_Dec."    "TwoY_length" "TwoY_RLN"    "NEM_Dec."   # [13] "Unity"       "PHY_Avg" 

a = plot_pca(cpmd, metadata,
             join_by_name = 'Sample',
             plotting_factors_in = 'col_names',
             plotting_factors_name = Hybrid,
             x = 'PC2', y = 'CG_Avg.',
             scale = T,
             center = T,
             color = 'Hybrid',
             plot_type = 'scatter',
             summarise_for_scatter = T)

b = plot_pca(cpmd, metadata,
             join_by_name = 'Sample',
             plotting_factors_in = 'col_names',
             plotting_factors_name = Hybrid,
             x = 'PC5',
             y = 'PHY_Avg',
             scale = T,
             center = T,
             color = 'Hybrid',
             plot_type = 'scatter',
             summarise_for_scatter = T)

c = plot_pca(cpmd, metadata,
             join_by_name = 'Sample',
             plotting_factors_in = 'col_names',
             plotting_factors_name = Hybrid, 
             x = 'PC1',
             y = 'TwoY_RLN',
             scale = T, 
             center = T, 
             color = 'Hybrid',
             plot_type = 'scatter',
             summarise_for_scatter = T)

arr = ggarrange(a$plot + ggtitle('Trait: Crown Gall Disease'),
                b$plot + ggtitle('Trait: Phytophthora Root Rot'),
                c$plot + ggtitle('Trait: Nematode Count'), 
                labels = c('A)', 'B)', 'C)'), ncol = 1)

sjPlot::save_plot('DGEresults/PCA_scatter_CG_PH_NEM.png', arr, height = 35, width = 18)

# save_plot('DGEresults/PCA_scatter_CG_PH_NEM.png', , base_height = 10, base_width = 6)

```


# Heatmap
```{r}
# library(pheatmap)
# 
# cor = cor(cpmd)
# 
# pheatmap(cor, annotation = select(metadata,Hybrid))
```

## Modeling against CG score

# Create design matrix for CG score
```{r}
mm = model.matrix(~CG_Avg., data = metadata_S)

head(mm)
```

## What is voom?

## 1. Counts are transformed to log2 counts per million reads (CPM), where "per million reads" is defined based on the normalization factors we calculated earlier.
## 2. A linear model is fitted to the log2 CPM for each gene, and the residuals are calculated.
## 3. A smoothed curve is fitted to the sqrt(residual standard deviation) by average expression.
## (see red line in plot below)
## 4. The smoothed curve is used to obtain weights for each gene and sample that are passed into limma along with the log2 CPMs.

## More details at "[voom: precision weights unlock linear model analysis tools for RNA-seq read counts](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29)"

# Filtered mean-variance trend CG
```{r}
y <- voom(d, mm, plot = T)
```

# Vs Unfiltered mean-variance trend CG
```{r}
# tmp <- voom(dds, mm, plot = T)
```

# Fitting linear models in limma with random effects CG
```{r}
## Need to tell limma where the within class correlation is coming from
dupcor_CG = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.CG = dupcor_CG$consensus.correlation

consensus.corr.CG

# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.CG) 
```

## The variance characteristics of low expressed genes are different from high expressed genes, if treated the same, the effect is to over represent low expressed genes in the DE list. This is corrected for by the log transformation and voom. However, some genes will have increased or decreased variance that is not a result of low expression, but due to other random factors. We are going to run empirical Bayes to adjust the variance of these genes.

## Empirical Bayes smoothing of standard errors (shifts standard errors that are much larger or smaller than those from other genes towards the average standard error) (see "[Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments](https://www.degruyter.com/doi/10.2202/1544-6115.1027)"
```{r}
BlockFit_CG = eBayes(fit)
```

# Limma results CG
```{r}
res_summaries_CG = BlockFit_CG %>% decideTests() %>% summary()

res_summaries_CG
```

# Table of results CG
```{r}
impCG = topTable(BlockFit_CG, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)

dim(impCG)
## adding Hybrid as a blocking variable reduced DEGs by several thousand
```

# PCA of CG DEGs
```{r}
ids = impCG %>% 
  rownames_to_column(var = 'GeneID') %>% 
  select(GeneID)

PCA_CG = cpmd %>% 
  as.data.frame() %>% 
  rownames_to_column(var = 'GeneID') %>% 
  right_join(ids) %>% 
  column_to_rownames(var = 'GeneID')

pca = plot_pca(PCA_CG, metadata,
               join_by_name = 'Sample',
               plotting_factors_in = 'col_names',
               plotting_factors_name = Group, 
               x = 'PC1',
               y = 'PC2',
               scale = T, 
               center = T, 
               color = 'CG_Avg.',
               fill = 'CG_Avg.',
               plot_type = '2D')

pca$plot

```


# Merge annotation with results CG
```{r}
impCG = impCG %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impCG)

length(unique(impCG$GeneID))

fwrite(impCG, 'DGEresults/Limma_results_table_CG.csv')
```

# Read in CG results
```{r}
impCG = fread('DGEresults/Limma_results_table_CG.csv')
```

# Positive CG DEGs for GO
```{r}
CG_GO_pos_Jm = impCG %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(CG_GO_pos_Jm$GeneID))

CG_GO_pos_both = impCG %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(CG_GO_pos_Jm)

length(unique(CG_GO_pos_both$GeneID))

head(CG_GO_pos_both)

fwrite(CG_GO_pos_both, "DGEresults/CG_DEGs_pos_GO.csv")

```

# Expression from alleles is causing duplication in GO results which means data loss as GO removes duplicates. However, these are not true duplicates.

# Maybe analyze separately and try to recombine?
```{r}
# CG_GO_pos_Jm = impCG %>%
#   filter(logFC > 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`) %>%
#   mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
#   rename(GeneID = `Jr-GeneID`)
# 
# 
# 
# CG_GO_pos_Jr= impCG %>% 
#   filter(logFC > 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID) %>%
#   mutate(GeneID = paste0("LOC", GeneID))
# 
# length(unique(CG_GO_pos_Jr$GeneID)) + length(unique(CG_GO_pos_Jm$GeneID))
# 
# fwrite(CG_GO_pos_Jm, 'GO_test_CG_pos_Jm.csv')
# 
# 
# fwrite(CG_GO_pos_Jr, 'GO_test_CG_pos_Jr.csv')
```

# Negative CG DEGs for GO
```{r}
CG_GO_neg_Jm = impCG %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(CG_GO_neg_Jm$GeneID))

CG_GO_neg_both = impCG %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(CG_GO_neg_Jm)

length(unique(CG_GO_neg_both$GeneID))

head(CG_GO_neg_both)

fwrite(CG_GO_neg_both, "DGEresults/CG_DEGs_neg_GO.csv")
```

# Expression from alleles is causing duplication in GO results which means data loss as GO removes duplicates. However, these are not true duplicates.
```{r}
# dups_neg_Jm = impCG %>%
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`, Parent_haplotype) %>%
#   rename(GeneID = `Jr-GeneID`)
# 
# dups_neg_both = impCG %>% 
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID, Parent_haplotype) %>% 
#   rbind(dups_neg_Jm)
```


# Maybe analyze separately and try to recombine?
```{r}
# CG_GO_neg_Jm = impCG %>%
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`) %>%
#   mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
#   rename(GeneID = `Jr-GeneID`)
# 
# 
# 
# CG_GO_neg_Jr= impCG %>% 
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID) %>%
#   mutate(GeneID = paste0("LOC", GeneID))
# 
# length(unique(CG_GO_neg_Jr$GeneID)) + length(unique(CG_GO_neg_Jm$GeneID))
# 
# fwrite(CG_GO_neg_Jm, 'GO_test_CG_neg_Jm.csv')
# 
# 
# fwrite(CG_GO_neg_Jr, 'GO_test_CG_neg_Jr.csv')
```


# Fit model for genes associated with PHY score
```{r}
mm = model.matrix(~PHY_Avg, data = metadata_S)

head(mm)
```

# Filtered mean-variance trend PHY
```{r}
y <- voom(d, mm, plot = T)
```

# Fitting linear models in limma with random effects PHY
```{r}
## Need to tell limma where the within class correlation is coming from
dupcor_PHY = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.PHY = dupcor_PHY$consensus.correlation

consensus.corr.PHY

# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.PHY) 

# Ebayes
BlockFit_PHY = eBayes(fit)
```

# Limma results PHY
```{r}
res_summaries_PHY = BlockFit_PHY %>% decideTests() %>% summary()

res_summaries_PHY
```

# Table of results PHY
```{r}
impPHY = topTable(BlockFit_PHY, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)

dim(impPHY)
## adding Hybrid as a blocking variable reduced DEGs by several thousand

head(impPHY)
```

# Merge annotation with results PHY
```{r}
impPHY = impPHY %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impPHY)

fwrite(impPHY, 'DGEresults/Limma_results_table_PHY.csv')

impPHY = fread('DGEresults/Limma_results_table_PHY.csv')
```

# Read in PHY results
```{r}
impPHY = fread('DGEresults/Limma_results_table_PHY.csv')
```

# Positive PHY DEGs for GO
```{r}
PHY_GO_pos_Jm = impPHY %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(PHY_GO_pos_Jm$GeneID))

PHY_GO_pos_both = impPHY %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(PHY_GO_pos_Jm)

length(unique(PHY_GO_pos_both$GeneID))

head(PHY_GO_pos_both)

fwrite(PHY_GO_pos_both, "DGEresults/PHY_DEGs_pos_GO.csv")

```

# Negative PHY DEGs for GO
```{r}
PHY_GO_neg_Jm = impPHY %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(PHY_GO_neg_Jm$GeneID))

PHY_GO_neg_both = impPHY %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(PHY_GO_neg_Jm)

length(unique(PHY_GO_neg_both$GeneID))

head(PHY_GO_neg_both)

fwrite(PHY_GO_neg_both, "DGEresults/PHY_DEGs_neg_GO.csv")
```


# Fit model for genes associated with NEM score
```{r}
mm = model.matrix(~TwoY_RLN, data = metadata_S)

head(mm)
```

# Filtered mean-variance trend NEM
```{r}
y <- voom(d, mm, plot = T)
```

# Fitting linear models in limma with random effects NEM
```{r}
## Need to tell limma where the within class correlation is coming from
dupcor_NEM = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.NEM = dupcor_NEM$consensus.correlation

consensus.corr.NEM

# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.NEM) 

# Ebayes
BlockFit_NEM = eBayes(fit)
```

## Limma results NEM
```{r}
res_summaries_NEM = BlockFit_NEM %>% decideTests() %>% summary()

res_summaries_NEM
```

## Table of results NEM
```{r}
impNEM = topTable(BlockFit_NEM, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)

dim(impNEM)
## adding Hybrid as a blocking variable reduced DEGs by several thousand

head(impNEM)
```

## Merge annotation with results NEM
```{r}
impNEM = impNEM %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impNEM)

fwrite(impNEM, 'DGEresults/Limma_results_table_NEM.csv')

impNEM = fread('DGEresults/Limma_results_table_NEM.csv')
```

# Read in NEM results
```{r}
impNEM = fread('DGEresults/Limma_results_table_NEM.csv')
```


# Positive NEM DEGs for GO
```{r}
NEM_GO_pos_Jm = impNEM %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(NEM_GO_pos_Jm$GeneID))

NEM_GO_pos_both = impNEM %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(NEM_GO_pos_Jm)

length(unique(NEM_GO_pos_both$GeneID))

head(NEM_GO_pos_both)

fwrite(NEM_GO_pos_both, "DGEresults/NEM_DEGs_pos_GO.csv")

```

# Negative NEM DEGs for GO
```{r}
NEM_GO_neg_Jm = impNEM %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(NEM_GO_neg_Jm$GeneID))

NEM_GO_neg_both = impNEM %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(NEM_GO_neg_Jm)

length(unique(NEM_GO_neg_both$GeneID))

head(NEM_GO_neg_both)

fwrite(NEM_GO_neg_both, "DGEresults/NEM_DEGs_neg_GO.csv")
```


# Any common DEGs between all pathosystems?
```{r}
CG = impCG %>%
  select(GeneID, name, logFC, `Jr-GeneID`, Parent_haplotype) %>%
  rename_with(~paste0(., '_CG'))

PHY = impPHY %>%
  select(GeneID, name, logFC, Parent_haplotype) %>%
  rename_with(~paste0(., '_PHY'))

NEM = impNEM %>%
  select(GeneID, name, logFC, Parent_haplotype) %>%
  rename_with(~paste0(., '_NEM'))

impUNITY = CG %>%
  inner_join(PHY, by = c('GeneID_CG' = 'GeneID_PHY')) %>%
  inner_join(NEM, by = c('GeneID_CG' = 'GeneID_NEM'))
```

# Common positive genes across pathosystems for GO
```{r}
UNITY_GO_pos_Jm = impUNITY %>%
  filter(logFC_CG > 0 & logFC_PHY > 0 & logFC_NEM > 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG, `Jr-GeneID_CG`) %>%
  filter(Parent_haplotype_CG == "J.microcarpa") %>%
  select(`Jr-GeneID_CG`) %>%
  mutate(`Jr-GeneID_CG` = paste0("LOC", `Jr-GeneID_CG`)) %>% 
  rename(GeneID = `Jr-GeneID_CG`)

length(unique(UNITY_GO_pos_Jm$GeneID))

UNITY_GO_pos_both = impUNITY %>% 
  filter(logFC_CG > 0 & logFC_PHY > 0 & logFC_NEM > 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG,) %>%
  filter(Parent_haplotype_CG == "J.regia") %>% 
  mutate(GeneID_CG = paste0("LOC", GeneID_CG)) %>% 
  select(GeneID_CG) %>%
  rename(GeneID = GeneID_CG)
  rbind(UNITY_GO_pos_Jm, fill = TRUE)

length(unique(UNITY_GO_pos_both$GeneID))

head(UNITY_GO_pos_both)

fwrite(UNITY_GO_pos_both, "DGEresults/UNITY_DEGs_pos_GO.csv")

```

# Common negative genes across pathosystems for GO
```{r}
UNITY_GO_neg_Jm = impUNITY %>%
  filter(logFC_CG < 0 & logFC_PHY < 0 & logFC_NEM < 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG, `Jr-GeneID_CG`) %>%
  filter(Parent_haplotype_CG == "J.microcarpa") %>%
  select(`Jr-GeneID_CG`) %>%
  mutate(`Jr-GeneID_CG` = paste0("LOC", `Jr-GeneID_CG`)) %>% 
  rename(GeneID = `Jr-GeneID_CG`)

length(unique(UNITY_GO_neg_Jm$GeneID))

UNITY_GO_neg_both = impUNITY %>% 
  filter(logFC_CG < 0 & logFC_PHY < 0 & logFC_NEM < 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG,) %>%
  filter(Parent_haplotype_CG == "J.regia") %>% 
  mutate(GeneID_CG = paste0("LOC", GeneID_CG)) %>% 
  select(GeneID_CG) %>%
  rename(GeneID = GeneID_CG)
  rbind(UNITY_GO_neg_Jm, fill = TRUE)

length(unique(UNITY_GO_neg_both$GeneID))

head(UNITY_GO_neg_both)

fwrite(UNITY_GO_neg_both, "DGEresults/UNITY_DEGs_neg_GO.csv")
```



# Summary of combined unique DEGs
```{r}
c(length(unique(UNITY_GO_pos_both)), length(unique(UNITY_GO_neg_both$Gen)))
```

# Summaries of each analysis
```{r}
summary = cbind(res_summaries_CG,res_summaries_PHY, res_summaries_NEM) %>% 
  as.data.frame() %>% 
  select(!`(Intercept)`) %>% 
  rename(PHY_Avg. = PHY_Avg) %>% 
  rownames_to_column(var = 'Association') %>% 
  mutate(Association = gsub('Down', 'Negative', gsub('Up', 'Positive', Association))) %>% 
  rename_with(~ str_extract(.x, 'Association|CG|PHY|RLN')) %>% 
  rename_with(~ gsub('RLN', 'NEM', .x))

head(summary)

write.csv(summary, 'DGEresults/Results_summaries.csv')
```


# Venn Diagram positive genes
```{r}
CG_pos = impCG %>%
  filter(logFC >= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)
  

PHY_pos = impPHY %>%
  filter(logFC >= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)

NEM_pos = impNEM %>%
  filter(logFC >= 0) %>% 
  distinct(GeneID)  %>% 
  pull(GeneID)

venn_pos = list(CG_pos = CG_pos, PHY_pos = PHY_pos, NEM_pos = NEM_pos)

```

# Venn Diagram negative genes
```{r}
CG_neg = impCG %>%
  filter(logFC <= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)
  

PHY_neg = impPHY %>%
  filter(logFC <= 0) %>% 
  distinct(GeneID) %>% 
  pull(GeneID)

NEM_neg = impNEM %>%
  filter(logFC <= 0) %>% 
  distinct(GeneID)  %>% 
  pull(GeneID)

venn_neg = list(CG_neg = CG_neg, PHY_neg = PHY_neg, NEM_neg = NEM_neg)

```

# Plot and save venn diagram figure with results summaries
```{r}
library(ggvenn)

a = ggvenn(venn_pos)

b = ggvenn(venn_neg)

p = tableGrob(summary)

c = grid.arrange(p)

ggarrange(a, b, c,  labels = c('A)', 'B)', 'C)'), label.y = 0.75)


save_plot('DGEresults/Venn_Pos_Neg.png',
          ggarrange(a, b, c,  labels = c('A)', 'B)', 'C)'), label.y = 0.75, heights = c(4,1)),
          width = 25, height = 23)
```




# Trait data have some collinearity
```{r}
library(ggcorrplot)

# Get matrix for correlation
cordat =  metadata %>%
  select(CG_Avg., Cinn.PRLR, TwoY_RLN, TwoY_length) %>%
  scale() %>%
  as.matrix()

# Calculate p-values
corpmat1 = cor_pmat(cordat)

# Calculate correlation coefficients
cormat1 =cor(cordat)

# Plot. All relationships are significant
p = ggcorrplot(cormat1, type = 'lower', hc.order = T, lab = T, p.mat = corpmat1)

p

save_plot('DGEresults/Pheno_corrplot.png', p)
```


