Jm3101\_v1.0\_and\_Walnut\_2.0\_ref\_MS1s\_only
================
Houston Saxe
1/28/2022

## Load neccessary libraries

``` r
pacman::p_load(data.table, ggplot2, ggfortify, stringr, rtracklayer, ape, GEOquery, Biobase, BiocManager, dplyr, statmod, tidytable, tibble, ggpubr, cowplot, tidyr, tidytext)
```

    ## Warning: unable to access index for repository http://www.stats.ox.ac.uk/pub/RWin/bin/windows/contrib/4.1:
    ##   cannot open URL 'http://www.stats.ox.ac.uk/pub/RWin/bin/windows/contrib/4.1/PACKAGES'

    ## Warning: cannot remove prior installation of package 'rlang'

    ## Warning in file.copy(savedcopy, lib, recursive = TRUE):
    ## problem copying C:\Users\hsaxe\OneDrive\Documents\R\win-
    ## library\4.1\00LOCK\rlang\libs\x64\rlang.dll to C:
    ## \Users\hsaxe\OneDrive\Documents\R\win-library\4.1\rlang\libs\x64\rlang.dll:
    ## Permission denied

    ## Warning: restored 'rlang'

    ## Warning in pacman::p_load(data.table, ggplot2, ggfortify, stringr, rtracklayer, : Failed to install/load:
    ## tidytable

``` r
## Helpful function for quick viewing of wide data
left = function(x, n = 10){head(x[,1:n])}
```

# Prepare expression data

``` r
## Read in data
dat = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/SCRI_ROOT_RNAseq_counts_combined_genomes.txt")

dat$GeneID = gsub("LOC", "", dat$GeneID)

dat = dat %>% 
  select(contains(c('GeneID', 'MS1')))

head(dat)
```

    ##       GeneID MS1-122-1 MS1-122-2 MS1-122-3 MS1-122-4 MS1-122-5 MS1-122-6
    ## 1: 121242473        23        29        50        26        54        23
    ## 2: 121268866         0         0         0         0         0         0
    ## 3: 121241406       622       674       618       233       986       368
    ## 4: 121250192        80       123        91        52       176        70
    ## 5: 121241144        13        17        20        13        32        11
    ## 6: 121255391       181       181       253        69       259       162
    ##    MS1-36-1 MS1-36-2 MS1-36-3 MS1-36-4 MS1-36-5 MS1-36-6 MS1-41-1 MS1-41-2
    ## 1:       13       11       15       13       12       33        9       14
    ## 2:        0        0        0        0        0        0        0        0
    ## 3:      787      342     1020      531      491      714      535      231
    ## 4:      153       91      228      148      129      173      101       54
    ## 5:        3       15       20        3        1       17       10        2
    ## 6:      331       76      386      180       14      126      115       88
    ##    MS1-41-3 MS1-41-4 MS1-41-5 MS1-41-6 MS1-56-1 MS1-56-2 MS1-56-3 MS1-56-4
    ## 1:        9        7       17        7       35       21       25       10
    ## 2:        0        0        0        0        0        0        0        0
    ## 3:      454      355      618      331     1122     1017      846      828
    ## 4:       82       73      107       75      196      242      156      152
    ## 5:        7        9        2        6        1        1        1        2
    ## 6:      181      122      235      162       43       20       77       26
    ##    MS1-56-5 MS1-56-6
    ## 1:        0       30
    ## 2:        0        0
    ## 3:      155     1116
    ## 4:       46      247
    ## 5:        1        0
    ## 6:       11       15

``` r
# Data has unnecessary attributes. Also, samples need to be column names
```

# Read in metadata

``` r
## Read in metadata
metadata = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/R/Phenotyping/SCRI/LongList3_2Y.csv", stringsAsFactors = T)

head(metadata)
```

    ##    CAL:_Wip_ID...1 CG_Avg. CG_Dec. Cinn.PCLR Cinn.PRLR Pini.PCLR Pini.PRLR
    ## 1:            I8-1    2.40       D         7        30        81        25
    ## 2:            I9-2    2.30       D        22        21        63        14
    ## 3:           Vlach    3.40       K        72        79        59        39
    ## 4:           VX211    3.10       K        57        65        64        44
    ## 5:          29JM-4    3.10       D        24        17        68        31
    ## 6:          29JM-8    2.75       D        55        50        35        34
    ##                  PHY_Dec. TwoY_length TwoY_RLN NEM_Dec. Unity
    ## 1:                      D      124.50      5.0        D D-D-D
    ## 2:                      D      139.75     58.5        D D-D-D
    ## 3:                   SAVE      170.57    112.2        K K-K-K
    ## 4:                   SAVE      133.30     87.4        K K-K-K
    ## 5:                      D       70.10     14.8        D D-D-D
    ## 6: save due to in trials?       77.50     18.7        K D-K-K

# Modify metadata for this experiment

``` r
#
metadata = data.frame(Sample = colnames(dat)[colnames(dat) != 'GeneID']) %>%
  mutate(Hybrid = as.factor(gsub("\\-\\d$", "", Sample))) %>%
  left_join(metadata, by = c('Hybrid' = 'CAL:_Wip_ID...1'))
  
row.names(metadata) = metadata$Sample

head(metadata)
```

    ##              Sample  Hybrid CG_Avg. CG_Dec. Cinn.PCLR Cinn.PRLR Pini.PCLR
    ## MS1-122-1 MS1-122-1 MS1-122     3.3       D        49        42        50
    ## MS1-122-2 MS1-122-2 MS1-122     3.3       D        49        42        50
    ## MS1-122-3 MS1-122-3 MS1-122     3.3       D        49        42        50
    ## MS1-122-4 MS1-122-4 MS1-122     3.3       D        49        42        50
    ## MS1-122-5 MS1-122-5 MS1-122     3.3       D        49        42        50
    ## MS1-122-6 MS1-122-6 MS1-122     3.3       D        49        42        50
    ##           Pini.PRLR PHY_Dec. TwoY_length TwoY_RLN NEM_Dec. Unity
    ## MS1-122-1        18        D         173     68.9        K D-D-K
    ## MS1-122-2        18        D         173     68.9        K D-D-K
    ## MS1-122-3        18        D         173     68.9        K D-D-K
    ## MS1-122-4        18        D         173     68.9        K D-D-K
    ## MS1-122-5        18        D         173     68.9        K D-D-K
    ## MS1-122-6        18        D         173     68.9        K D-D-K

# Plotting metadata

``` r
metaLong = metadata %>% 
  select(!Sample) %>% 
  distinct() %>% 
  pivot_longer(where(is.numeric), names_to = 'Trait')

p =ggplot(metaLong, aes(reorder_within(Hybrid, value, Trait), value, fill = Hybrid))+
  geom_col()+
  scale_x_reordered()+
  theme(axis.text.x = element_text(angle = 30))+
  facet_wrap(~Trait, scales = 'free', ncol = 4)
p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-5-1.png)<!-- -->

``` r
save_plot('metadata_plot.png', p, base_height = 5, base_width = 10)
```

# Center and scale predictors to make comparisons more interpretable in results

``` r
#
metadata_S = metadata %>%
  mutate_if(is.numeric, scale)
```

# Limma needs GeneIDs as rownames in expression data. Also, colnames of expression data need to match rownames of metadata

``` r
dat1 = dat %>%
  column_to_rownames(var =  "GeneID") %>%
  as.matrix()

## Do colnames in data match rownames in metadata? If they don't, use match(x,y) produces the order of y required to match the order of x

all(colnames(dat1) == rownames(metadata_S))
```

    ## [1] TRUE

``` r
# Names match

# If they didn't match, use below code
## What order do rows of metadat need to be in to match colnames of dat1?
# match(colnames(dat1), rownames(metadata))

## Reset rownames
# metadata = metadata[match(colnames(dat1), rownames(metadata)),]

# all(colnames(dat1) == rownames(metadata))
# now they match
```

# Microcarpa annotation

``` r
annotation_Jm = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jm_x_Jr/Jm_x_Jr_Genomic_annotation_12_16_2021.csv")

## Extract everything but class mRNA and other isoforms. This reduces duplication in the data
annotation_Jm = annotation_Jm %>%
  filter(feature != "mRNA", !grepl('\\sX[2-9]$|\\sX1[0-9]$', name)) %>%
  mutate(GeneID = as.character(GeneID)) %>% 
  mutate(Parent_haplotype = "J.microcarpa")

head(annotation_Jm)
```

    ##    feature        class chromosome  start    end
    ## 1:   ncRNA       lncRNA         1D      7   1181
    ## 2:   ncRNA       lncRNA         1D  14257  15054
    ## 3:     CDS with_protein         1D  27395  59712
    ## 4:     CDS with_protein         1D  85492 108529
    ## 5:     CDS with_protein         1D 137771 144108
    ## 6:     CDS with_protein         1D 164081 167500
    ##                                                         name    GeneID
    ## 1:                              uncharacterized LOC121242473 121242473
    ## 2:                              uncharacterized LOC121268866 121268866
    ## 3: LOW QUALITY PROTEIN: uncharacterized protein LOC121241406 121241406
    ## 4:          membrane-bound O-acyltransferase gup1 isoform X1 121250192
    ## 5:                  pectin acetylesterase 12-like isoform X1 121241144
    ## 6:                                  titin homolog isoform X1 121255391
    ##    product_accession Jr-GeneID Jr-chromosome           Subcell_loc Score
    ## 1:    XR_005935900.1        NA                                        NA
    ## 2:    XR_005941271.1        NA                                        NA
    ## 3:    XP_040995100.1 108981175             7             C:nucleus  1.00
    ## 4:    XP_041005135.1 108980963             7 C:endomembrane system  0.92
    ## 5:    XP_040994724.1 108980844             7 C:extracellular space  0.77
    ## 6:    XP_041011626.1 109018740             7             C:nucleus  1.00
    ##    Parent_haplotype
    ## 1:     J.microcarpa
    ## 2:     J.microcarpa
    ## 3:     J.microcarpa
    ## 4:     J.microcarpa
    ## 5:     J.microcarpa
    ## 6:     J.microcarpa

# Regia annotation

``` r
annotation_Jr = fread("C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jr/Jr_Genomic_annotation_1_21_2022.csv")

## Extract everything but class mRNA and other isoforms. This reduces duplication in the data
annotation_Jr = annotation_Jr %>%
  filter(feature != "mRNA", !grepl('\\sX[2-9]$|\\sX1[0-9]$', name)) %>%
  mutate(GeneID = as.character(GeneID)) %>% 
  mutate(Parent_haplotype = "J.regia")

head(annotation_Jr)
```

    ##    feature        class chromosome start
    ## 1:     CDS with_protein          1     6
    ## 2:     CDS with_protein          1  7068
    ## 3:     CDS with_protein          1 18754
    ## 4:     CDS with_protein          1 18754
    ## 5:     CDS with_protein          1 18754
    ## 6:     CDS with_protein          1 18754
    ##                                                 name    GeneID
    ## 1:             auxin efflux carrier component 2-like 118344036
    ## 2:             probable WRKY transcription factor 17 118348327
    ## 3: KH domain-containing protein HEN4-like isoform X1 108990460
    ## 4: KH domain-containing protein HEN4-like isoform X1 108990460
    ## 5: KH domain-containing protein HEN4-like isoform X1 108990460
    ## 6: KH domain-containing protein HEN4-like isoform X1 108990460
    ##    product_accession           Subcell_loc Score Parent_haplotype
    ## 1:    XP_035539669.1 C:endomembrane system  0.78          J.regia
    ## 2:    XP_035545773.1             C:nucleus  1.00          J.regia
    ## 3:    XP_018819971.1             C:nucleus  1.00          J.regia
    ## 4:    XP_018819972.1             C:nucleus  1.00          J.regia
    ## 5:    XP_035550599.1             C:nucleus  1.00          J.regia
    ## 6:    XP_035550603.1             C:nucleus  1.00          J.regia

# Combine Jm and Jr annotations

``` r
annotation_combined = annotation_Jm %>%
  rbind(annotation_Jr, fill = T)
```

# Create DGEList object

``` r
library(edgeR)
```

    ## Loading required package: limma

    ## 
    ## Attaching package: 'limma'

    ## The following object is masked from 'package:BiocGenerics':
    ## 
    ##     plotMA

``` r
dds = DGEList(dat1)

dim(dds$counts)
```

    ## [1] 67924    24

``` r
## Calculate library normalization factors (does not do anything to data)
dds = calcNormFactors(dds)

## These are the size factors (normalization factors) for each sample
# dds$samples
```

# Filter by max CPM of 75 (This is used for DGE analysis)

``` r
drop = which(apply(cpm(dds), 1, max) < 75)

d = dds[-drop,]

dim(d) # number of genes left
```

    ## [1] 6455   24

``` r
## CPM normalized counts of all data
cpm = cpm(dds, prior.count = 2, log = F) 

## CPM normalized counts of filtered data
cpmd = cpm[-drop,]
```

# Less conservative threshold for looking at expressed genes

``` r
drop = which(apply(cpm(dds), 1, mean) < 20)

d2 = dds[-drop,]

dim(d2) # number of genes left
```

    ## [1] 13312    24

# How many genes are being expressed from each haplotype?

``` r
expressed = d2$counts %>% 
  as.data.frame() %>% 
  rownames_to_column(var = 'GeneID') %>% 
  select(GeneID) %>% 
  left_join(annotation_combined) %>%
  distinct(GeneID, Parent_haplotype) %>% 
  count(Parent_haplotype)
```

    ## Joining, by = "GeneID"

``` r
p = ggplot(expressed, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='Expressed genes')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-14-1.png)<!-- -->

``` r
save_plot('Expression_by_haplotype_CPM20.png', p)
```

## Cluster visualization

``` r
## MDS visualization
# plotMDS(d, col = as.numeric(metadata$Hybrid), cex=1)
```

# Make function for PCA analysis/plotting

``` r
plot_pca = function(dat, metadata, sample_colname = 'Group', sample_names_in = 'row_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'Group', fill = 'Group', plot_type = '2D', group_by = 'Group', summarise_for_scatter = T) {
  
  if(sample_names_in == 'col_names') {
    dat = t(dat) %>%
      as.data.frame() %>%
    mutate(across(everything(), as.numeric)) %>%
    scale(scale = scale, center = center) %>%
    as.matrix()
  } else {
    dat = dat %>%
    as.data.frame() %>% 
    column_to_rownames(var = group) %>% 
    select_if(is.character) %>%
    mutate(across(everything(), as.numeric)) %>%
    scale(scale = scale, center = center) %>%
    as.matrix()
  }
  
  dat = prcomp(dat, scale = F, center = F)
  
  scores = dat$x
  
  eigs = dat$sdev^2 
  
  var_exp = paste('(', formatC((eigs/sum(eigs))*100, format = 'f', digits = 2), '%', ')', sep = '')

  names(var_exp) = colnames(scores)
  
  plot_dat = scores %>%
  data.frame() %>%
  rownames_to_column(var = sample_colname) %>%
  left_join(metadata)
 
  if(plot_type == '2D') {
  p = ggplot(plot_dat, aes(get(x), get(y), color = get(color), group = get(group_by), fill = get(fill)))+
  geom_point()+
  stat_ellipse(geom = 'polygon', alpha = 0.5, level = 0.65)+
    # ggforce::geom_mark_ellipse(aes(fill = get(fill), label = get(group_by)))+
  geom_text(aes(label = get(group_by)), color = 'black', size = 2.5)+
  labs(x = paste(x, var_exp[x]), y = paste(y, var_exp[y]), fill = color, color = color)
  print(p)
  } 
  
  if(plot_type == 'boxplot') {
    p = ggplot(plot_dat, aes(reorder(get(color), get(x)), get(x), fill = get(color)))+
    geom_boxplot()+
    labs(x = color, y = paste(x, var_exp[x]), fill = color)
    print(p)
  }
  
  if(plot_type == 'scatter') {
    if(summarise_for_scatter == T) {
      
      plot_dat = plot_dat %>%
      group_by(get(group_by)) %>%
      summarise_if(is.numeric, mean)  
  
  colnames(plot_dat)[1] = group_by
  
  ggscatter(plot_dat, x = x, y = y, add = 'reg.line', color = color)+
      stat_cor(label.x = 0, label.y = max(plot_dat[y])*1.1)+ 
      stat_smooth(method = 'lm')+
      labs(x = paste(x, var_exp[x]))
      
    } else {
      ggscatter(plot_dat, x = x, y = y, add = 'reg.line', color = color)+
      stat_cor(label.x = 0, label.y = max(plot_dat[y])*1.1)+ 
      stat_smooth(method = 'lm')+
      labs(x = paste(x, var_exp[x]))
    }
    
  }
  
}
```

# Plot PCA to look in R

``` r
plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'Hybrid', fill = 'Hybrid', plot_type = '2D', group_by = 'Hybrid')
```

    ## Joining, by = "Sample"

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-17-1.png)<!-- -->

# Plot PCA to save

``` r
png('PCA_Hybrids.png', width = 18, height = 16, units = 'cm', res = 300)

plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'Hybrid', fill = 'Hybrid', plot_type = '2D', group_by = 'Hybrid')
```

    ## Joining, by = "Sample"

``` r
dev.off()
```

    ## png 
    ##   2

# PCA boxplot

``` r
# plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC2', y = 'PC2', scale = T, center = T, color = 'Cinn.PRLR', plot_type = 'boxplot')
```

# PCA scatterplot with fitted line and correlation coefficient

``` r
## Available plotting data: [1] "Hybrid"      "Sample"      "CG_Avg."     "CG_Dec."     "Cinn.PCLR"   "Cinn.PRLR"  
 # [7] "Pini.PCLR"   "Pini.PRLR"   "PHY_Dec."    "TwoY_length" "TwoY_RLN"    "NEM_Dec."   
# [13] "Unity"
p = plot_pca(cpmd, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'CG_Avg.', scale = T, center = T, color = 'Hybrid', plot_type = 'scatter', group_by = 'Hybrid', summarise_for_scatter = T)+
  ggtitle('Trait: Crown Gall Disease')
```

    ## Joining, by = "Sample"

``` r
p
```

    ## `geom_smooth()` using formula 'y ~ x'

    ## `geom_smooth()` using formula 'y ~ x'

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-20-1.png)<!-- -->

``` r
save_plot('Scatterplot_PC1_CG.png', p)
```

    ## `geom_smooth()` using formula 'y ~ x'
    ## `geom_smooth()` using formula 'y ~ x'

# Heatmap

``` r
# library(pheatmap)
# 
# cor = cor(cpmd)
# 
# pheatmap(cor, annotation = select(metadata,Hybrid))
```

# Create design matrix for CG score

``` r
mm = model.matrix(~CG_Avg., data = metadata_S)

head(mm)
```

    ##           (Intercept)  CG_Avg.
    ## MS1-122-1           1 1.452745
    ## MS1-122-2           1 1.452745
    ## MS1-122-3           1 1.452745
    ## MS1-122-4           1 1.452745
    ## MS1-122-5           1 1.452745
    ## MS1-122-6           1 1.452745

## What is voom?

# 1. Counts are transformed to log2 counts per million reads (CPM), where “per million reads” is defined based on the normalization factors we calculated earlier.

# 2. A linear model is fitted to the log2 CPM for each gene, and the residuals are calculated.

# 3. A smoothed curve is fitted to the sqrt(residual standard deviation) by average expression.

# (see red line in plot above)

# 4. The smoothed curve is used to obtain weights for each gene and sample that are passed into limma along with the log2 CPMs.

# More details at “[voom: precision weights unlock linear model analysis tools for RNA-seq read counts](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29)”

# Filtered mean-variance trend

``` r
y <- voom(d, mm, plot = T)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-23-1.png)<!-- -->

# Vs Unfiltered mean-variance trend

``` r
# tmp <- voom(dds, mm, plot = T)
```

## 5. Fitting linear models in limma with random effects

``` r
## Need to tell limma where the within class correlation is coming from
dupcor_CG = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.CG = dupcor_CG$consensus.correlation

consensus.corr.CG
```

    ## [1] 0.434176

``` r
# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.CG) 
```

# The variance characteristics of low expressed genes are different from high expressed genes, if treated the same, the effect is to over represent low expressed genes in the DE list. This is corrected for by the log transformation and voom. However, some genes will have increased or decreased variance that is not a result of low expression, but due to other random factors. We are going to run empirical Bayes to adjust the variance of these genes.

# Empirical Bayes smoothing of standard errors (shifts standard errors that are much larger or smaller than those from other genes towards the average standard error) (see “[Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments](https://www.degruyter.com/doi/10.2202/1544-6115.1027)”

``` r
BlockFit_CG = eBayes(fit)
```

# Limma results

``` r
res_summaries_CG = BlockFit_CG %>% decideTests() %>% summary()

res_summaries_CG
```

    ##        (Intercept) CG_Avg.
    ## Down             1    1945
    ## NotSig         115    3785
    ## Up            6339     725

# Table of results

``` r
impCG = topTable(BlockFit_CG, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)
```

    ## Removing intercept from test coefficients

``` r
dim(impCG)
```

    ## [1] 2670    7

``` r
## adding Hybrid as a blocking variable reduced DEGs by several thousand
```

# PCA of CG DEGs

``` r
ids = impCG %>% 
  rownames_to_column(var = 'GeneID') %>% 
  select(GeneID)

PCA_CG = cpmd %>% 
  as.data.frame() %>% 
  rownames_to_column(var = 'GeneID') %>% 
  right_join(ids) %>% 
  column_to_rownames(var = 'GeneID')
```

    ## Joining, by = "GeneID"

``` r
plot_pca(PCA_CG, metadata, sample_colname = 'Sample', sample_names_in = 'col_names', x = 'PC1', y = 'PC2', scale = T, center = T, color = 'CG_Avg.', fill = 'CG_Avg.', plot_type = '2D', group_by = 'Hybrid')
```

    ## Joining, by = "Sample"

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-29-1.png)<!-- -->

# Merge annotation with results

``` r
impCG = impCG %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impCG)
```

    ##      GeneID     logFC    AveExpr         t      P.Value    adj.P.Val        B
    ## 1 109016198 -6.542261  1.8501302 -6.505399 6.199358e-07 1.215702e-04 5.430395
    ## 2 109017709 -5.686888  5.0164589 -6.917797 2.180009e-07 7.406292e-05 6.725269
    ## 3 109004718 -5.442998  9.3770746 -5.749991 4.397483e-06 2.689897e-04 4.258588
    ## 4 109021176 -5.244791  0.6362362 -7.030110 1.645567e-07 6.638836e-05 5.591110
    ## 5 109022106 -4.892365  0.9183470 -4.974968 3.423363e-05 7.015177e-04 1.879880
    ## 6 121237960 -4.883703 16.1279239 -7.332406 7.777680e-08 4.183744e-05 8.099022
    ##           R feature        class chromosome    start      end
    ## 1 0.7170025     CDS with_protein          8 25240305       NA
    ## 2 0.7380441     CDS with_protein         15  5713036       NA
    ## 3 0.6726981     CDS with_protein          7 17915184       NA
    ## 4 0.7434279     CDS with_protein         11 30074105       NA
    ## 5 0.6182575     CDS with_protein         10  8877504       NA
    ## 6 0.7572304     CDS with_protein         1D 17558112 17559107
    ##                                               name product_accession Jr-GeneID
    ## 1                               dehydrin DHN1-like    XP_035549466.1        NA
    ## 2             uncharacterized protein LOC109017709    XP_035541988.1        NA
    ## 3         late embryogenesis abundant protein D-29    XP_018838898.2        NA
    ## 4 E3 ubiquitin-protein ligase UPL1-like isoform X1    XP_018859300.1        NA
    ## 5                        cytochrome P450 78A7-like    XP_018860457.1        NA
    ## 6    late embryogenesis abundant protein D-29-like    XP_040990832.1 109004718
    ##   Jr-chromosome           Subcell_loc Score Parent_haplotype
    ## 1          <NA>             C:nucleus  1.00          J.regia
    ## 2          <NA>             C:nucleus  1.00          J.regia
    ## 3          <NA> C:extracellular space  0.82          J.regia
    ## 4          <NA> C:extracellular space  0.60          J.regia
    ## 5          <NA> C:endomembrane system  0.78          J.regia
    ## 6             7 C:extracellular space  0.80     J.microcarpa

``` r
length(unique(impCG$GeneID))
```

    ## [1] 2670

``` r
fwrite(impCG, 'Limma_results_table_CG.csv')

# impCG = fread('Limma_results_table_CG.csv')
```

## Total CG DEGs by haplotype

``` r
ASEsums = impCG %>% distinct(GeneID, Parent_haplotype) %>% count(Parent_haplotype)

p = ggplot(ASEsums, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('Total CG DEGs')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-31-1.png)<!-- -->

``` r
save_plot('impCG_both_EBH.png', p)
```

# Positive CG DEGs for GO

``` r
CG_GO_pos_Jm = impCG %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(CG_GO_pos_Jm$GeneID))
```

    ## [1] 317

``` r
CG_GO_pos_both = impCG %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(CG_GO_pos_Jm)

length(unique(CG_GO_pos_both$GeneID))
```

    ## [1] 530

``` r
head(CG_GO_pos_both)
```

    ##         GeneID
    ## 1 LOC108987644
    ## 2 LOC108985473
    ## 3 LOC109002793
    ## 4 LOC108989061
    ## 5 LOC109006610
    ## 6 LOC109018703

``` r
fwrite(CG_GO_pos_both, "CG_DEGs_pos_GO.csv")
```

# Maybe analyze separately and try to recombine?

``` r
# CG_GO_pos_Jm = impCG %>%
#   filter(logFC > 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`) %>%
#   mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
#   rename(GeneID = `Jr-GeneID`)
# 
# 
# 
# CG_GO_pos_Jr= impCG %>% 
#   filter(logFC > 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID) %>%
#   mutate(GeneID = paste0("LOC", GeneID))
# 
# length(unique(CG_GO_pos_Jr$GeneID)) + length(unique(CG_GO_pos_Jm$GeneID))
# 
# fwrite(CG_GO_pos_Jm, 'GO_test_CG_pos_Jm.csv')
# 
# 
# fwrite(CG_GO_pos_Jr, 'GO_test_CG_pos_Jr.csv')
```

# Negative CG DEGs for GO

``` r
CG_GO_neg_Jm = impCG %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(CG_GO_neg_Jm$GeneID))
```

    ## [1] 962

``` r
CG_GO_neg_both = impCG %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(CG_GO_neg_Jm)

length(unique(CG_GO_neg_both$GeneID))
```

    ## [1] 1415

``` r
head(CG_GO_neg_both)
```

    ##         GeneID
    ## 1 LOC109016198
    ## 2 LOC109017709
    ## 3 LOC109004718
    ## 4 LOC109021176
    ## 5 LOC109022106
    ## 6 LOC118349596

``` r
fwrite(CG_GO_neg_both, "CG_DEGs_neg_GO.csv")
```

# Expression from alleles is causing duplication in GO results which means data loss as GO removes duplicates. However, these are not true duplicates.

``` r
# dups_neg_Jm = impCG %>%
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`, Parent_haplotype) %>%
#   rename(GeneID = `Jr-GeneID`)
# 
# dups_neg_both = impCG %>% 
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID, Parent_haplotype) %>% 
#   rbind(dups_neg_Jm)
```

# Maybe analyze separately and try to recombine?

``` r
# CG_GO_neg_Jm = impCG %>%
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
#   filter(Parent_haplotype == "J.microcarpa") %>%
#   select(`Jr-GeneID`) %>%
#   mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
#   rename(GeneID = `Jr-GeneID`)
# 
# 
# 
# CG_GO_neg_Jr= impCG %>% 
#   filter(logFC < 0) %>%
#   distinct(GeneID, Parent_haplotype) %>%
#   filter(Parent_haplotype == "J.regia") %>% 
#   select(GeneID) %>%
#   mutate(GeneID = paste0("LOC", GeneID))
# 
# length(unique(CG_GO_neg_Jr$GeneID)) + length(unique(CG_GO_neg_Jm$GeneID))
# 
# fwrite(CG_GO_neg_Jm, 'GO_test_CG_neg_Jm.csv')
# 
# 
# fwrite(CG_GO_neg_Jr, 'GO_test_CG_neg_Jr.csv')
```

# CG positive genes EBH

``` r
impCG_pos = impCG %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>% 
  count(Parent_haplotype)

sum(impCG_pos$n)
```

    ## [1] 725

``` r
p = ggplot(impCG_pos, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('CG DEGs (+)')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-37-1.png)<!-- -->

``` r
save_plot('impCG_pos_EBH.png', p)
```

# CG negative genes EBH

``` r
impCG_neg = impCG %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>% 
  count(Parent_haplotype)

p = ggplot(impCG_neg, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('CG DEGs (-)')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-38-1.png)<!-- -->

``` r
save_plot('impCG_neg_EBH.png', p)
```

# CG Subcellular localization by logFC boxplot

``` r
subcell_CG = impCG %>% 
  distinct(GeneID, logFC, Subcell_loc, Parent_haplotype)

p = ggplot(subcell_CG, aes(reorder(Subcell_loc, logFC), logFC))+
  geom_boxplot()+
  theme(axis.text.x = element_text(angle = 30))

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-39-1.png)<!-- -->

# Fit model for genes associated with PHY score

``` r
mm = model.matrix(~Cinn.PRLR, data = metadata_S)

head(mm)
```

    ##           (Intercept) Cinn.PRLR
    ## MS1-122-1           1 0.7679476
    ## MS1-122-2           1 0.7679476
    ## MS1-122-3           1 0.7679476
    ## MS1-122-4           1 0.7679476
    ## MS1-122-5           1 0.7679476
    ## MS1-122-6           1 0.7679476

# Filtered mean-variance trend

``` r
y <- voom(d, mm, plot = T)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-41-1.png)<!-- -->

# Fitting linear models in limma with random effects

``` r
## Need to tell limma where the within class correlation is coming from
dupcor_PHY = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.PHY = dupcor_PHY$consensus.correlation

consensus.corr.PHY
```

    ## [1] 0.5669524

``` r
# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.PHY) 

# Ebayes
BlockFit_PHY = eBayes(fit)
```

# Limma results

``` r
res_summaries_PHY = BlockFit_PHY %>% decideTests() %>% summary()

res_summaries_PHY
```

    ##        (Intercept) Cinn.PRLR
    ## Down             0       870
    ## NotSig         175      5477
    ## Up            6280       108

# Table of results

``` r
impPHY = topTable(BlockFit_PHY, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)
```

    ## Removing intercept from test coefficients

``` r
dim(impPHY)
```

    ## [1] 978   7

``` r
## adding Hybrid as a blocking variable reduced DEGs by several thousand

head(impPHY)
```

    ##               logFC   AveExpr         t      P.Value  adj.P.Val          B
    ## 109004718 -7.504556  9.377075 -3.161211 3.902295e-03 0.03917467 -1.9478293
    ## 121249183 -6.874128  6.595917 -3.163075 3.884270e-03 0.03911682 -2.0544315
    ## 109016198 -6.606870  1.850130 -2.989328 5.957926e-03 0.04475293 -2.7279124
    ## 121268603 -5.571834 11.529381 -2.937189 6.762960e-03 0.04786722 -2.3316091
    ## 121251444 -4.847617  4.097486 -4.654596 7.980904e-05 0.01144816  0.2427923
    ## 108984104 -4.496372 61.234832 -3.572894 1.376890e-03 0.02719727 -1.0099440
    ##                   R
    ## 109004718 0.4470929
    ## 121249183 0.4473038
    ## 109016198 0.4273256
    ## 121268603 0.4212043
    ## 121251444 0.5927366
    ## 108984104 0.4918637

# Merge annotation with results

``` r
impPHY = impPHY %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impPHY)
```

    ##      GeneID     logFC   AveExpr         t      P.Value  adj.P.Val          B
    ## 1 109004718 -7.504556  9.377075 -3.161211 3.902295e-03 0.03917467 -1.9478293
    ## 2 121249183 -6.874128  6.595917 -3.163075 3.884270e-03 0.03911682 -2.0544315
    ## 3 109016198 -6.606870  1.850130 -2.989328 5.957926e-03 0.04475293 -2.7279124
    ## 4 121268603 -5.571834 11.529381 -2.937189 6.762960e-03 0.04786722 -2.3316091
    ## 5 121251444 -4.847617  4.097486 -4.654596 7.980904e-05 0.01144816  0.2427923
    ## 6 108984104 -4.496372 61.234832 -3.572894 1.376890e-03 0.02719727 -1.0099440
    ##           R feature        class chromosome    start     end
    ## 1 0.4470929     CDS with_protein          7 17915184      NA
    ## 2 0.4473038     CDS with_protein         2D   325909  326637
    ## 3 0.4273256     CDS with_protein          8 25240305      NA
    ## 4 0.4212043     CDS with_protein         5S  8096066 8096422
    ## 5 0.5927366     CDS with_protein         2S  5610154 5611522
    ## 6 0.4918637   ncRNA       lncRNA          1  2406665      NA
    ##                                              name product_accession Jr-GeneID
    ## 1        late embryogenesis abundant protein D-29    XP_018838898.2        NA
    ## 2        light-regulated protein 1, chloroplastic    XP_041003832.1 109001297
    ## 3                              dehydrin DHN1-like    XP_035549466.1        NA
    ## 4            uncharacterized protein LOC121268603    XP_041028861.1 108993380
    ## 5 uncharacterized protein LOC121251444 isoform X1    XP_041006639.1 109007630
    ## 6                    uncharacterized LOC108984104    XR_001994969.2        NA
    ##   Jr-chromosome           Subcell_loc Score Parent_haplotype
    ## 1          <NA> C:extracellular space  0.82          J.regia
    ## 2             1         C:chloroplast  1.00     J.microcarpa
    ## 3          <NA>             C:nucleus  1.00          J.regia
    ## 4             4 C:extracellular space  0.70     J.microcarpa
    ## 5            10             C:nucleus  1.00     J.microcarpa
    ## 6          <NA>                          NA          J.regia

``` r
fwrite(impPHY, 'Limma_results_table_PHY.csv')
```

## Total PHY DEGs by haplotype

``` r
ASEsums = impPHY %>% distinct(GeneID, Parent_haplotype) %>% count(Parent_haplotype)

p = ggplot(ASEsums, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('Total PHY DEGs')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-46-1.png)<!-- -->

``` r
save_plot('impPHY_both_EBH.png', p)
```

# Positive PHY DEGs for GO

``` r
PHY_GO_pos_Jm = impPHY %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(PHY_GO_pos_Jm$GeneID))
```

    ## [1] 52

``` r
PHY_GO_pos_both = impPHY %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(PHY_GO_pos_Jm)

length(unique(PHY_GO_pos_both$GeneID))
```

    ## [1] 91

``` r
head(PHY_GO_pos_both)
```

    ##         GeneID
    ## 1 LOC109008986
    ## 2 LOC108985005
    ## 3 LOC108992399
    ## 4 LOC109006717
    ## 5 LOC108997033
    ## 6 LOC109020737

``` r
fwrite(PHY_GO_pos_both, "PHY_DEGs_pos_GO.csv")
```

# Negative PHY DEGs for GO

``` r
PHY_GO_neg_Jm = impPHY %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(PHY_GO_neg_Jm$GeneID))
```

    ## [1] 455

``` r
PHY_GO_neg_both = impPHY %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(PHY_GO_neg_Jm)

length(unique(PHY_GO_neg_both$GeneID))
```

    ## [1] 698

``` r
head(PHY_GO_neg_both)
```

    ##         GeneID
    ## 1 LOC109004718
    ## 2 LOC109016198
    ## 3 LOC108984104
    ## 4 LOC118345731
    ## 5 LOC109021176
    ## 6 LOC108998169

``` r
fwrite(PHY_GO_neg_both, "PHY_DEGs_neg_GO.csv")
```

# PHY positive genes EBH

``` r
impPHY_pos = impPHY %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>% 
  count(Parent_haplotype)

p = ggplot(impPHY_pos, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('PHY DEGs (+)')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-49-1.png)<!-- -->

``` r
save_plot('impPHY_pos_EBH.png', p)
```

# PHY negative genes EBH

``` r
impPHY_neg = impPHY %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>% 
  count(Parent_haplotype)

p = ggplot(impPHY_neg, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('PHY DEGs (-)')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-50-1.png)<!-- -->

``` r
save_plot('impPHY_neg_EBH.png', p)
```

# Fit model for genes associated with NEM score

``` r
mm = model.matrix(~TwoY_RLN, data = metadata_S)

head(mm)
```

    ##           (Intercept) TwoY_RLN
    ## MS1-122-1           1 1.029263
    ## MS1-122-2           1 1.029263
    ## MS1-122-3           1 1.029263
    ## MS1-122-4           1 1.029263
    ## MS1-122-5           1 1.029263
    ## MS1-122-6           1 1.029263

# Filtered mean-variance trend

``` r
y <- voom(d, mm, plot = T)
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-52-1.png)<!-- -->

# Fitting linear models in limma with random effects

``` r
## Need to tell limma where the within class correlation is coming from
dupcor_NEM = duplicateCorrelation(y, mm, block = metadata$Hybrid)

## How correlated are the hybrid replicates on average?
consensus.corr.NEM = dupcor_NEM$consensus.correlation

consensus.corr.NEM
```

    ## [1] 0.4381766

``` r
# lmFit fits a linear model using weighted least squares for each gene:
fit = lmFit(y, design = mm, block = metadata$Hybrid, correlation = consensus.corr.NEM) 

# Ebayes
BlockFit_NEM = eBayes(fit)
```

## Limma results

``` r
res_summaries_NEM = BlockFit_NEM %>% decideTests() %>% summary()

res_summaries_NEM
```

    ##        (Intercept) TwoY_RLN
    ## Down             1     1948
    ## NotSig         122     3863
    ## Up            6332      644

## Table of results

``` r
impNEM = topTable(BlockFit_NEM, sort.by = "logFC", p.value = 0.05, adjust.method = "BH", number = Inf) %>%
  mutate(R = sqrt(t^2/(t^2 + 40)), AveExpr = 2^AveExpr)
```

    ## Removing intercept from test coefficients

``` r
dim(impNEM)
```

    ## [1] 2592    7

``` r
## adding Hybrid as a blocking variable reduced DEGs by several thousand

head(impNEM)
```

    ##               logFC    AveExpr         t      P.Value    adj.P.Val        B
    ## 109016198 -7.291493  1.8501302 -5.885534 3.079250e-06 0.0004622456 3.724739
    ## 109017709 -6.743005  5.0164589 -6.127437 1.639444e-06 0.0003413745 4.928009
    ## 109004718 -6.544578  9.3770746 -5.574462 6.973081e-06 0.0006339611 3.786021
    ## 109021176 -5.386092  0.6362362 -6.582409 5.085675e-07 0.0003353604 4.208761
    ## 118349596 -5.235293  9.5030706 -5.158162 2.101019e-05 0.0009575994 2.799097
    ## 121268603 -5.220237 11.5293807 -5.566640 7.118488e-06 0.0006381922 3.797426
    ##                   R
    ## 109016198 0.6812421
    ## 109017709 0.6958259
    ## 109004718 0.6612195
    ## 109021176 0.7210894
    ## 118349596 0.6320279
    ## 121268603 0.6606969

## Merge annotation with results

``` r
impNEM = impNEM %>%
  rownames_to_column(var = "GeneID") %>%
  left_join(annotation_combined, by = "GeneID")

head(impNEM)
```

    ##      GeneID     logFC    AveExpr         t      P.Value    adj.P.Val        B
    ## 1 109016198 -7.291493  1.8501302 -5.885534 3.079250e-06 0.0004622456 3.724739
    ## 2 109017709 -6.743005  5.0164589 -6.127437 1.639444e-06 0.0003413745 4.928009
    ## 3 109004718 -6.544578  9.3770746 -5.574462 6.973081e-06 0.0006339611 3.786021
    ## 4 109021176 -5.386092  0.6362362 -6.582409 5.085675e-07 0.0003353604 4.208761
    ## 5 118349596 -5.235293  9.5030706 -5.158162 2.101019e-05 0.0009575994 2.799097
    ## 6 121268603 -5.220237 11.5293807 -5.566640 7.118488e-06 0.0006381922 3.797426
    ##           R feature        class chromosome    start     end
    ## 1 0.6812421     CDS with_protein          8 25240305      NA
    ## 2 0.6958259     CDS with_protein         15  5713036      NA
    ## 3 0.6612195     CDS with_protein          7 17915184      NA
    ## 4 0.7210894     CDS with_protein         11 30074105      NA
    ## 5 0.6320279   ncRNA       lncRNA         10  6758733      NA
    ## 6 0.6606969     CDS with_protein         5S  8096066 8096422
    ##                                               name product_accession Jr-GeneID
    ## 1                               dehydrin DHN1-like    XP_035549466.1        NA
    ## 2             uncharacterized protein LOC109017709    XP_035541988.1        NA
    ## 3         late embryogenesis abundant protein D-29    XP_018838898.2        NA
    ## 4 E3 ubiquitin-protein ligase UPL1-like isoform X1    XP_018859300.1        NA
    ## 5                     uncharacterized LOC118349596    XR_004802575.1        NA
    ## 6             uncharacterized protein LOC121268603    XP_041028861.1 108993380
    ##   Jr-chromosome           Subcell_loc Score Parent_haplotype
    ## 1          <NA>             C:nucleus  1.00          J.regia
    ## 2          <NA>             C:nucleus  1.00          J.regia
    ## 3          <NA> C:extracellular space  0.82          J.regia
    ## 4          <NA> C:extracellular space  0.60          J.regia
    ## 5          <NA>                          NA          J.regia
    ## 6             4 C:extracellular space  0.70     J.microcarpa

``` r
fwrite(impNEM, 'Limma_results_table_NEM.csv')
```

## Total NEM DEGs by haplotype

``` r
ASEsums = impNEM %>% distinct(GeneID, Parent_haplotype) %>% count(Parent_haplotype)

p = ggplot(ASEsums, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('Total NEM DEGs')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-57-1.png)<!-- -->

``` r
save_plot('impNEM_both_EBH.png', p)
```

# Positive NEM DEGs for GO

``` r
NEM_GO_pos_Jm = impNEM %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(NEM_GO_pos_Jm$GeneID))
```

    ## [1] 290

``` r
NEM_GO_pos_both = impNEM %>% 
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(NEM_GO_pos_Jm)

length(unique(NEM_GO_pos_both$GeneID))
```

    ## [1] 467

``` r
head(NEM_GO_pos_both)
```

    ##         GeneID
    ## 1 LOC108989061
    ## 2 LOC109011537
    ## 3 LOC109002793
    ## 4 LOC108993193
    ## 5 LOC109019577
    ## 6 LOC108991653

``` r
fwrite(NEM_GO_pos_both, "NEM_DEGs_pos_GO.csv")
```

# Negative NEM DEGs for GO

``` r
NEM_GO_neg_Jm = impNEM %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype, `Jr-GeneID`) %>%
  filter(Parent_haplotype == "J.microcarpa") %>%
  select(`Jr-GeneID`) %>%
  mutate(`Jr-GeneID` = paste0("LOC", `Jr-GeneID`)) %>% 
  rename(GeneID = `Jr-GeneID`)

length(unique(NEM_GO_neg_Jm$GeneID))
```

    ## [1] 971

``` r
NEM_GO_neg_both = impNEM %>% 
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>%
  filter(Parent_haplotype == "J.regia") %>% 
  select(GeneID) %>%
  mutate(GeneID = paste0("LOC", GeneID)) %>% 
  rbind(NEM_GO_neg_Jm)

length(unique(NEM_GO_neg_both$GeneID))
```

    ## [1] 1420

``` r
head(NEM_GO_neg_both)
```

    ##         GeneID
    ## 1 LOC109016198
    ## 2 LOC109017709
    ## 3 LOC109004718
    ## 4 LOC109021176
    ## 5 LOC118349596
    ## 6 LOC109022106

``` r
fwrite(NEM_GO_neg_both, "NEM_DEGs_neg_GO.csv")
```

# NEM positive genes EBH

``` r
impNEM_pos = impNEM %>%
  filter(logFC > 0) %>%
  distinct(GeneID, Parent_haplotype) %>% 
  count(Parent_haplotype)

p = ggplot(impNEM_pos, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('NEM DEGs (+)')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-60-1.png)<!-- -->

``` r
save_plot('impNEM_pos_EBH.png', p)
```

# NEM negative genes EBH

``` r
impNEM_neg = impNEM %>%
  filter(logFC < 0) %>%
  distinct(GeneID, Parent_haplotype) %>% 
  count(Parent_haplotype)

p = ggplot(impNEM_neg, aes(Parent_haplotype, n, fill = Parent_haplotype))+
  geom_col(color = 'black')+
  labs(y='DEGs')+
  ggtitle('NEM DEGs (-)')

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-61-1.png)<!-- -->

``` r
save_plot('impNEM_neg_EBH.png', p)
```

# Any common DEGs between all pathosystems?

``` r
CG = impCG %>%
  select(GeneID, name, logFC, `Jr-GeneID`, Parent_haplotype) %>%
  rename_with(~paste0(., '_CG'))

PHY = impPHY %>%
  select(GeneID, name, logFC, Parent_haplotype) %>%
  rename_with(~paste0(., '_PHY'))

NEM = impNEM %>%
  select(GeneID, name, logFC, Parent_haplotype) %>%
  rename_with(~paste0(., '_NEM'))

impUNITY = CG %>%
  inner_join(PHY, by = c('GeneID_CG' = 'GeneID_PHY')) %>%
  inner_join(NEM, by = c('GeneID_CG' = 'GeneID_NEM'))
```

# Common positive genes across pathosystems for GO

``` r
UNITY_GO_pos_Jm = impUNITY %>%
  filter(logFC_CG > 0 & logFC_PHY > 0 & logFC_NEM > 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG, `Jr-GeneID_CG`) %>%
  filter(Parent_haplotype_CG == "J.microcarpa") %>%
  select(`Jr-GeneID_CG`) %>%
  mutate(`Jr-GeneID_CG` = paste0("LOC", `Jr-GeneID_CG`)) %>% 
  rename(GeneID = `Jr-GeneID_CG`)

length(unique(UNITY_GO_pos_Jm$GeneID))
```

    ## [1] 45

``` r
UNITY_GO_pos_both = impUNITY %>% 
  filter(logFC_CG > 0 & logFC_PHY > 0 & logFC_NEM > 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG,) %>%
  filter(Parent_haplotype_CG == "J.regia") %>% 
  mutate(GeneID_CG = paste0("LOC", GeneID_CG)) %>% 
  select(GeneID_CG) %>%
  rename(GeneID = GeneID_CG)
  rbind(UNITY_GO_pos_Jm, fill = TRUE)
```

    ##          GeneID
    ## 1  LOC109003827
    ## 2  LOC108982593
    ## 3  LOC109015093
    ## 4  LOC109010746
    ## 5  LOC109012310
    ## 6  LOC109001915
    ## 7  LOC109002543
    ## 8  LOC108997930
    ## 9  LOC108999916
    ## 10 LOC108997210
    ## 11 LOC108997940
    ## 12 LOC109007613
    ## 13 LOC108990945
    ## 14 LOC109018844
    ## 15 LOC108990002
    ## 16 LOC109005825
    ## 17 LOC108979956
    ## 18 LOC108981100
    ## 19 LOC109003782
    ## 20 LOC109011698
    ## 21 LOC109000510
    ## 22 LOC109008795
    ## 23 LOC108982574
    ## 24 LOC109012759
    ## 25 LOC109002071
    ## 26 LOC109019074
    ## 27 LOC108983125
    ## 28 LOC109003886
    ## 29 LOC108997102
    ## 30 LOC108985058
    ## 31 LOC109005021
    ## 32 LOC108993925
    ## 33 LOC109011295
    ## 34 LOC108990908
    ## 35 LOC108994817
    ## 36 LOC108997367
    ## 37 LOC109013651
    ## 38 LOC108992210
    ## 39 LOC108989520
    ## 40 LOC109000757
    ## 41 LOC108989877
    ## 42 LOC109005051
    ## 43 LOC118349257
    ## 44 LOC108983614
    ## 45 LOC109012510
    ## 46         TRUE

``` r
length(unique(UNITY_GO_pos_both$GeneID))
```

    ## [1] 46

``` r
head(UNITY_GO_pos_both)
```

    ##         GeneID
    ## 1 LOC108996790
    ## 2 LOC108993986
    ## 3 LOC109020737
    ## 4 LOC108992461
    ## 5 LOC108995160
    ## 6 LOC108997218

``` r
fwrite(UNITY_GO_pos_both, "UNITY_DEGs_pos_GO.csv")
```

# Common negative genes across pathosystems for GO

``` r
UNITY_GO_neg_Jm = impUNITY %>%
  filter(logFC_CG < 0 & logFC_PHY < 0 & logFC_NEM < 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG, `Jr-GeneID_CG`) %>%
  filter(Parent_haplotype_CG == "J.microcarpa") %>%
  select(`Jr-GeneID_CG`) %>%
  mutate(`Jr-GeneID_CG` = paste0("LOC", `Jr-GeneID_CG`)) %>% 
  rename(GeneID = `Jr-GeneID_CG`)

length(unique(UNITY_GO_neg_Jm$GeneID))
```

    ## [1] 440

``` r
UNITY_GO_neg_both = impUNITY %>% 
  filter(logFC_CG < 0 & logFC_PHY < 0 & logFC_NEM < 0) %>%
  distinct(GeneID_CG, Parent_haplotype_CG,) %>%
  filter(Parent_haplotype_CG == "J.regia") %>% 
  mutate(GeneID_CG = paste0("LOC", GeneID_CG)) %>% 
  select(GeneID_CG) %>%
  rename(GeneID = GeneID_CG)
  rbind(UNITY_GO_neg_Jm, fill = TRUE)
```

    ##           GeneID
    ## 1   LOC108993380
    ## 2   LOC109001297
    ## 3          LOCNA
    ## 4   LOC109007630
    ## 5   LOC108989475
    ## 6   LOC108993039
    ## 7   LOC108981955
    ## 8   LOC108989931
    ## 9   LOC109012906
    ## 10  LOC108982143
    ## 11  LOC109001005
    ## 12         LOCNA
    ## 13         LOCNA
    ## 14  LOC108998480
    ## 15  LOC109006840
    ## 16  LOC108982093
    ## 17  LOC109012018
    ## 18  LOC108982084
    ## 19  LOC109003786
    ## 20  LOC109008522
    ## 21  LOC109002934
    ## 22  LOC109021202
    ## 23  LOC108988732
    ## 24  LOC108979561
    ## 25  LOC108984668
    ## 26  LOC108984319
    ## 27  LOC108989646
    ## 28  LOC108984182
    ## 29  LOC108989742
    ## 30  LOC109006543
    ## 31  LOC108996276
    ## 32  LOC108985495
    ## 33  LOC109018407
    ## 34  LOC109002246
    ## 35  LOC108990109
    ## 36  LOC108997381
    ## 37  LOC108999522
    ## 38  LOC109004546
    ## 39  LOC109013436
    ## 40  LOC109006606
    ## 41  LOC109005433
    ## 42  LOC108994407
    ## 43  LOC108983723
    ## 44  LOC109016875
    ## 45  LOC108986174
    ## 46  LOC108995181
    ## 47  LOC109008509
    ## 48  LOC109019756
    ## 49  LOC108995747
    ## 50  LOC108992835
    ## 51  LOC108999631
    ## 52  LOC109010059
    ## 53  LOC109007655
    ## 54  LOC109011702
    ## 55  LOC108998412
    ## 56  LOC108979130
    ## 57  LOC109013742
    ## 58  LOC108989149
    ## 59  LOC109010779
    ## 60  LOC109013567
    ## 61  LOC109004351
    ## 62  LOC109013347
    ## 63  LOC108982222
    ## 64  LOC109022012
    ## 65  LOC109020092
    ## 66  LOC109004890
    ## 67  LOC109018722
    ## 68  LOC108981203
    ## 69  LOC108996821
    ## 70  LOC109013638
    ## 71  LOC108997086
    ## 72  LOC108991266
    ## 73  LOC108999584
    ## 74         LOCNA
    ## 75  LOC108998827
    ## 76  LOC109016625
    ## 77  LOC108983500
    ## 78  LOC109008844
    ## 79  LOC108993691
    ## 80  LOC108981103
    ## 81  LOC108996328
    ## 82  LOC108998645
    ## 83  LOC109006392
    ## 84  LOC108991963
    ## 85  LOC109007373
    ## 86  LOC108992034
    ## 87  LOC109007171
    ## 88  LOC109002331
    ## 89  LOC108989290
    ## 90  LOC108992092
    ## 91  LOC109013810
    ## 92  LOC109001451
    ## 93  LOC108993512
    ## 94  LOC108982117
    ## 95  LOC108998090
    ## 96  LOC108993360
    ## 97  LOC108988864
    ## 98  LOC109005646
    ## 99  LOC109017779
    ## 100 LOC109013856
    ## 101 LOC108993589
    ## 102 LOC109003182
    ## 103        LOCNA
    ## 104 LOC109007725
    ## 105 LOC108983004
    ## 106 LOC109020206
    ## 107 LOC109010491
    ## 108 LOC108992039
    ## 109 LOC108997260
    ## 110 LOC109010735
    ## 111 LOC108995480
    ## 112 LOC109007734
    ## 113 LOC108985286
    ## 114 LOC108992577
    ## 115 LOC109008595
    ## 116 LOC109003842
    ## 117 LOC108998859
    ## 118 LOC109013069
    ## 119 LOC109011631
    ## 120 LOC109003980
    ## 121 LOC109003667
    ## 122 LOC109013071
    ## 123 LOC108981052
    ## 124 LOC108983397
    ## 125 LOC109011635
    ## 126 LOC109002235
    ## 127 LOC108998883
    ## 128 LOC108979952
    ## 129 LOC109014561
    ## 130 LOC108992852
    ## 131 LOC108998109
    ## 132 LOC109001799
    ## 133 LOC109007038
    ## 134 LOC109020143
    ## 135 LOC109006531
    ## 136 LOC109003048
    ## 137 LOC108991371
    ## 138 LOC108985117
    ## 139 LOC108989884
    ## 140 LOC109019613
    ## 141 LOC109004550
    ## 142 LOC109010944
    ## 143 LOC108988697
    ## 144 LOC108991841
    ## 145 LOC108994090
    ## 146 LOC108997972
    ## 147 LOC108998864
    ## 148 LOC108995638
    ## 149 LOC109006960
    ## 150 LOC108988041
    ## 151        LOCNA
    ## 152 LOC109006004
    ## 153 LOC108991875
    ## 154 LOC108983014
    ## 155        LOCNA
    ## 156 LOC108988782
    ## 157 LOC109011878
    ## 158 LOC109012389
    ## 159 LOC109019694
    ## 160 LOC108992206
    ## 161 LOC108986384
    ## 162 LOC109013014
    ## 163 LOC108997061
    ## 164 LOC108990726
    ## 165 LOC109014312
    ## 166 LOC109000761
    ## 167 LOC108989214
    ## 168 LOC108999736
    ## 169 LOC109005306
    ## 170        LOCNA
    ## 171 LOC108987501
    ## 172 LOC109002343
    ## 173 LOC108986275
    ## 174 LOC108995011
    ## 175 LOC108988968
    ## 176 LOC108999289
    ## 177 LOC109004812
    ## 178 LOC108988191
    ## 179 LOC109017709
    ## 180 LOC109007373
    ## 181 LOC109003483
    ## 182 LOC109000197
    ## 183 LOC108986369
    ## 184 LOC108988557
    ## 185 LOC108992981
    ## 186 LOC108995170
    ## 187 LOC109010264
    ## 188 LOC109005757
    ## 189 LOC109000588
    ## 190 LOC108999413
    ## 191 LOC109013127
    ## 192 LOC109011769
    ## 193 LOC109005343
    ## 194        LOCNA
    ## 195 LOC108994288
    ## 196 LOC108999340
    ## 197 LOC109001884
    ## 198 LOC109014169
    ## 199 LOC109002143
    ## 200 LOC108984185
    ## 201 LOC108996455
    ## 202 LOC109007327
    ## 203 LOC108990017
    ## 204 LOC108983287
    ## 205 LOC109012033
    ## 206 LOC108983520
    ## 207 LOC109001168
    ## 208 LOC109010064
    ## 209 LOC109006613
    ## 210 LOC109018855
    ## 211 LOC108984432
    ## 212 LOC108993392
    ## 213 LOC109001794
    ## 214 LOC108999247
    ## 215 LOC108996384
    ## 216 LOC109014195
    ## 217 LOC109013082
    ## 218 LOC109012840
    ## 219 LOC109013907
    ## 220 LOC109000868
    ## 221 LOC109003008
    ## 222 LOC108992839
    ## 223 LOC109003753
    ## 224 LOC109006710
    ## 225 LOC108990241
    ## 226 LOC108989147
    ## 227 LOC108985047
    ## 228 LOC109006347
    ## 229 LOC108993388
    ## 230 LOC109013177
    ## 231 LOC108980851
    ## 232 LOC108984218
    ## 233 LOC109009517
    ## 234 LOC108982179
    ## 235 LOC108980791
    ## 236 LOC109000623
    ## 237 LOC109012431
    ## 238 LOC108989898
    ## 239 LOC109008005
    ## 240 LOC108982656
    ## 241 LOC108989529
    ## 242 LOC109020913
    ## 243 LOC108979666
    ## 244 LOC108983732
    ## 245 LOC108996462
    ## 246        LOCNA
    ## 247 LOC108983415
    ## 248 LOC108985991
    ## 249 LOC109009187
    ## 250 LOC109002655
    ## 251 LOC108996170
    ## 252 LOC109019146
    ## 253 LOC108994271
    ## 254 LOC109004964
    ## 255 LOC108996258
    ## 256 LOC108995197
    ## 257 LOC108986222
    ## 258 LOC108989746
    ## 259 LOC109009323
    ## 260 LOC109014491
    ## 261 LOC108982988
    ## 262 LOC108987972
    ## 263 LOC109008128
    ## 264 LOC108984027
    ## 265 LOC109000709
    ## 266 LOC108981368
    ## 267 LOC108996282
    ## 268 LOC109018985
    ## 269 LOC109013123
    ## 270 LOC109006415
    ## 271 LOC108990471
    ## 272 LOC109009709
    ## 273 LOC109000748
    ## 274 LOC109008641
    ## 275 LOC109005011
    ## 276 LOC109007341
    ## 277 LOC118349544
    ## 278        LOCNA
    ## 279 LOC108998261
    ## 280 LOC108980543
    ## 281 LOC109005468
    ## 282 LOC109000865
    ## 283 LOC109014492
    ## 284 LOC108979055
    ## 285 LOC108981202
    ## 286 LOC109003095
    ## 287 LOC108981407
    ## 288 LOC109013840
    ## 289 LOC108997113
    ## 290 LOC109014310
    ## 291 LOC109019839
    ## 292 LOC109021938
    ## 293 LOC108994687
    ## 294 LOC109006118
    ## 295        LOCNA
    ## 296 LOC109021960
    ## 297 LOC108997386
    ## 298 LOC108986551
    ## 299 LOC109005432
    ## 300 LOC108989168
    ## 301 LOC108987896
    ## 302 LOC108999142
    ## 303 LOC108997859
    ## 304 LOC108992108
    ## 305 LOC108998282
    ## 306 LOC109001448
    ## 307 LOC108990019
    ## 308 LOC108998559
    ## 309 LOC108995647
    ## 310 LOC109011929
    ## 311 LOC109002014
    ## 312 LOC108989201
    ## 313 LOC109012357
    ## 314 LOC108987147
    ## 315 LOC109010749
    ## 316 LOC109005883
    ## 317 LOC109009875
    ## 318 LOC109005439
    ## 319 LOC109007557
    ## 320 LOC109019296
    ## 321 LOC108985530
    ## 322 LOC109013066
    ## 323 LOC109000678
    ## 324 LOC108986652
    ## 325 LOC108981023
    ## 326 LOC108997237
    ## 327 LOC109012785
    ## 328 LOC109000780
    ## 329 LOC108979278
    ## 330 LOC109012889
    ## 331 LOC109007805
    ## 332 LOC108987298
    ## 333 LOC109011892
    ## 334 LOC109021825
    ## 335 LOC108998863
    ## 336 LOC109014339
    ## 337 LOC118343674
    ## 338 LOC109004579
    ## 339 LOC109022244
    ## 340 LOC108992032
    ## 341 LOC109014658
    ## 342 LOC109012766
    ## 343 LOC108993918
    ## 344 LOC108987142
    ## 345 LOC108988731
    ## 346 LOC109000916
    ## 347 LOC108984162
    ## 348 LOC108990907
    ## 349 LOC108991213
    ## 350 LOC109007382
    ## 351 LOC108990414
    ## 352 LOC109019725
    ## 353 LOC109005469
    ## 354 LOC109013525
    ## 355 LOC109015105
    ## 356 LOC109021182
    ## 357 LOC108988183
    ## 358 LOC109018928
    ## 359 LOC108984384
    ## 360 LOC109010996
    ## 361 LOC108982573
    ## 362 LOC109005990
    ## 363 LOC108992024
    ## 364 LOC109020583
    ## 365 LOC118343679
    ## 366 LOC108992143
    ## 367 LOC109009932
    ## 368 LOC109013918
    ## 369 LOC109000516
    ## 370        LOCNA
    ## 371 LOC108998840
    ## 372 LOC109014533
    ## 373 LOC109021983
    ## 374 LOC109003986
    ## 375 LOC108991260
    ## 376 LOC108996410
    ## 377 LOC108999895
    ## 378 LOC109014211
    ## 379 LOC109006350
    ## 380 LOC109009005
    ## 381 LOC109013954
    ## 382 LOC109009438
    ## 383 LOC109013138
    ## 384 LOC109009728
    ## 385 LOC109006942
    ## 386 LOC109004403
    ## 387 LOC109011240
    ## 388 LOC108988140
    ## 389 LOC108985722
    ## 390 LOC108995657
    ## 391 LOC109020673
    ## 392 LOC108980005
    ## 393 LOC109004233
    ## 394 LOC109010191
    ## 395 LOC109020299
    ## 396 LOC109010863
    ## 397 LOC109009471
    ## 398 LOC109020427
    ## 399 LOC108990620
    ## 400 LOC108979338
    ## 401 LOC108987487
    ## 402 LOC108985414
    ## 403 LOC109009593
    ## 404 LOC109005028
    ## 405 LOC109006318
    ## 406 LOC118344194
    ## 407 LOC108995974
    ## 408 LOC108987897
    ## 409 LOC108990852
    ## 410 LOC108987139
    ## 411        LOCNA
    ## 412 LOC109001572
    ## 413 LOC108990616
    ## 414 LOC108987299
    ## 415 LOC109009507
    ## 416 LOC108993632
    ## 417 LOC109014105
    ## 418 LOC109003529
    ## 419 LOC109002666
    ## 420 LOC109005208
    ## 421 LOC108981085
    ## 422 LOC109009402
    ## 423 LOC109007458
    ## 424 LOC108981678
    ## 425 LOC108985848
    ## 426 LOC108987681
    ## 427 LOC108995577
    ## 428 LOC108991872
    ## 429 LOC109011306
    ## 430 LOC109012760
    ## 431 LOC108987669
    ## 432 LOC108997239
    ## 433 LOC109012820
    ## 434 LOC108988966
    ## 435 LOC108985860
    ## 436 LOC108984977
    ## 437 LOC109005156
    ## 438 LOC108986766
    ## 439 LOC109007024
    ## 440 LOC109009498
    ## 441 LOC109001305
    ## 442 LOC108984879
    ## 443 LOC108986066
    ## 444 LOC108994752
    ## 445 LOC108986719
    ## 446 LOC108987177
    ## 447 LOC108987071
    ## 448 LOC108997553
    ## 449 LOC108997371
    ## 450 LOC108983948
    ## 451 LOC109006335
    ## 452 LOC109006779
    ## 453 LOC108984797
    ## 454 LOC108993579
    ## 455         TRUE

``` r
length(unique(UNITY_GO_neg_both$GeneID))
```

    ## [1] 386

``` r
head(UNITY_GO_neg_both)
```

    ##         GeneID
    ## 1 LOC109016198
    ## 2 LOC109004718
    ## 3 LOC109021176
    ## 4 LOC109014998
    ## 5 LOC108984104
    ## 6 LOC108998169

``` r
fwrite(UNITY_GO_neg_both, "UNITY_DEGs_neg_GO.csv")
```

# Summary of combined unique DEGs

``` r
c(length(unique(UNITY_GO_pos_both)), length(unique(UNITY_GO_neg_both$Gen)))
```

    ## [1]   1 386

# Summaries of each analysis

``` r
summary = cbind(res_summaries_CG,res_summaries_PHY, res_summaries_NEM) %>% 
  as.data.frame() %>% 
  select(!`(Intercept)`)

head(summary)
```

    ##        CG_Avg. Cinn.PRLR TwoY_RLN
    ## Down      1945       870     1948
    ## NotSig    3785      5477     3863
    ## Up         725       108      644

``` r
write.csv(summary, 'Results_summaries.csv')
```

# Venn Diagram

``` r
CG = impCG %>%
  distinct(GeneID) %>% 
  pull(GeneID)
  

PHY = impPHY %>%
  distinct(GeneID) %>% 
  pull(GeneID)

NEM = impNEM %>%
  distinct(GeneID)  %>% 
  pull(GeneID)

venn = list(CG = CG, PHY = PHY, NEM = NEM)
```

# Plot and save venn diagram

``` r
library(ggVennDiagram)
library(ggvenn)
```

    ## Loading required package: grid

``` r
p = ggvenn(venn)


save_plot('Venn_Up_Down.png', p, base_width = 6, base_height = 6)
```

# Trait data have some collinearity

``` r
library(ggcorrplot)

# Get matrix for correlation
cordat =  metadata %>%
  select(CG_Avg., Cinn.PRLR, TwoY_RLN, TwoY_length) %>%
  scale() %>%
  as.matrix()

# Calculate p-values
corpmat1 = cor_pmat(cordat)

# Calculate correlation coefficients
cormat1 =cor(cordat)

# Plot. All relationships are significant
p = ggcorrplot(cormat1, type = 'lower', hc.order = T, lab = T, p.mat = corpmat1)

p
```

![](Jm3101_v1.0_and_Walnut_2.0_ref_MS1s_only_files/figure-gfm/unnamed-chunk-69-1.png)<!-- -->

``` r
save_plot('Pheno_corrplot.png', p)
```

# Troubleshooting space for pca function

``` r
# dat = cpmd
# scale = T
# center = T
# metadata = metadata
# sample_colname = 'Sample'
# group_by = 'Hybrid'
# color = 'Hybrid'
# x = 'PC2'
# y = 'Cinn.PRLR'
# 
# dat = t(dat) %>%
#       as.data.frame() %>%
#     mutate(across(everything(), as.numeric)) %>%
#     scale(scale = scale, center = center) %>%
#     as.matrix()
# 
# dat = prcomp(dat, scale = F, center = F)
#   
#   scores = dat$x
#   
#   eigs = dat$sdev^2 
#   
#   var_exp = paste('(', formatC((eigs/sum(eigs))*100, format = 'f', digits = 2), '%', ')', sep = '')
# 
#   names(var_exp) = colnames(scores)
#   
#   plot_dat = scores %>%
#   data.frame() %>%
#   rownames_to_column(var = sample_colname) %>%
#   left_join(metadata)
#   
#   plot_dat = plot_dat %>%
#       group_by(get(group_by)) %>%
#       summarise_if(is.numeric, mean)  
#   
#   colnames(plot_dat)[1] = group_by
  
  # ggscatter(plot_dat, x = x, y = y, add = 'reg.line', color = color)+
  #     stat_cor()+ 
  #     stat_smooth(method = 'lm')
```

``` r
# Get matrix for correlation
# cordat =  plot_dat %>%
#   select(CG_Avg., Cinn.PRLR, TwoY_RLN, PC1, PC2) %>%
#   scale() %>%
#   as.matrix()
# 
# # Calculate p-values
# corpmat = cor_pmat(cordat)
# 
# # Calculate correlation coefficients
# cormat =cor(cordat)
# 
# # Plot. All relationships are significant
# p = ggcorrplot(cormat, type = 'lower', hc.order = T, lab = T, p.mat = corpmat)
# 
# p

# save_plot('Pheno_Geno_corrplot.png', p)
```
