<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.475">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Houston Saxe">
<meta name="dcterms.date" content="2022-01-28">

<title>SCRI_ROOT_RNAs</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="SCRI_ROOT_RNAs_files/libs/clipboard/clipboard.min.js"></script>
<script src="SCRI_ROOT_RNAs_files/libs/quarto-html/quarto.js"></script>
<script src="SCRI_ROOT_RNAs_files/libs/quarto-html/popper.min.js"></script>
<script src="SCRI_ROOT_RNAs_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="SCRI_ROOT_RNAs_files/libs/quarto-html/anchor.min.js"></script>
<link href="SCRI_ROOT_RNAs_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="SCRI_ROOT_RNAs_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="SCRI_ROOT_RNAs_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="SCRI_ROOT_RNAs_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="SCRI_ROOT_RNAs_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">SCRI_ROOT_RNAs</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Houston Saxe </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 28, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-neccessary-libraries" class="level1">
<h1>Load neccessary libraries</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(base,</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>               edgeR,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>               tidytable,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>               data.table,</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>               dplyr,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>               stats,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>               ggpubr,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>               tidyr,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>               ggplot2,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>               gridExtra,</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>               limma,</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>               OmicsAnalyst,</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>               sjPlot,</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>               tibble,</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>               tidytext,</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>               utils,</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>               statmod,</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>               corrplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-and-prepare-expression-data" class="level1">
<h1>Read in and prepare expression data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/SCRI_ROOT_RNAseq_counts_combined_genomes.txt"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>GeneID <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"LOC"</span>, <span class="st">""</span>, dat<span class="sc">$</span>GeneID)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># dat = dat %&gt;% </span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(!matches('STJM'))</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      GeneID 29JM-11-1 29JM-11-2 29JM-11-3 29JM-11-4 29JM-11-5 29JM-11-6
1: 121242473         6         5         8        10        16        20
2: 121268866         0         0         0         0         0         0
3: 121241406       210        70       274       445       564       409
4: 121250192        38        12        40       111       109        76
5: 121241144         8         6        17        13        33        13
6: 121255391        76        28       146       196       206       102
   JMS-12-1 JMS-12-2 JMS-12-3 JMS-12-4 JMS-12-5 JMS-12-6 MS1-122-1 MS1-122-2
1:       11        6       11       14       16       15        23        29
2:        0        0        0        0        0        0         0         0
3:      373      410      415      261      439      609       622       674
4:       88       81      106       79       92      108        80       123
5:       11        4        5        7        9       16        13        17
6:       23      102      110      106       85      275       181       181
   MS1-122-3 MS1-122-4 MS1-122-5 MS1-122-6 MS1-36-1 MS1-36-2 MS1-36-3 MS1-36-4
1:        50        26        54        23       13       11       15       13
2:         0         0         0         0        0        0        0        0
3:       618       233       986       368      787      342     1020      531
4:        91        52       176        70      153       91      228      148
5:        20        13        32        11        3       15       20        3
6:       253        69       259       162      331       76      386      180
   MS1-36-5 MS1-36-6 MS1-41-1 MS1-41-2 MS1-41-3 MS1-41-4 MS1-41-5 MS1-41-6
1:       12       33        9       14        9        7       17        7
2:        0        0        0        0        0        0        0        0
3:      491      714      535      231      454      355      618      331
4:      129      173      101       54       82       73      107       75
5:        1       17       10        2        7        9        2        6
6:       14      126      115       88      181      122      235      162
   MS1-56-1 MS1-56-2 MS1-56-3 MS1-56-4 MS1-56-5 MS1-56-6 STJM-4-1 STJM-4-2
1:       35       21       25       10        0       30        0        6
2:        0        0        0        0        0        0        0        0
3:     1122     1017      846      828      155     1116       18       50
4:      196      242      156      152       46      247       17       33
5:        1        1        1        2        1        0        0        0
6:       43       20       77       26       11       15        0        0
   STJM-4-3 STJM-4-4 STJM-4-5 STJM-4-6
1:        3        9        3        4
2:        0        0        0        0
3:       52      184       83       69
4:       19       58       18       23
5:        1        6        0        0
6:        5        7        2        3</code></pre>
</div>
</div>
</section>
<section id="read-in-metadata" class="level1">
<h1>Read in metadata</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"SCRI_Phenotype_Analysis/2023_metadata.csv"</span>, <span class="at">stringsAsFactors =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CAL_Wip_ID =</span> <span class="fu">gsub</span>(<span class="st">' '</span>, <span class="st">'-'</span>, CAL_Wip_ID) <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>           <span class="fu">gsub</span>(<span class="st">'M4'</span>, <span class="st">'M-4'</span>, .)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">matches</span>(<span class="st">'&gt;'</span>))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   CAL_Wip_ID genotype Female_Parent length_2Y RLN_2Y length_3Y RLN_3Y
1:    29JM-11   29JM11         29.11     96.30  68.90    152.80   17.9
2:     JMS-12   JMS 12         29.11     63.40  26.70        NA     NA
3:     MS1-36   MS1 36         31.01     24.33   1.29     37.50   34.1
4:     MS1-41   MS1 41         31.01    117.50 247.80    139.56   63.9
5:     MS1-56   MS1 56         31.01     44.33  16.40     44.94   13.0
6:    MS1-122  MS1 122         31.01    149.33   3.70    173.00   68.9
   Over_length_3Y Over_RLN_3Y   CG_Avg  PHY_Avg
1:             NA          NA 2.800000 32.70450
2:             NA          NA 3.400000 26.29900
3:             NA          NA 1.500000 15.97925
4:             NA          NA 2.133333 35.02627
5:             NA          NA 1.075000 19.50000
6:         100.56        26.7 3.300000 39.70000</code></pre>
</div>
</div>
<section id="modify-metadata-for-this-analysis" class="level2">
<h2 class="anchored" data-anchor-id="modify-metadata-for-this-analysis">Modify metadata for this analysis</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Sample =</span> <span class="fu">colnames</span>(dat)[<span class="fu">colnames</span>(dat) <span class="sc">!=</span> <span class="st">'GeneID'</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Hybrid =</span> <span class="fu">as.factor</span>(<span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">-</span><span class="sc">\\</span><span class="st">d$"</span>, <span class="st">""</span>, Sample))) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'Hybrid'</span> <span class="ot">=</span> <span class="st">'CAL_Wip_ID'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">gsub</span>(<span class="st">'2Y-length'</span>, <span class="st">'Length_2Y'</span>, .x) <span class="sc">%&gt;%</span> </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>                <span class="fu">gsub</span>(<span class="st">'3Y-length'</span>, <span class="st">'Length_3Y'</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>                <span class="fu">gsub</span>(<span class="st">'2Y-RLN'</span>, <span class="st">'RLN_2Y'</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>                <span class="fu">gsub</span>(<span class="st">'3Y-RLN'</span>, <span class="st">'RLN_3Y'</span>, .)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">RLN_2_3Y =</span> <span class="fu">rowMeans</span>(<span class="fu">select</span>(., <span class="fu">matches</span>(<span class="st">'RLN'</span>))))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(RLN_2Y = log10(RLN_2Y + 1))</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(metadata) <span class="ot">=</span> metadata<span class="sc">$</span>Sample</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(metadata, <span class="st">'Plotting/metadata.csv'</span>)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             Sample  Hybrid genotype Female_Parent length_2Y RLN_2Y length_3Y
29JM-11-1 29JM-11-1 29JM-11   29JM11         29.11      96.3   68.9     152.8
29JM-11-2 29JM-11-2 29JM-11   29JM11         29.11      96.3   68.9     152.8
29JM-11-3 29JM-11-3 29JM-11   29JM11         29.11      96.3   68.9     152.8
29JM-11-4 29JM-11-4 29JM-11   29JM11         29.11      96.3   68.9     152.8
29JM-11-5 29JM-11-5 29JM-11   29JM11         29.11      96.3   68.9     152.8
29JM-11-6 29JM-11-6 29JM-11   29JM11         29.11      96.3   68.9     152.8
          RLN_3Y Over_length_3Y Over_RLN_3Y CG_Avg PHY_Avg RLN_2_3Y
29JM-11-1   17.9             NA          NA    2.8 32.7045       NA
29JM-11-2   17.9             NA          NA    2.8 32.7045       NA
29JM-11-3   17.9             NA          NA    2.8 32.7045       NA
29JM-11-4   17.9             NA          NA    2.8 32.7045       NA
29JM-11-5   17.9             NA          NA    2.8 32.7045       NA
29JM-11-6   17.9             NA          NA    2.8 32.7045       NA</code></pre>
</div>
</div>
</section>
<section id="plotting-metadata" class="level2">
<h2 class="anchored" data-anchor-id="plotting-metadata">Plotting metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>metaLong <span class="ot">=</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>Sample) <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hybrid, </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>         CG_Avg,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>         PHY_Avg,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>         RLN_2Y,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>         RLN_3Y,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>         length_3Y) <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">where</span>(is.numeric), </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">'Trait'</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(metaLong,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>          <span class="fu">aes</span>(<span class="fu">reorder_within</span>(Hybrid, value, Trait), </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>              value, </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">fill =</span> Hybrid))<span class="sc">+</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()<span class="sc">+</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_reordered</span>()<span class="sc">+</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">30</span>))<span class="sc">+</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Trait, <span class="at">scales =</span> <span class="st">'free'</span>, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 4 rows containing missing values (`position_stack()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_plot</span>(<span class="st">'DGEresults/metadata_plot.png'</span>, p, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">width =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 4 rows containing missing values (`position_stack()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div>
</section>
<section id="center-and-scale-predictors-to-make-comparisons-more-interpretable-in-results" class="level2">
<h2 class="anchored" data-anchor-id="center-and-scale-predictors-to-make-comparisons-more-interpretable-in-results">Center and scale predictors to make comparisons more interpretable in results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>metadata_S <span class="ot">=</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.numeric, scale) </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metadata_S)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             Sample  Hybrid genotype Female_Parent length_2Y    RLN_2Y
29JM-11-1 29JM-11-1 29JM-11   29JM11     -1.140871 0.3700379 0.1440493
29JM-11-2 29JM-11-2 29JM-11   29JM11     -1.140871 0.3700379 0.1440493
29JM-11-3 29JM-11-3 29JM-11   29JM11     -1.140871 0.3700379 0.1440493
29JM-11-4 29JM-11-4 29JM-11   29JM11     -1.140871 0.3700379 0.1440493
29JM-11-5 29JM-11-5 29JM-11   29JM11     -1.140871 0.3700379 0.1440493
29JM-11-6 29JM-11-6 29JM-11   29JM11     -1.140871 0.3700379 0.1440493
          length_3Y     RLN_3Y Over_length_3Y Over_RLN_3Y    CG_Avg   PHY_Avg
29JM-11-1 0.7477421 -0.9237004             NA          NA 0.4502366 0.5030946
29JM-11-2 0.7477421 -0.9237004             NA          NA 0.4502366 0.5030946
29JM-11-3 0.7477421 -0.9237004             NA          NA 0.4502366 0.5030946
29JM-11-4 0.7477421 -0.9237004             NA          NA 0.4502366 0.5030946
29JM-11-5 0.7477421 -0.9237004             NA          NA 0.4502366 0.5030946
29JM-11-6 0.7477421 -0.9237004             NA          NA 0.4502366 0.5030946
          RLN_2_3Y
29JM-11-1       NA
29JM-11-2       NA
29JM-11-3       NA
29JM-11-4       NA
29JM-11-5       NA
29JM-11-6       NA</code></pre>
</div>
</div>
</section>
<section id="plotting-metadata-1" class="level2">
<h2 class="anchored" data-anchor-id="plotting-metadata-1">Plotting metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>metaLongS <span class="ot">=</span> metadata_S <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>Sample) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hybrid, CG_Avg, PHY_Avg, RLN_2Y, RLN_3Y) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">where</span>(is.numeric), <span class="at">names_to =</span> <span class="st">'Trait'</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span><span class="fu">ggplot</span>(metaLongS, <span class="fu">aes</span>(<span class="fu">reorder_within</span>(Hybrid, value, Trait), value, <span class="at">fill =</span> Hybrid))<span class="sc">+</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()<span class="sc">+</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_reordered</span>()<span class="sc">+</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">30</span>))<span class="sc">+</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Trait, <span class="at">scales =</span> <span class="st">'free'</span>, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (`position_stack()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="get-annotation" class="level1">
<h1>Get annotation</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>Jm_anno <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jm_x_Jr/Jm_x_Jr_Genomic_annotation.csv'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parent_haplotype =</span> <span class="st">'J.microcarpa'</span>) </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>Jr_anno <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'C:/Users/hsaxe/OneDrive/Documents/ALAB/Genome_info/Genomic_Annotation_2/Jr/Jr_Genomic_annotation.csv'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parent_haplotype =</span> <span class="st">'J.regia'</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Rbind annotations into one file</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>annotation_combined <span class="ot">=</span> Jm_anno <span class="sc">%&gt;%</span> </span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(Jr_anno, <span class="at">fill =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">as.character</span>(GeneID)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>product_accession) </span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove transcript variants since we did not map to mRNAs</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>annotation_combined <span class="ot">=</span> annotation_combined <span class="sc">%&gt;%</span> </span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>feature <span class="sc">%like%</span> <span class="st">'CDS'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">', transcript variant X([1-9]|1[0-9])'</span>, <span class="st">''</span>, name) <span class="sc">%&gt;%</span> </span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>           <span class="fu">gsub</span>(<span class="st">' isoform X([1-9]|1[0-9])'</span>, <span class="st">''</span>, .)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(annotation_combined <span class="sc">%&gt;%</span> </span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>         <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>         <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">paste0</span>(<span class="st">'LOC'</span>, GeneID)), <span class="st">'For_Panther/annotation_combined_GeneIDs.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="limma-needs-geneids-as-rownames-in-expression-data.-also-colnames-of-expression-data-need-to-match-rownames-of-metadata" class="level2">
<h2 class="anchored" data-anchor-id="limma-needs-geneids-as-rownames-in-expression-data.-also-colnames-of-expression-data-need-to-match-rownames-of-metadata">Limma needs GeneIDs as rownames in expression data. Also, colnames of expression data need to match rownames of metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>dat1 <span class="ot">=</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span>  <span class="st">"GeneID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Do colnames in data match rownames in metadata? If they don't, use match(x,y) produces the order of y required to match the order of x</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(dat1) <span class="sc">==</span> <span class="fu">rownames</span>(metadata_S))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Names match</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co"># If they didn't match, use below code</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="do">## What order do rows of metadat need to be in to match colnames of dat1?</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># match(colnames(dat1), rownames(metadata))</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Reset rownames</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co"># metadata = metadata[match(colnames(dat1), rownames(metadata)),]</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co"># all(colnames(dat1) == rownames(metadata))</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co"># now they match</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="create-dgelist-object" class="level1">
<h1>Create DGEList object</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DGEList</span>(dat1)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dds<span class="sc">$</span>counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 67924    42</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Calculate library normalization factors (does not do anything to data)</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">calcNormFactors</span>(dds)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="do">## These are the size factors (normalization factors) for each sample</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co"># dds$samples</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="filter-by-max-cpm-this-is-used-for-dge-analysis" class="level2">
<h2 class="anchored" data-anchor-id="filter-by-max-cpm-this-is-used-for-dge-analysis">Filter by max CPM (This is used for DGE analysis)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>d <span class="ot">=</span> <span class="fu">expression_filter</span>(dds, <span class="at">DGEList =</span> T, <span class="at">FilterFUN =</span> max, <span class="at">FilterThreshold =</span> <span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will negatively impact filtering</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Data has no missing values"
[1] "Removed 48725 features based on max threshold of 30, 19199 remaining"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># d = expression_filter(dds, DGEList = T, FilterFUN = IQR, RankThreshold = 20000)</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(d<span class="sc">$</span>counts <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>         <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">paste0</span>(<span class="st">'LOC'</span>, GeneID)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">select</span>(GeneID),</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>       <span class="st">'For_Panther/expressed_genes.csv'</span>)</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="do">## CPM normalized counts of all data</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>cpm <span class="ot">=</span> <span class="fu">cpm</span>(dds, <span class="at">prior.count =</span> <span class="dv">2</span>, <span class="at">log =</span> F) </span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>cpm <span class="ot">=</span> cpm <span class="sc">%&gt;%</span> </span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">gsub</span>(<span class="st">'X'</span>, <span class="st">''</span>, <span class="fu">gsub</span>(<span class="st">'</span><span class="sc">\\</span><span class="st">.'</span>, <span class="st">'-'</span>, .x)))</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(cpm, <span class="st">'Plotting/cpm_plotting_data.csv'</span>)</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a><span class="do">## CPM normalized counts of filtered data</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>cpmd <span class="ot">=</span> <span class="fu">cpm</span>(d)</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(cpmd <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>), <span class="st">'DGEresults/DGE_CPM_data.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-pca-to-check-samples" class="level2">
<h2 class="anchored" data-anchor-id="plot-pca-to-check-samples">Plot PCA to check samples</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd, metadata,</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>               <span class="co"># Using 'Group' here becuase 'Hybrid' is already in metadata. Will cause error if I used 'Hybrid'</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'2D'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>pca_scatter <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd, metadata,</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>               <span class="co"># Using 'Group' here becuase 'Hybrid' is already in metadata. Will cause error if I used 'Hybrid'</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'scatter'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>plot_dat <span class="ot">=</span> pca_scatter<span class="sc">$</span>plot_dat <span class="sc">%&gt;%</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Hybrid =</span> Group) <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hybrid, </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>         CG_Avg,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>         PC1,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>         PC2,</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>         PC3, </span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>         PC4, </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>         PC5) <span class="sc">%&gt;%</span> </span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(Hybrid, CG_Avg), <span class="at">names_to =</span> <span class="st">'PC'</span>)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_dat, <span class="fu">aes</span>(value, CG_Avg))<span class="sc">+</span></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'lm'</span>)<span class="sc">+</span></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>PC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>plot_dat <span class="ot">=</span> pca_scatter<span class="sc">$</span>plot_dat <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Hybrid =</span> Group) <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hybrid, </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>         PHY_Avg,</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>         PC1,</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>         PC2,</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>         PC3, </span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>         PC4, </span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>         PC5) <span class="sc">%&gt;%</span> </span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(Hybrid, PHY_Avg), <span class="at">names_to =</span> <span class="st">'PC'</span>)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_dat, <span class="fu">aes</span>(value, PHY_Avg))<span class="sc">+</span></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'lm'</span>)<span class="sc">+</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>PC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
</section>
<section id="do-any-pcs-correlate-with-phenotypes" class="level2">
<h2 class="anchored" data-anchor-id="do-any-pcs-correlate-with-phenotypes">Do any PCs correlate with phenotypes?</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd, metadata,</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>             <span class="co"># Having to use 'Hybrid' here to get it into the plotting data because the function drops everything but the join_by_name when summarizing for scatter</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">join_by_name =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_name =</span> Hybrid,</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="st">'PC2'</span>, <span class="at">y =</span> <span class="st">'CG_Avg'</span>,</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">scale =</span> T,</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">center =</span> T,</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot_type =</span> <span class="st">'scatter'</span>,</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">summarise_for_scatter =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>b <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd, metadata,</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>             <span class="co"># Having to use 'Hybrid' here to get it into the plotting data because the function drops everything but the join_by_name when summarizing for scatter</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">join_by_name =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_name =</span> Hybrid,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="st">'PC2'</span>,</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> <span class="st">'PHY_Avg'</span>,</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">scale =</span> T,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">center =</span> T,</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot_type =</span> <span class="st">'scatter'</span>,</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">summarise_for_scatter =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>arr <span class="ot">=</span> ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(a<span class="sc">$</span>plot,</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>                b<span class="sc">$</span>plot,</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'A)'</span>, <span class="st">'B)'</span>), <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>arr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>sjPlot<span class="sc">::</span><span class="fu">save_plot</span>(<span class="st">'DGEresults/PCA_scatter_CG_PH_NEM.png'</span>, arr, <span class="at">height =</span> <span class="dv">24</span>, <span class="at">width =</span> <span class="dv">14</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div>
</section>
</section>
<section id="section" class="level1">
<h1>*</h1>
</section>
<section id="differential-expression-analysis" class="level1">
<h1>*<em>*Differential expression analysis</em></h1>
</section>
<section id="cg-fit-model-for-genes-associated-with-cg-score" class="level1">
<h1>CG Fit model for genes associated with CG score</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mm_CG = model.matrix(~poly(CG_Avg, 2), data = metadata_S)</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>mm_CG <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>CG_Avg, <span class="at">data =</span> metadata_S)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mm_CG)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          (Intercept)    CG_Avg
29JM-11-1           1 0.4502366
29JM-11-2           1 0.4502366
29JM-11-3           1 0.4502366
29JM-11-4           1 0.4502366
29JM-11-5           1 0.4502366
29JM-11-6           1 0.4502366</code></pre>
</div>
</div>
<section id="what-is-voom" class="level2">
<h2 class="anchored" data-anchor-id="what-is-voom">What is voom?</h2>
<section id="counts-are-transformed-to-log2-counts-per-million-reads-cpm-where-per-million-reads-is-defined-based-on-the-normalization-factors-we-calculated-earlier." class="level3">
<h3 class="anchored" data-anchor-id="counts-are-transformed-to-log2-counts-per-million-reads-cpm-where-per-million-reads-is-defined-based-on-the-normalization-factors-we-calculated-earlier.">1. Counts are transformed to log2 counts per million reads (CPM), where “per million reads” is defined based on the normalization factors we calculated earlier.</h3>
</section>
<section id="a-linear-model-is-fitted-to-the-log2-cpm-for-each-gene-and-the-residuals-are-calculated." class="level3">
<h3 class="anchored" data-anchor-id="a-linear-model-is-fitted-to-the-log2-cpm-for-each-gene-and-the-residuals-are-calculated.">2. A linear model is fitted to the log2 CPM for each gene, and the residuals are calculated.</h3>
</section>
<section id="a-smoothed-curve-is-fitted-to-the-sqrtresidual-standard-deviation-by-average-expression." class="level3">
<h3 class="anchored" data-anchor-id="a-smoothed-curve-is-fitted-to-the-sqrtresidual-standard-deviation-by-average-expression.">3. A smoothed curve is fitted to the sqrt(residual standard deviation) by average expression.</h3>
</section>
<section id="see-red-line-in-plot-below" class="level3">
<h3 class="anchored" data-anchor-id="see-red-line-in-plot-below">(see red line in plot below)</h3>
</section>
<section id="the-smoothed-curve-is-used-to-obtain-weights-for-each-gene-and-sample-that-are-passed-into-limma-along-with-the-log2-cpms." class="level3">
<h3 class="anchored" data-anchor-id="the-smoothed-curve-is-used-to-obtain-weights-for-each-gene-and-sample-that-are-passed-into-limma-along-with-the-log2-cpms.">4. The smoothed curve is used to obtain weights for each gene and sample that are passed into limma along with the log2 CPMs.</h3>
</section>
<section id="more-details-at-voom-precision-weights-unlock-linear-model-analysis-tools-for-rna-seq-read-counts" class="level3">
<h3 class="anchored" data-anchor-id="more-details-at-voom-precision-weights-unlock-linear-model-analysis-tools-for-rna-seq-read-counts">More details at “<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29">voom: precision weights unlock linear model analysis tools for RNA-seq read counts</a>”</h3>
</section>
</section>
<section id="filtered-mean-variance-trend-cg" class="level2">
<h2 class="anchored" data-anchor-id="filtered-mean-variance-trend-cg">Filtered mean-variance trend CG</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>y_CG <span class="ot">&lt;-</span> <span class="fu">voom</span>(d, mm_CG , <span class="at">plot =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="vs-unfiltered-mean-variance-trend-cg" class="level2">
<h2 class="anchored" data-anchor-id="vs-unfiltered-mean-variance-trend-cg">Vs Unfiltered mean-variance trend CG</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">voom</span>(dds, mm_CG, <span class="at">plot =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="fitting-linear-models-in-limma-with-random-effects-cg" class="level2">
<h2 class="anchored" data-anchor-id="fitting-linear-models-in-limma-with-random-effects-cg">Fitting linear models in limma with random effects CG</h2>
<section id="we-used-six-biological-replicates-but-these-replicates-tell-us-little-about-the-variation-due-to-genotype.-therefore-they-are-more-like-technical-replicate-used-to-account-for-variation-we-see-in-expression-likely-due-to-environmental-pots-soil-type-water-amount-etc.-and-sampling-technical-variation-in-sampling-method-and-rna-extraction-effects.-these-replicates-should-not-be-used-in-calculating-a-p-value-testing-the-hypothesis-of-relationship-between-genotype-and-mrna-expression." class="level3">
<h3 class="anchored" data-anchor-id="we-used-six-biological-replicates-but-these-replicates-tell-us-little-about-the-variation-due-to-genotype.-therefore-they-are-more-like-technical-replicate-used-to-account-for-variation-we-see-in-expression-likely-due-to-environmental-pots-soil-type-water-amount-etc.-and-sampling-technical-variation-in-sampling-method-and-rna-extraction-effects.-these-replicates-should-not-be-used-in-calculating-a-p-value-testing-the-hypothesis-of-relationship-between-genotype-and-mrna-expression.">We used six ‘biological replicates’ but these replicates tell us little about the variation due to genotype. Therefore, they are more like technical replicate used to account for variation we see in expression likely due to environmental (pots, soil type, water amount etc.) and sampling (technical variation in sampling method and RNA extraction) effects. These replicates should not be used in calculating a p-value testing the hypothesis of relationship between genotype and mRNA expression.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Need to tell limma where the within class correlation is coming from</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>dupcor_CG <span class="ot">=</span> <span class="fu">duplicateCorrelation</span>(y_CG,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>                                 mm_CG,</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">block =</span> metadata_S<span class="sc">$</span>Hybrid)</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="do">## How correlated are the hybrid replicates on average?</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>consensus.corr.CG <span class="ot">=</span> dupcor_CG<span class="sc">$</span>consensus.correlation</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>consensus.corr.CG</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.5354868</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lmFit fits a linear model using weighted least squares for each gene:</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>fit_CG <span class="ot">=</span> <span class="fu">lmFit</span>(y_CG, </span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">design =</span> mm_CG,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">block =</span> metadata_S<span class="sc">$</span>Hybrid, </span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">correlation =</span> consensus.corr.CG) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="the-variance-characteristics-of-low-expressed-genes-are-different-from-high-expressed-genes-if-treated-the-same-the-effect-is-to-over-represent-low-expressed-genes-in-the-de-list." class="level2">
<h2 class="anchored" data-anchor-id="the-variance-characteristics-of-low-expressed-genes-are-different-from-high-expressed-genes-if-treated-the-same-the-effect-is-to-over-represent-low-expressed-genes-in-the-de-list.">The variance characteristics of low expressed genes are different from high expressed genes, if treated the same, the effect is to over represent low expressed genes in the DE list.</h2>
<section id="this-is-corrected-for-by-the-log-transformation-and-voom.-however-some-genes-will-have-increased-or-decreased-variance-that-is-not-a-result-of-low-expression-but-due-to-other-random-factors.-we-are-going-to-run-empirical-bayes-to-adjust-the-variance-of-these-genes." class="level3">
<h3 class="anchored" data-anchor-id="this-is-corrected-for-by-the-log-transformation-and-voom.-however-some-genes-will-have-increased-or-decreased-variance-that-is-not-a-result-of-low-expression-but-due-to-other-random-factors.-we-are-going-to-run-empirical-bayes-to-adjust-the-variance-of-these-genes.">This is corrected for by the log transformation and voom. However, some genes will have increased or decreased variance that is not a result of low expression, but due to other random factors. We are going to run empirical Bayes to adjust the variance of these genes.</h3>
</section>
<section id="empirical-bayes-smoothing-of-standard-errors-shifts-standard-errors-that-are-much-larger-or-smaller-than-those-from-other-genes-towards-the-average-standard-error-see-linear-models-and-empirical-bayes-methods-for-assessing-differential-expression-in-microarray-experiments" class="level3">
<h3 class="anchored" data-anchor-id="empirical-bayes-smoothing-of-standard-errors-shifts-standard-errors-that-are-much-larger-or-smaller-than-those-from-other-genes-towards-the-average-standard-error-see-linear-models-and-empirical-bayes-methods-for-assessing-differential-expression-in-microarray-experiments">Empirical Bayes smoothing of standard errors (shifts standard errors that are much larger or smaller than those from other genes towards the average standard error) (see “<a href="https://www.degruyter.com/doi/10.2202/1544-6115.1027">Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments</a>”</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>BlockFit_CG <span class="ot">=</span> <span class="fu">eBayes</span>(fit_CG)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="limma-results-cg" class="level1">
<h1>Limma results CG</h1>
<section id="results-summary" class="level2">
<h2 class="anchored" data-anchor-id="results-summary">Results summary</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>res_summaries_CG <span class="ot">=</span> BlockFit_CG <span class="sc">%&gt;%</span> <span class="fu">decideTests</span>() <span class="sc">%&gt;%</span> <span class="fu">summary</span>()</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>res_summaries_CG</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       (Intercept) CG_Avg
Down            30   3888
NotSig         469  13290
Up           18700   2021</code></pre>
</div>
</div>
</section>
<section id="table-of-results-cg" class="level2">
<h2 class="anchored" data-anchor-id="table-of-results-cg">Table of results CG</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>impCG <span class="ot">=</span> limma<span class="sc">::</span><span class="fu">topTable</span>(BlockFit_CG, </span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">'CG_Avg'</span>,</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sort.by =</span> <span class="st">"logFC"</span>,</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">p.value =</span> <span class="fl">0.05</span>, </span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">adjust.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">number =</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">R =</span> <span class="fu">sqrt</span>(t<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>(t<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">40</span>)), <span class="at">AveExpr =</span> <span class="dv">2</span><span class="sc">^</span>AveExpr)</span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(impCG, <span class="st">'DGEresults/unAnnotated_results_CG.csv'</span>)</span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>rawCG <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'DGEresults/unAnnotated_results_CG.csv'</span>)</span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-14"><a href="#cb57-14" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(impCG)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5909    8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="do">## adding Hybrid as a blocking variable reduced DEGs by several thousand</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="merge-annotation-with-results-cg" class="level2">
<h2 class="anchored" data-anchor-id="merge-annotation-with-results-cg">Merge annotation with results CG</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>impCG <span class="ot">=</span> impCG <span class="sc">%&gt;%</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(annotation_combined, <span class="at">by =</span> <span class="st">"GeneID"</span>)</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(impCG)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID     logFC   AveExpr         t      P.Value    adj.P.Val         B
1 109017709 -5.436123  1.749451 -4.790291 1.896594e-05 7.476944e-04  2.452608
2 118345731 -4.164794 30.981442 -7.175920 6.084723e-09 1.257174e-05 10.352673
3 109020534 -4.000459  7.660034 -4.193793 1.294339e-04 2.246837e-03  1.038583
4 109001297 -3.667421 12.793231 -5.105935 6.686095e-06 4.447445e-04  3.740688
5 108988244  3.617572  2.359144  5.219340 4.581922e-06 3.759330e-04  3.378459
6 108988244  3.617572  2.359144  5.219340 4.581922e-06 3.759330e-04  3.378459
          R feature  class chromosome   start     end
1 0.6037745    mRNA                15 5712898 5713715
2 0.7502080    mRNA                     31540   33123
3 0.5526388   ncRNA lncRNA          1 2469864 2470874
4 0.6281614    mRNA                 1  797531  798444
5 0.6364977   ncRNA lncRNA          1 7310979 7313148
6 0.6364977   ncRNA lncRNA          1 7310979 7313147
                                      name Jr-GeneID Jr-chromosome
1             uncharacterized LOC109017709        NA          &lt;NA&gt;
2                     endochitinase 2-like        NA          &lt;NA&gt;
3             uncharacterized LOC109020534        NA          &lt;NA&gt;
4 light-regulated protein 1, chloroplastic        NA          &lt;NA&gt;
5             uncharacterized LOC108988244        NA          &lt;NA&gt;
6             uncharacterized LOC108988244        NA          &lt;NA&gt;
          Subcell_loc Parent_haplotype
1             nucleus          J.regia
2 extracellular space          J.regia
3                              J.regia
4       mitochondrion          J.regia
5                              J.regia
6                              J.regia</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(impCG<span class="sc">$</span>GeneID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5909</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(impCG, <span class="st">'DGEresults/Limma_results_table_CG.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pca-of-cg-degs" class="level2">
<h2 class="anchored" data-anchor-id="pca-of-cg-degs">PCA of CG DEGs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">=</span> impCG <span class="sc">%&gt;%</span> </span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">as.character</span>(GeneID))</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>PCA_CG <span class="ot">=</span> cpmd <span class="sc">%&gt;%</span> </span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(ids) <span class="sc">%&gt;%</span> </span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(GeneID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>pca_plot_CG <span class="ot">=</span> <span class="fu">plot_pca</span>(PCA_CG, metadata,</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'CG_Avg'</span>,</span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'CG_Avg'</span>,</span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'2D'</span>,</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">sep =</span> <span class="st">'-'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># View(pca_plot_CG$plot_dat)</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>pca_plot_CG</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$plot_dat
      Sample        PC1        PC2        PC3         PC4          PC5
1  29JM-11-1 -47.952689 -12.994505   2.515359 -19.2441809  16.99294347
2  29JM-11-2 -42.554941 -23.102363   4.896068 -21.4896859  15.06195316
3  29JM-11-3 -31.474196 -20.110970  11.571161 -15.2562012  15.31584729
4  29JM-11-4 -35.660492 -17.199818   5.922734 -11.4924113  12.29285535
5  29JM-11-5 -46.093425 -20.833699   2.262672 -17.4205062  17.10523048
6  29JM-11-6 -46.091409   3.635911  -2.441807  -3.6387419  10.66086657
7   JMS-12-1 -66.840628  24.366074 -27.412874  31.2814094  14.29402516
8   JMS-12-2 -15.575958  -1.167821   6.886078   3.0264107   4.52527280
9   JMS-12-3 -40.158280   8.619779  -1.565090  14.1518457  12.55956968
10  JMS-12-4 -57.894585   9.779490 -13.521174  20.9238840  13.77737677
11  JMS-12-5 -36.988684   1.365370  -7.238464  11.4247196   7.98646036
12  JMS-12-6 -47.347009  -6.139928  -2.653706   6.4029723  12.33451685
13 MS1-122-1 -49.375284 -19.986212 -20.795643  -4.1961333 -21.70286209
14 MS1-122-2 -37.934269 -13.938136 -23.863905  -2.0677229 -24.88320648
15 MS1-122-3 -74.634109 -13.703149 -35.926506   2.7590831 -19.46266315
16 MS1-122-4 -67.000067  -3.628495 -25.104737   1.0325682 -18.45366217
17 MS1-122-5 -48.279782 -15.749296 -26.333591  -0.6926176 -15.50105481
18 MS1-122-6 -50.704861 -18.592823 -17.185477  -8.0672730 -17.42968398
19  MS1-36-1  59.871240 -26.869775  40.770569  30.9946764  -9.62763638
20  MS1-36-2   7.909877  -4.812610  27.048814  15.5309382 -12.87347602
21  MS1-36-3  -2.156044 -20.666562  34.423556  12.9333868  -4.13808535
22  MS1-36-4  35.985472 -15.677974  33.450310  17.7110970 -10.40869218
23  MS1-36-5  88.922080  -6.966268   3.230852  12.7068466  -6.24383320
24  MS1-36-6  15.663225  -9.857184  13.695415  10.1740317  -9.34677746
25  MS1-41-1   1.507105 -13.946464  15.903916  -5.7757404  -3.58942618
26  MS1-41-2  -3.274775 -16.028616  24.754381  -8.8449287   0.13366081
27  MS1-41-3 -12.974982 -16.723878  19.012151  -6.8681422   0.16700091
28  MS1-41-4  -8.086957 -16.842859  15.894868 -11.7138430  -4.50108053
29  MS1-41-5  -7.890610 -21.161418  18.268240  -9.8143948   0.72526520
30  MS1-41-6 -18.376734 -17.548398  21.769121  -9.8844800   1.38067194
31  MS1-56-1 105.026278 -21.319268 -24.341773  -5.8527950   5.54073538
32  MS1-56-2 126.301101 -15.474329 -35.165023  -2.9184203  10.37924976
33  MS1-56-3 112.596078 -21.159096 -13.902742   8.8349076   7.88341080
34  MS1-56-4 124.739574 -21.046242 -28.304774   0.5277552  11.41604458
35  MS1-56-5  73.096842   9.810112  -1.651278  -3.2208177  -8.37876862
36  MS1-56-6 104.534505   8.033279 -23.648985  -9.1606766  -0.09373784
37  STJM-4-1  26.189351  74.385960  15.496110 -32.5949725  -9.09451469
38  STJM-4-2  28.385179  94.064886   6.949376  -6.1560437 -10.47187400
39  STJM-4-3 -17.189789  67.233470  -0.488691  19.2280659   8.87646033
40  STJM-4-4  -3.788064  42.323082   3.525531   0.3210571   4.81451554
41  STJM-4-5  37.947422  58.453467   5.398696 -21.1670154  -7.63666384
42  STJM-4-6 -32.376706  51.177279  -2.099741  17.5720890   9.61376580
           PC6         PC7         PC8         PC9        PC10         PC11
1   -5.7650733   6.7350968 -11.9629731   9.9930894  -2.1251676   3.82647746
2  -10.9770527  -9.7687011  -6.4225956   8.5032293  17.2084427   2.55108265
3   -8.9679009  -4.7027011  -1.5620887   3.3713327   2.6256569  -0.38724590
4   -8.0633148  -4.1591137  -0.7986678   1.6678297   3.3034802   1.65875262
5   -9.4727616  -0.5694457  -7.3750766  11.6492671  -0.5460094  -0.15991734
6   -2.7719846  18.7227627 -14.0670595  -0.5276709 -11.5776093   1.02611947
7   -2.9037483  -7.2400703  13.4278534  -7.2303819  -2.8765179  -5.72444768
8   16.6244099  -5.4256707  13.0961986  24.2151739 -11.9534582  -1.46319055
9   -3.6314651  -4.5397463   9.3763820  -6.2258885  -1.4303068  -3.37810049
10  -6.3608697  -6.0654512  11.2048249  -5.9973870  -7.3469224  -5.60086854
11   1.1965721  -8.1559118   9.9680515   0.9808348  -0.1784275  -2.93251711
12  -5.0485614  -4.1742005   5.2732663  -3.0007021  -7.8112237  -7.11341529
13   5.5947384  -9.7812597   4.4074221   8.0369740   4.3717225   0.93687132
14  14.9238771  -4.4849429   3.2243975  15.6678884   0.1630310   0.73674017
15 -10.3704153   0.6160974 -10.2906375  -8.1637999  -5.1571671  -1.18844253
16  -4.1328695  13.3371279 -13.0355018  -5.4620748  -6.9931018   4.71628157
17  -3.8537508  -4.0563803  -3.4345156  -9.9960243   4.0353586   0.06820826
18  -0.3393818  -3.3631030  -1.2979345   3.4157049   2.4142848   2.77772544
19  -3.8251044  -9.4698066 -20.5271250   4.8203377 -16.8161822   1.20720435
20  -8.5391840   4.1905077   1.7053257  -2.1688043   9.9861555  -0.08624665
21  -8.9104360  -5.4222419  -3.3660032   0.2425781   1.6654337  -3.33526853
22  -7.5452340  -9.4001828  -0.2135553  -1.5871062   5.6582919  -3.49036563
23  -8.0550799  -5.4738442   0.1034343   9.9231337   4.5951073   3.12866523
24  -4.4078670  10.8927196   3.8203611  -4.0924448  17.6743873  13.96228734
25  15.7082613   2.2085325   9.7957930  -6.2033688   2.4261142  -1.43022887
26  12.2524073  13.3786441   2.2661419 -14.3419991  -6.8453354  -3.45178601
27   9.2833709   4.2222542   5.0183413 -11.9590940   1.6023350  -2.23597777
28  23.2105798   3.9748798  11.0658318  10.4652415  -7.3404319   0.65232861
29  11.2996357   2.8453762   3.2329208 -12.8755685   3.8798766  -1.78996745
30   4.6876663   8.0710752   1.2341524 -11.5924030  -2.5954306  -2.97514465
31   5.3720717 -10.3460075   2.2301563  -7.7603769  14.4206221  -3.31441322
32   3.7355297  -5.9858118  -1.4350827  -0.7267940   2.4659848   1.64287129
33   2.6660517  -2.8329589  -7.8302311   0.8283843 -10.6364549   6.98768728
34   7.9201431  -0.2280498  -7.7958965  -2.5997302  -6.6353155   4.25675998
35 -26.4915998  29.0197582  26.0260870  15.1523499  -1.4064224  -3.70062787
36  -5.2868598  17.4966843   4.4031499  -6.3551395  -2.2985112  -5.85630171
37 -11.4816885 -22.0375123  12.8961056 -12.2515259 -15.0043858  26.58402762
38  -2.4828831  -1.1824968  -9.9592704   7.3173949  -3.7667813 -12.45185832
39  11.5239321   3.6403064  -4.0907464   5.1484685  17.9525842  11.18639365
40   7.9593666   2.2240147  -7.8573876  -2.1335455   3.7597553   0.22928648
41   2.4751558  -8.5493579 -10.4871359  -1.2810261   5.7785556 -28.47045817
42  13.2513168  15.8391310  -9.9667126   3.1336433   5.3539822  12.40101951
          PC12        PC13         PC14         PC15         PC16        PC17
1   -2.2129625   4.2711172  -5.79654511   2.56854614   0.31215320   7.0522443
2    6.0060893 -11.1949377  12.09993249 -22.37102951  -5.03273945  -4.6222533
3    4.5716175   1.4100573  -2.27410948   6.44485216   1.85647966  -2.5747252
4    1.0990996   0.9918079   1.22308250   6.57926121  -0.42479106  -4.2199175
5   -0.0901624   7.5819060  -7.04040943   4.40379159   0.65536771   5.0946427
6  -15.6482497   3.6698622   0.43504638   5.05784361  -2.41522566  -3.6012337
7    0.3294233   0.4951503  -1.30936514  -5.12671006  -6.32299534   6.2783685
8   -8.2589778   4.6768342  -1.98939317  -2.65359767   0.52098696  -4.7525094
9   -1.2418591  -0.6609378   1.38910319   0.31988726   1.38609082  -3.6300072
10   0.9456517   1.4640763   1.74083885  -1.46342302   0.33636976   0.4417269
11  -8.9548373  -1.1201695   3.08200103   2.54963080   1.05328257  -7.8200733
12  -1.4815448   2.3561908   1.01729896   0.81592892  -0.14512377  -1.2507293
13   4.4505968  -5.6256069  -3.99792452   0.86066256   0.12406167  -0.1312542
14  -3.0612184  -4.3256349  -4.28936753  -2.13438352  -2.38027305  -2.5295653
15   7.5027167   3.0711931   3.69405250  -1.54326256   4.14723407   4.2796437
16  -1.5491793   2.4000931   0.98139464   0.74461663   3.20383822   2.0515021
17   0.9022365  -2.7331130   3.69330470   2.28887626  -4.90478316  -2.4520439
18   1.2343568  -0.3497704  -2.04630723   4.76003998  -0.02255995  -1.1640191
19   3.8445512  -8.8285260  -2.85823727  -2.70892508 -14.57826403  -0.2628792
20  -5.8723784   3.5579896  -5.32004462  -2.28571413  10.51956737  -5.8041258
21   3.4570906  -1.6781619 -17.26836842  -1.54596205   8.24441517   2.6415365
22  -1.6114310  -0.8667036  -2.07324094   0.57587274   8.49985266   0.8993399
23  -8.3253290  12.7375245  18.16962141  -0.53656029  -1.60196307  22.6176546
24 -16.1651186  13.9888193   3.51076393   1.45289942 -10.61254800 -12.2965266
25   1.0667126   1.4919363   3.09075681   1.63377368  -0.15310782   1.3572501
26   4.6242628  -0.5942515   4.71503656  -0.49906777  -1.04789325   0.9357402
27   4.3772716   0.6598416   4.42023648   0.16897822   0.34758248   2.9968480
28   2.9300421   5.7757942   0.99608712  -3.65422078  -0.05649143   4.4402657
29   4.5794887  -1.9810677   4.52491442  -0.02552001  -1.09468303   2.4974247
30   8.1423760  -1.1342220  -0.04151355  -0.63345699   1.93960951   1.9485900
31  -4.5958915   3.1994837  -7.41128321   2.25838043   4.76617397   0.8023279
32   2.8966233   1.3157091  -2.25413130   2.94715937   8.22484543   1.1342758
33   6.8292401  -6.2885196  10.38109331   6.03806958   3.61195334 -10.4860465
34   1.6717902  -1.2804699   0.34059009  -0.09995392  -0.72377116  -3.8461215
35   8.2428686 -14.9190884   4.46566423   7.42398134  -0.39032793  -0.1473594
36  -2.1476275   4.6697754 -17.78133282 -18.17665408  -9.81319181   1.5448478
37  -5.1637842  -3.6506712  -3.71239046  -0.43229455  -0.27208818   2.4375137
38  15.1719296  19.6547478   7.04307778  -7.01757275   7.70837145 -10.0842993
39  21.0419775   0.8783624  -7.76809043  11.28163335 -10.59630510   3.9800884
40  -3.3854635  -2.5446981   1.58935679   1.70023429  -4.56805923  -1.6868844
41 -15.7406950 -13.1862610   1.28923189   8.53680429  -5.72712661   2.8990996
42 -10.4113028 -17.3554613   1.33956855  -8.50341509  15.42607610   5.0316430
          PC18         PC19        PC20         PC21          PC22       PC23
1   -3.3673759  -3.78868501  3.49425084   0.53338459   0.623691480  6.7956039
2   -0.6000636   5.75760086  6.07292058   5.46176098   0.680648573 -0.6398073
3   -0.1319972  -2.11280476 -6.73890851 -11.16756141   3.705058657 -3.3524734
4    2.6251196  -1.62344055 -4.38742190  -4.70157949  -0.131698745 -1.7106351
5   -1.5971080  -5.35393108  0.09892793   0.16788704  -3.245940823  2.2422995
6   11.7074683   1.70541485 12.12994526   1.60753827  -1.260502210 -3.4581264
7   -7.3423551 -12.20541390 12.26879998   1.55580951  -1.253456153  1.9202739
8   -2.0567914  10.18018708 -5.50672006   4.55950478   4.340453955 -4.5438847
9   -0.3493547   0.84178103 -3.05000263  -2.26014973   3.892316840 -4.8112568
10  -2.5829567  -0.05029379 -2.48092697  -1.42733014   2.270499214  1.8828229
11   9.3611235   8.89812172 -2.61996272   1.70965419 -10.809489920  7.7891736
12   2.3738530   4.28576095 -5.79896036   0.52331761   1.712907535 -3.5841345
13  -0.1762092  -5.04162719 -0.33883330  -5.51027809  -1.110858579 -5.2758292
14   3.9471931  -5.19588327  7.67735522  -6.66435723  -0.336191063 -4.9388422
15  -5.2100884   9.00400709 -7.00046043   4.55590203   2.261141306  3.7677598
16  -3.8783071   5.96589504  1.14997360   5.37417206   7.022706172 -2.3654378
17   5.7671694  -3.51984134 -2.18879782  -4.47453905  -0.038387972 -2.1461366
18   2.7762932  -1.11855060 -2.31172543   3.26723769  -8.269223450  8.4193094
19   5.3961777  -4.58098057 -4.43647682   5.12437716   1.230344455  0.6892376
20  -4.6887487   7.77396606 12.60221955  -3.62988094   1.616932912  1.2059936
21  -6.0621347  -2.57244683 -1.61325693   8.05104707  -6.738049568 -4.6593760
22   5.5817568   4.35203337  4.03845686  -8.17003802   3.888740360  3.6726490
23   3.3727402   2.25708947 -0.84085062  -5.06318064  -2.395343692 -2.9982877
24 -12.2565631  -6.56664106 -6.49893747   3.04275091  -0.424575695 -0.7842712
25   3.3270509  -2.22725344  0.03243440   2.85665118  -4.307647571  4.5968169
26   2.4474182   3.27773732  5.77732258  -0.02248344  -4.483312475 -3.9672931
27   0.6151297  -1.75026072 -2.85559592   2.85854678  -1.791958315 -0.9212551
28  -7.4299942  -1.14948408  1.53856418   1.33830717   6.277618152  7.5545952
29   2.5914228  -1.54364053 -2.89911719   2.07747480  -1.302674805 -0.6212875
30  -4.2357009  -0.99109380  0.67541544  -5.71305239   2.645860022 -2.8106283
31  13.0707608  -4.69946211  0.56787983   9.22102352  10.100981430  4.1333761
32  -3.5375224   0.52266717  1.20128968   8.25902267  -1.864541637 -8.9028478
33  -6.7087286  -2.47799174  3.39682028  -5.43529660   4.649105747  9.8658184
34  -7.8941389   3.86593089  1.94244374  -4.41706602  -9.796741384 -4.6955064
35   2.3796273  -2.84010697  0.03917470   5.21571339   0.636578530 -0.7210575
36   3.6460163   4.49871862 -6.05726775  -9.00354395  -0.067429369  4.0981376
37  -0.1277518   1.26768225  2.28101687   0.91594842   0.005243495 -0.1286151
38   4.3233904  -8.22024895 -0.86217203   0.88209611  -3.334860834 -0.4296775
39   0.9663346  12.06638904  2.35754695  -0.63078174   0.012657931  0.7670095
40   4.6104928  -3.97380133 -0.46917307   2.43736744   7.950592249 -4.4048736
41 -11.8486581   3.40809933 -0.52853763  -0.28982967  -0.305526528  1.9696429
42   1.1960103  -6.32519851 -9.85865292  -3.01554681  -2.255668226  1.5010212
          PC24         PC25        PC26         PC27        PC28        PC29
1  -4.49373228 -12.86868095 -3.60650520   1.18583030   0.4634050  1.94291953
2   1.19945621   2.42209979 -0.92688466  -1.05451375   0.6091443  1.23497680
3  -2.24761318   3.08283435  6.55668633  -2.24512098   1.0882157 -1.74989324
4   0.35446548   3.05559548  7.06218758  -0.76910478  -1.0779811  0.53017991
5  -1.12748195   0.67932760 -2.10538208   5.03766906  -1.4707078 -0.61349847
6   1.88690502   8.80829866 -1.20184262  -0.36420267  -1.2936832  1.60779869
7  -0.90900911  -1.06134801  3.38686170  -0.65159182   1.5997428  1.23860873
8  -0.31978496  -4.63087384 -3.14717128  -1.79438429   3.8782570  1.26538262
9   0.84645164  -1.76518827 -2.50797174  -4.20869830   1.4558427  0.05217784
10 -0.30141808   1.55575839 -1.45049278  -1.37767197  -0.5419390  0.18012622
11 -1.02347892   0.39261007  4.57374341   5.44264133  -4.1387838 -1.85838549
12 -0.37695692   0.16145727 -6.66501951  -0.73888794   0.3878795 -1.76346857
13 -0.68982834  -3.16081217 -3.69530475   4.00982569  -2.0493763 -2.34247353
14 -2.35691639   3.07489963 -3.91324021   2.47012044  -1.2370659  0.51794165
15 -1.83434616   0.38465670  0.30667628   8.97911675  -2.7821774 -3.47993617
16  0.60812469   0.98490258  3.50029369  -3.54666552   4.7818647  6.26417005
17 -0.83938124   0.31226085  2.52781341  -0.37819901   0.3086847  2.05818164
18  7.39232985  -3.70464400 -0.31089386 -15.53674250   2.7927774 -2.49811232
19 -7.15370224  -1.43381132  4.27383690  -1.73134538  -0.7166134 -2.33884818
20 -3.17373557  -5.18471346  4.91072140  -1.54595739  -1.2028638 -9.53821813
21 10.26496951   7.95760271 -4.17880455   2.49250147   0.5635568  0.01775095
22  1.18012767  -3.68529157  0.02900731   1.81284460   0.5428741 13.62702618
23  3.34953403   1.94489373 -1.84679859  -1.54263167   0.1655033 -3.44273249
24 -2.96610660   1.08313879 -3.09308527   1.28417740  -0.6390051  1.62293903
25  1.43975259  -0.14256661  1.67560598   1.39370914  -0.7902024  4.29855563
26 -5.11962209  -3.28573738 -5.22796573  -0.51164974  -1.2157883 -4.36560038
27  1.06629076  -1.59173496 -1.41506116   1.97238385   0.2820246  3.29120171
28  1.43058501   8.22129552  9.72874037   0.51219199  -2.2096330 -0.54335063
29 -1.52104997  -3.36920736 -1.25005223   1.25530012   0.6384861  2.35466757
30 -0.06956993   2.86956411 -1.40585973  -3.90854490   1.0818296 -4.41728705
31 -4.11601757   3.29818386 -2.15748396   1.58932864   8.0769516 -4.90128663
32 -5.03367748  -1.83375789  3.84772041  -6.04076863 -12.9402264  3.93261030
33  5.34425937   2.94495735 -9.69332684   0.08458291  -3.6513981 -0.76163429
34  2.56434705  -2.90234395  6.22981333   5.09269624  12.3889515  0.25824971
35 -0.52568978  -0.88342663  1.49118954   2.02672904   1.9326519  0.17483454
36  2.55816570   0.02070899 -1.28037395  -1.56778825  -3.8626423  1.02938025
37 -1.17326061   0.74635424 -0.59452075   0.80442154   0.3082003 -0.13475604
38 -2.36202935   0.03348225 -0.09335870  -0.19182065   1.7822421  0.74068266
39 -1.83719552   3.27576827 -2.42193786  -0.47562940   0.4146335  0.75759650
40 16.43664788  -9.86149959  4.86450721   4.66172668  -4.5256906 -3.98390572
41 -2.31254814   2.36090951 -0.92450744  -1.00745238   0.0894806  0.55069750
42 -4.03826010   1.69407726  0.14844060  -0.91842526   0.7125783 -0.81526889
          PC30        PC31        PC32        PC33        PC34        PC35
1   1.98909766   4.7260318  6.48484691   1.4490503 -5.63356066 -1.78399036
2   0.56565387  -0.4882059  0.06167291   0.7443615  0.47451268 -0.67406089
3  -4.65986975  -5.2880644  0.33457821   1.0728234  1.04986851 -2.26846419
4  -2.83551509   1.4415868 -2.86193910   2.2585881 -3.97475020 -0.05040777
5  -1.17872048  -1.5300158 -4.74829932  -5.1835874  9.51040301  0.44638424
6   3.39733050  -0.9792258 -4.60348317   1.3428782 -4.58818557  2.57367564
7  -5.96066747  -3.7701195 -1.59826924  -1.0671532 -0.61859747  4.22463264
8  -5.29328140  -2.3509153 -1.87032020  -5.0378461 -2.41453072  2.50179516
9  -1.81936985   3.9787419 -1.74273153   0.2562946 -0.92239335 -2.07272082
10  8.86636832   1.0710847  1.42671454   3.8104629  1.00930252 -7.07797067
11 -4.60632258   4.3018614  7.48308610   2.9321763  2.42985407  3.40828665
12  8.29631736   0.3092183 -0.23581712  -0.3685377  2.01350050  0.22005717
13 -1.28726760   2.1652825 -5.14667467  10.4946703 -4.53981570  2.03196527
14  2.96704264   1.1375870  3.70022861  -2.3668685  5.72053569 -2.18673165
15  1.42801320  -6.2552013 -3.02940803  -1.4182173 -1.67051225  1.99408130
16 -7.54911969   5.4063614  2.35616123   4.4869653  5.34968845 -2.26274166
17 -0.09518943   1.6860481  5.01495256 -12.7892810 -5.72975327 -2.53633536
18  3.12673310  -4.7940448 -1.74261965   0.5378946  1.01237135  1.68399827
19  0.62543406   1.3247692 -1.32447610   0.8931138  1.35917790 -0.07266038
20  1.14266399   4.2356563 -4.97638429  -3.6753881 -0.02964374 -2.00725378
21 -2.63673480   0.9840005  4.24963063  -0.9433468 -3.96915663 -2.30170295
22  3.10636431  -6.9197447  1.11150907   1.4432212  0.95843236  3.26403914
23 -1.98375761   1.8308995  0.91141514   0.4389247 -0.65519604 -0.48646533
24  1.60383488  -2.7214682  1.76363821   1.8513036  0.58998447  1.04002134
25 -1.94272692   2.2712491 -5.49395198   0.2413451  2.40048507 -8.33110458
26 -5.26661455 -10.5762982  4.47428671   2.3453534 -0.31123076 -5.01525395
27  1.04676051   5.5850430 -5.40837658  -2.0008502  2.10406111  2.21403157
28  4.97768532  -0.9974523  2.34715594   0.6227548 -3.41427761 -0.24058343
29  0.21017185   2.9765138 -3.27997127  -2.0561031 -2.91985625  4.05918116
30  1.77884841   3.8406665  6.34766231   0.3502175  4.02495836  9.42878881
31 -0.73995225  -1.1025223  1.07230333   1.5674210 -0.09614324  0.11444108
32  1.97760245  -0.4866522  0.88394102  -0.3292164  0.54941608  0.88404500
33 -4.86224361   1.6961836 -1.12826893  -1.5238481 -0.66011710 -0.32935962
34  4.89983267  -0.5181957 -0.26885435   0.7077649  0.29324448 -0.36781328
35  1.11423733  -0.4415861  0.55464442  -0.5189614  0.15284715  0.36503844
36 -2.58750888   0.9849878 -1.65928918   0.2035562 -0.41150116 -0.16223061
37  0.41992239  -0.1238660 -0.24123640  -0.4631898  0.33880931  0.12192526
38 -0.05129538   0.9044180  0.56106808   0.5909385 -0.74257323  0.33537011
39  0.62328619   0.4121197 -0.26218006  -0.4947233 -0.41108708 -0.02870325
40  1.07027851  -2.3477427  2.00865214   0.3159687  2.57329229 -0.30244740
41 -0.24029867   0.1317647 -0.56802066   0.2182748 -0.18987218 -0.41373681
42  0.36297648  -1.7107546 -0.95757622  -0.9392056 -0.01199114  0.06098048
           PC36        PC37         PC38        PC39         PC40         PC41
1  -1.968373066  0.08743453  2.183284000  1.38512281  0.657904098 -1.367653014
2   0.496071017  0.45738545  0.688974500 -0.50069000 -0.009600091 -0.134431653
3  -7.254188081 -0.87953501  5.254089295 -5.06310456 -1.874494952 -1.297001041
4  -0.002815320  2.12330544 -8.565729148  8.58846345  3.720107875 -1.134189866
5   6.857337016 -1.12373892 -2.492367893 -1.10104754 -1.657842884  2.907622762
6  -0.399028677  1.29301214  2.895845846 -0.87396720 -1.702447136  0.184672925
7  -2.248154438  0.21379701 -0.092092437 -0.46740722  2.182677059 -1.038378919
8  -0.270965570  5.00139082 -0.740223559 -0.20482031 -1.243252006 -0.212252386
9   5.751702300 -8.07028310  6.894571527  6.59394772 -2.226526423  2.039050951
10 -0.873187625  5.92107302 -4.139671908 -0.89668731 -6.508383855  1.712829146
11  0.018798868 -1.47960277  0.664499634 -1.43674910 -0.894106159 -0.042052758
12 -1.349175086 -3.99912413 -1.580093949 -4.25775719  9.786208491 -1.620295031
13  5.292226647  1.03987726  0.026459416 -4.75125842 -0.140153248  0.215067591
14 -6.662578320 -2.06216215 -0.156746180  5.28864832 -0.293867184 -0.344734418
15 -2.383135531  1.03610497  3.560699420  4.38029302 -0.244483024 -0.222110485
16 -0.013680698 -1.34547715 -2.950921834 -2.50088378  0.971825582  0.744165626
17  5.428969151  1.61580344 -0.591522944 -3.70053735 -0.033986573 -0.301461311
18 -0.488427152 -0.97997108 -0.022586658  0.61818386 -0.888131738 -0.206280873
19  0.468068053  0.06549281  0.726755731  0.34635322 -0.212980499 -0.069093034
20 -0.979522497 -0.02468945 -1.365916169 -1.33030040  0.416219432 -0.022258077
21 -1.688311138 -0.34068219 -0.031161975 -0.04175512 -0.054647286  0.054414697
22  2.162172352 -0.18975509 -0.214576384  0.46804979  0.182633997  0.037654304
23 -0.650926651  0.02156579  0.007655565  0.12130968 -0.093599995  0.152411353
24  0.072783992  0.15319058  0.488455037  0.01973074 -0.014169458  0.057861403
25  0.044053906  6.58487095  6.554990946  0.81806868  5.236516385  1.327603017
26  2.876883827 -1.79837015 -3.433268825  0.72207101 -0.496024046 -0.820387628
27 -1.488539019 -1.98865100 -1.273963204 -0.50531050 -4.755980682 -9.691054441
28  3.533429569 -5.23432587 -0.378739308 -1.27704730  0.282394014  0.112270817
29 -7.416444067 -3.38141570 -3.208278133 -1.56499184 -1.627341256  8.587114543
30  2.552458900  7.33770737  3.024846835  2.05529869  0.906811169  0.903192156
31  0.533465536  0.55038275 -0.402579158  0.54185930 -0.423337461 -0.581651952
32  0.024069031  0.46156857  0.800027053 -0.29774260 -0.041887910  0.006100387
33 -1.074163097 -0.30215185 -1.029860378 -0.91210017  0.332776421 -0.187712323
34  0.931013462 -0.56297632  0.477197306  0.33005106  0.049714387  0.287185452
35 -0.002180162 -0.11993097  0.344445625 -0.02222316 -0.054037292  0.194533887
36 -0.386775249 -0.30666457 -0.707878030  0.08471710  0.081957203 -0.130536334
37 -0.259745546 -0.06880176  0.084821298 -0.07184577  0.043686686  0.026724165
38  0.002056925 -0.16927098  0.017402117 -0.18636517  0.068600195  0.193848986
39  0.186286825 -0.08974013 -0.071708997 -0.25077191  0.084627388 -0.190296008
40 -0.334902270  0.42920553 -0.342826901 -0.11817729 -0.051562582  0.248806625
41  0.200412966 -0.04431145 -0.231757615  0.15196991 -0.200716212 -0.091958720
42  0.762958917  0.16846337 -0.670549563 -0.18059716  0.738899567 -0.287340521
            PC42   Group  Hybrid genotype Female_Parent length_2Y RLN_2Y
1   7.776258e-14 29JM-11 29JM-11   29JM11         29.11     96.30  68.90
2  -1.248402e-14 29JM-11 29JM-11   29JM11         29.11     96.30  68.90
3   3.372330e-15 29JM-11 29JM-11   29JM11         29.11     96.30  68.90
4   1.319108e-14 29JM-11 29JM-11   29JM11         29.11     96.30  68.90
5   7.119872e-14 29JM-11 29JM-11   29JM11         29.11     96.30  68.90
6   9.874100e-16 29JM-11 29JM-11   29JM11         29.11     96.30  68.90
7  -1.141261e-13  JMS-12  JMS-12   JMS 12         29.11     63.40  26.70
8   1.825566e-14  JMS-12  JMS-12   JMS 12         29.11     63.40  26.70
9   2.123407e-14  JMS-12  JMS-12   JMS 12         29.11     63.40  26.70
10 -6.477105e-14  JMS-12  JMS-12   JMS 12         29.11     63.40  26.70
11 -2.366399e-14  JMS-12  JMS-12   JMS 12         29.11     63.40  26.70
12  5.686662e-14  JMS-12  JMS-12   JMS 12         29.11     63.40  26.70
13  4.303256e-14 MS1-122 MS1-122  MS1 122         31.01    149.33   3.70
14  1.393318e-14 MS1-122 MS1-122  MS1 122         31.01    149.33   3.70
15  5.055080e-14 MS1-122 MS1-122  MS1 122         31.01    149.33   3.70
16 -3.917789e-14 MS1-122 MS1-122  MS1 122         31.01    149.33   3.70
17  8.252552e-14 MS1-122 MS1-122  MS1 122         31.01    149.33   3.70
18  6.655965e-14 MS1-122 MS1-122  MS1 122         31.01    149.33   3.70
19  6.472226e-15  MS1-36  MS1-36   MS1 36         31.01     24.33   1.29
20 -3.206078e-14  MS1-36  MS1-36   MS1 36         31.01     24.33   1.29
21  3.138250e-15  MS1-36  MS1-36   MS1 36         31.01     24.33   1.29
22 -3.111273e-14  MS1-36  MS1-36   MS1 36         31.01     24.33   1.29
23 -7.561147e-14  MS1-36  MS1-36   MS1 36         31.01     24.33   1.29
24 -1.725855e-14  MS1-36  MS1-36   MS1 36         31.01     24.33   1.29
25 -3.421200e-14  MS1-41  MS1-41   MS1 41         31.01    117.50 247.80
26 -3.189444e-14  MS1-41  MS1-41   MS1 41         31.01    117.50 247.80
27  2.823894e-14  MS1-41  MS1-41   MS1 41         31.01    117.50 247.80
28  1.105233e-14  MS1-41  MS1-41   MS1 41         31.01    117.50 247.80
29  4.923714e-15  MS1-41  MS1-41   MS1 41         31.01    117.50 247.80
30 -1.814496e-14  MS1-41  MS1-41   MS1 41         31.01    117.50 247.80
31 -1.386015e-13  MS1-56  MS1-56   MS1 56         31.01     44.33  16.40
32  1.229678e-14  MS1-56  MS1-56   MS1 56         31.01     44.33  16.40
33 -8.224958e-14  MS1-56  MS1-56   MS1 56         31.01     44.33  16.40
34 -1.394758e-14  MS1-56  MS1-56   MS1 56         31.01     44.33  16.40
35 -6.085505e-14  MS1-56  MS1-56   MS1 56         31.01     44.33  16.40
36  6.903882e-14  MS1-56  MS1-56   MS1 56         31.01     44.33  16.40
37 -5.098135e-14  STJM-4  STJM-4   STJM 4         29.11     74.30  35.20
38  1.801022e-14  STJM-4  STJM-4   STJM 4         29.11     74.30  35.20
39  4.076774e-14  STJM-4  STJM-4   STJM 4         29.11     74.30  35.20
40 -3.455526e-14  STJM-4  STJM-4   STJM 4         29.11     74.30  35.20
41 -1.188448e-14  STJM-4  STJM-4   STJM 4         29.11     74.30  35.20
42  5.797636e-14  STJM-4  STJM-4   STJM 4         29.11     74.30  35.20
   length_3Y RLN_3Y Over_length_3Y Over_RLN_3Y   CG_Avg  PHY_Avg RLN_2_3Y
1     152.80   17.9             NA          NA 2.800000 32.70450       NA
2     152.80   17.9             NA          NA 2.800000 32.70450       NA
3     152.80   17.9             NA          NA 2.800000 32.70450       NA
4     152.80   17.9             NA          NA 2.800000 32.70450       NA
5     152.80   17.9             NA          NA 2.800000 32.70450       NA
6     152.80   17.9             NA          NA 2.800000 32.70450       NA
7         NA     NA             NA          NA 3.400000 26.29900       NA
8         NA     NA             NA          NA 3.400000 26.29900       NA
9         NA     NA             NA          NA 3.400000 26.29900       NA
10        NA     NA             NA          NA 3.400000 26.29900       NA
11        NA     NA             NA          NA 3.400000 26.29900       NA
12        NA     NA             NA          NA 3.400000 26.29900       NA
13    173.00   68.9         100.56        26.7 3.300000 39.70000     33.1
14    173.00   68.9         100.56        26.7 3.300000 39.70000     33.1
15    173.00   68.9         100.56        26.7 3.300000 39.70000     33.1
16    173.00   68.9         100.56        26.7 3.300000 39.70000     33.1
17    173.00   68.9         100.56        26.7 3.300000 39.70000     33.1
18    173.00   68.9         100.56        26.7 3.300000 39.70000     33.1
19     37.50   34.1             NA          NA 1.500000 15.97925       NA
20     37.50   34.1             NA          NA 1.500000 15.97925       NA
21     37.50   34.1             NA          NA 1.500000 15.97925       NA
22     37.50   34.1             NA          NA 1.500000 15.97925       NA
23     37.50   34.1             NA          NA 1.500000 15.97925       NA
24     37.50   34.1             NA          NA 1.500000 15.97925       NA
25    139.56   63.9             NA          NA 2.133333 35.02627       NA
26    139.56   63.9             NA          NA 2.133333 35.02627       NA
27    139.56   63.9             NA          NA 2.133333 35.02627       NA
28    139.56   63.9             NA          NA 2.133333 35.02627       NA
29    139.56   63.9             NA          NA 2.133333 35.02627       NA
30    139.56   63.9             NA          NA 2.133333 35.02627       NA
31     44.94   13.0             NA          NA 1.075000 19.50000       NA
32     44.94   13.0             NA          NA 1.075000 19.50000       NA
33     44.94   13.0             NA          NA 1.075000 19.50000       NA
34     44.94   13.0             NA          NA 1.075000 19.50000       NA
35     44.94   13.0             NA          NA 1.075000 19.50000       NA
36     44.94   13.0             NA          NA 1.075000 19.50000       NA
37        NA     NA             NA          NA 2.767500 31.54750       NA
38        NA     NA             NA          NA 2.767500 31.54750       NA
39        NA     NA             NA          NA 2.767500 31.54750       NA
40        NA     NA             NA          NA 2.767500 31.54750       NA
41        NA     NA             NA          NA 2.767500 31.54750       NA
42        NA     NA             NA          NA 2.767500 31.54750       NA

$plot</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="read-in-cg-results" class="level2">
<h2 class="anchored" data-anchor-id="read-in-cg-results">Read in CG results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># impCG = fread('DGEresults/Limma_results_table_CG.csv')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="section-1" class="level1">
<h1>*</h1>
</section>
<section id="phy-fit-model-for-genes-associated-with-phy-score" class="level1">
<h1>PHY Fit model for genes associated with PHY score</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mm_PHY = model.matrix(~poly(PHY_Avg, 2), </span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                       data = metadata_S)</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>mm_PHY <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>PHY_Avg,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> metadata_S)</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mm_PHY)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          (Intercept)   PHY_Avg
29JM-11-1           1 0.5030946
29JM-11-2           1 0.5030946
29JM-11-3           1 0.5030946
29JM-11-4           1 0.5030946
29JM-11-5           1 0.5030946
29JM-11-6           1 0.5030946</code></pre>
</div>
</div>
<section id="filtered-mean-variance-trend-phy" class="level2">
<h2 class="anchored" data-anchor-id="filtered-mean-variance-trend-phy">Filtered mean-variance trend PHY</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>y_PHY <span class="ot">&lt;-</span> <span class="fu">voom</span>(d, </span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>              mm_PHY,</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">plot =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="fitting-linear-models-in-limma-with-random-effects-phy" class="level2">
<h2 class="anchored" data-anchor-id="fitting-linear-models-in-limma-with-random-effects-phy">Fitting linear models in limma with random effects PHY</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Need to tell limma where the within class correlation is coming from</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>dupcor_PHY <span class="ot">=</span> <span class="fu">duplicateCorrelation</span>(y_PHY,</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>                                  mm_PHY,</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">block =</span> metadata_S<span class="sc">$</span>Hybrid)</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="do">## How correlated are the hybrid replicates on average?</span></span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>consensus.corr.PHY <span class="ot">=</span> dupcor_PHY<span class="sc">$</span>consensus.correlation</span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>consensus.corr.PHY</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.6073402</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lmFit fits a linear model using weighted least squares for each gene:</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>fit_PHY <span class="ot">=</span> <span class="fu">lmFit</span>(y_PHY,</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">design =</span> mm_PHY,</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">block =</span> metadata_S<span class="sc">$</span>Hybrid, </span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> consensus.corr.PHY) </span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Ebayes</span></span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>BlockFit_PHY <span class="ot">=</span> <span class="fu">eBayes</span>(fit_PHY)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="limma-results-phy" class="level1">
<h1>Limma results PHY</h1>
<section id="results-summary-1" class="level2">
<h2 class="anchored" data-anchor-id="results-summary-1">Results summary</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>res_summaries_PHY <span class="ot">=</span> BlockFit_PHY <span class="sc">%&gt;%</span> <span class="fu">decideTests</span>() <span class="sc">%&gt;%</span> <span class="fu">summary</span>()</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>res_summaries_PHY</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       (Intercept) PHY_Avg
Down            25    1061
NotSig         534   17731
Up           18640     407</code></pre>
</div>
</div>
</section>
<section id="table-of-results-phy" class="level2">
<h2 class="anchored" data-anchor-id="table-of-results-phy">Table of results PHY</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>impPHY <span class="ot">=</span> limma<span class="sc">::</span><span class="fu">topTable</span>(BlockFit_PHY, </span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">'PHY_Avg'</span>,</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sort.by =</span> <span class="st">"logFC"</span>,</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">p.value =</span> <span class="fl">0.05</span>, </span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">adjust.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">number =</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">R =</span> <span class="fu">sqrt</span>(t<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>(t<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">40</span>)), <span class="at">AveExpr =</span> <span class="dv">2</span><span class="sc">^</span>AveExpr)</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(impPHY, <span class="st">'DGEresults/unAnnotated_results_PHY.csv'</span>)</span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>rawPHY <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'DGEresults/unAnnotated_results_PHY.csv'</span>)</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(impPHY)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1468    8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="do">## adding Hybrid as a blocking variable reduced DEGs by several thousand</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(impPHY)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID     logFC    AveExpr         t      P.Value    adj.P.Val         B
1 121261538  4.159943  0.9133184  6.689148 3.165144e-08 1.012793e-04 6.2778728
2 108996798 -3.856073  1.1670564 -6.986438 1.154440e-08 7.354236e-05 7.5594054
3 118345731 -3.813991 30.9814417 -5.690169 9.408471e-07 1.062548e-03 5.5241911
4 109019660  3.658527  0.7832938  6.929394 1.400615e-08 7.354236e-05 6.6664262
5 108983669 -3.641581  9.4120554 -4.282414 9.777842e-05 1.289556e-02 1.2883673
6 109001297 -3.578161 12.7932313 -3.993382 2.420869e-04 1.822677e-02 0.4764714
          R
1 0.7266320
2 0.7413514
3 0.6688394
4 0.7386066
5 0.5606720
6 0.5338903</code></pre>
</div>
</div>
</section>
<section id="merge-annotation-with-results-phy" class="level2">
<h2 class="anchored" data-anchor-id="merge-annotation-with-results-phy">Merge annotation with results PHY</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>impPHY <span class="ot">=</span> impPHY <span class="sc">%&gt;%</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(annotation_combined, <span class="at">by =</span> <span class="st">"GeneID"</span>)</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(impPHY)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID     logFC    AveExpr         t      P.Value    adj.P.Val        B
1 121261538  4.159943  0.9133184  6.689148 3.165144e-08 1.012793e-04 6.277873
2 108996798 -3.856073  1.1670564 -6.986438 1.154440e-08 7.354236e-05 7.559405
3 108996798 -3.856073  1.1670564 -6.986438 1.154440e-08 7.354236e-05 7.559405
4 118345731 -3.813991 30.9814417 -5.690169 9.408471e-07 1.062548e-03 5.524191
5 109019660  3.658527  0.7832938  6.929394 1.400615e-08 7.354236e-05 6.666426
6 108983669 -3.641581  9.4120554 -4.282414 9.777842e-05 1.289556e-02 1.288367
          R feature  class chromosome    start      end
1 0.7266320    mRNA                4D 30605557 30611409
2 0.7413514    mRNA                 6  2953242  2958879
3 0.7413514    mRNA                 6  2953242  2958786
4 0.6688394    mRNA                      31540    33123
5 0.7386066    mRNA                13  9506430  9511614
6 0.5606720   ncRNA lncRNA         12 30459235 30462813
                                                                 name Jr-GeneID
1 endoplasmic reticulum-Golgi intermediate compartment protein 3-like 109003827
2                              wall-associated receptor kinase 2-like        NA
3                              wall-associated receptor kinase 2-like        NA
4                                                endochitinase 2-like        NA
5                                  lipoxygenase 6, chloroplastic-like        NA
6                                        uncharacterized LOC108983669        NA
  Jr-chromosome         Subcell_loc Parent_haplotype
1             8 endomembrane system     J.microcarpa
2          &lt;NA&gt;     plasma membrane          J.regia
3          &lt;NA&gt;     plasma membrane          J.regia
4          &lt;NA&gt; extracellular space          J.regia
5          &lt;NA&gt;             nucleus          J.regia
6          &lt;NA&gt;                              J.regia</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(impPHY, <span class="st">'DGEresults/Limma_results_table_PHY.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pca-of-phy-degs" class="level2">
<h2 class="anchored" data-anchor-id="pca-of-phy-degs">PCA of PHY DEGs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">=</span> impPHY <span class="sc">%&gt;%</span> </span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">as.character</span>(GeneID))</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>PCA_PHY <span class="ot">=</span> cpmd <span class="sc">%&gt;%</span> </span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(ids) <span class="sc">%&gt;%</span> </span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(GeneID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>pca_plot_PHY <span class="ot">=</span> <span class="fu">plot_pca</span>(PCA_PHY, metadata,</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'PHY_Avg'</span>,</span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'PHY_Avg'</span>,</span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'2D'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>pca_plot_PHY<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="read-in-phy-results" class="level2">
<h2 class="anchored" data-anchor-id="read-in-phy-results">Read in PHY results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># impPHY = fread('DGEresults/Limma_results_table_PHY.csv')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="section-2" class="level1">
<h1>*</h1>
</section>
<section id="length_2y-fit-model-for-genes-associated-with-length_2y-score" class="level1">
<h1>length_2Y Fit model for genes associated with length_2Y score</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mm_length_2Y = model.matrix(~poly(length_2Y, 2), </span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                       data = metadata_S)</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>mm_length_2Y <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>length_2Y,</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> metadata_S)</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mm_length_2Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          (Intercept) length_2Y
29JM-11-1           1 0.3700379
29JM-11-2           1 0.3700379
29JM-11-3           1 0.3700379
29JM-11-4           1 0.3700379
29JM-11-5           1 0.3700379
29JM-11-6           1 0.3700379</code></pre>
</div>
</div>
<section id="filtered-mean-variance-trend-length_2y" class="level2">
<h2 class="anchored" data-anchor-id="filtered-mean-variance-trend-length_2y">Filtered mean-variance trend length_2Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>y_length_2Y <span class="ot">&lt;-</span> <span class="fu">voom</span>(d, </span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>              mm_length_2Y,</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">plot =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="fitting-linear-models-in-limma-with-random-effects-length_2y" class="level2">
<h2 class="anchored" data-anchor-id="fitting-linear-models-in-limma-with-random-effects-length_2y">Fitting linear models in limma with random effects length_2Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Need to tell limma where the within class correlation is coming from</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>dupcor_length_2Y <span class="ot">=</span> <span class="fu">duplicateCorrelation</span>(y_length_2Y,</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>                                  mm_length_2Y,</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">block =</span> metadata_S<span class="sc">$</span>Hybrid)</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="do">## How correlated are the hybrid replicates on average?</span></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>consensus.corr.length_2Y <span class="ot">=</span> dupcor_length_2Y<span class="sc">$</span>consensus.correlation</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>consensus.corr.length_2Y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.6138877</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lmFit fits a linear model using weighted least squares for each gene:</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>fit_length_2Y <span class="ot">=</span> <span class="fu">lmFit</span>(y_length_2Y,</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">design =</span> mm_length_2Y,</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">block =</span> metadata_S<span class="sc">$</span>Hybrid, </span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> consensus.corr.length_2Y) </span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Ebayes</span></span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>BlockFit_length_2Y <span class="ot">=</span> <span class="fu">eBayes</span>(fit_length_2Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="limma-results-length_2y" class="level1">
<h1>Limma results length_2Y</h1>
<section id="results-summary-2" class="level2">
<h2 class="anchored" data-anchor-id="results-summary-2">Results summary</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>res_summaries_length_2Y <span class="ot">=</span> BlockFit_length_2Y <span class="sc">%&gt;%</span> <span class="fu">decideTests</span>() <span class="sc">%&gt;%</span> <span class="fu">summary</span>()</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>res_summaries_length_2Y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       (Intercept) length_2Y
Down            25       357
NotSig         553     18653
Up           18621       189</code></pre>
</div>
</div>
</section>
<section id="table-of-results-length_2y" class="level2">
<h2 class="anchored" data-anchor-id="table-of-results-length_2y">Table of results length_2Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>implength_2Y <span class="ot">=</span> limma<span class="sc">::</span><span class="fu">topTable</span>(BlockFit_length_2Y, </span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">'length_2Y'</span>,</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sort.by =</span> <span class="st">"logFC"</span>,</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">p.value =</span> <span class="fl">0.05</span>, </span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">adjust.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">number =</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">R =</span> <span class="fu">sqrt</span>(t<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>(t<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">40</span>)), <span class="at">AveExpr =</span> <span class="dv">2</span><span class="sc">^</span>AveExpr)</span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(implength_2Y, <span class="st">'DGEresults/unAnnotated_results_length_2Y.csv'</span>)</span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a>rawlength_2Y <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'DGEresults/unAnnotated_results_length_2Y.csv'</span>)</span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(implength_2Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 546   8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="do">## adding Hybrid as a blocking variable reduced DEGs by several thousand</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(implength_2Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID     logFC    AveExpr         t      P.Value    adj.P.Val         B
1 118345731 -4.809887 30.9814417 -5.437351 2.211158e-06 0.0021226012  4.660267
2 108996798 -4.267855  1.1670564 -7.003638 1.093682e-08 0.0001186481  7.275782
3 109001297 -3.774253 12.7932313 -3.410120 1.395388e-03 0.0496357850 -1.060016
4 108984807 -3.621439  2.6203150 -4.777141 1.982674e-05 0.0086689593  2.196260
5 108987644  3.556584  0.7223902  6.778789 2.343547e-08 0.0001417472  6.797599
6 108985225 -3.533080  0.8902558 -5.554941 1.488571e-06 0.0019132387  3.713741
          R
1 0.6519175
2 0.7421717
3 0.4745950
4 0.6027197
5 0.7311798
6 0.6599134</code></pre>
</div>
</div>
</section>
<section id="merge-annotation-with-results-length_2y" class="level2">
<h2 class="anchored" data-anchor-id="merge-annotation-with-results-length_2y">Merge annotation with results length_2Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>implength_2Y <span class="ot">=</span> implength_2Y <span class="sc">%&gt;%</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(annotation_combined, <span class="at">by =</span> <span class="st">"GeneID"</span>)</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(implength_2Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID     logFC    AveExpr         t      P.Value    adj.P.Val         B
1 118345731 -4.809887 30.9814417 -5.437351 2.211158e-06 0.0021226012  4.660267
2 108996798 -4.267855  1.1670564 -7.003638 1.093682e-08 0.0001186481  7.275782
3 108996798 -4.267855  1.1670564 -7.003638 1.093682e-08 0.0001186481  7.275782
4 109001297 -3.774253 12.7932313 -3.410120 1.395388e-03 0.0496357850 -1.060016
5 108984807 -3.621439  2.6203150 -4.777141 1.982674e-05 0.0086689593  2.196260
6 108987644  3.556584  0.7223902  6.778789 2.343547e-08 0.0001417472  6.797599
          R feature  class chromosome    start      end
1 0.6519175    mRNA                      31540    33123
2 0.7421717    mRNA                 6  2953242  2958879
3 0.7421717    mRNA                 6  2953242  2958786
4 0.4745950    mRNA                 1   797531   798444
5 0.6027197   ncRNA lncRNA          2 36722389 36725188
6 0.7311798    mRNA                10 34437942 34441136
                                      name Jr-GeneID Jr-chromosome
1                     endochitinase 2-like        NA          &lt;NA&gt;
2   wall-associated receptor kinase 2-like        NA          &lt;NA&gt;
3   wall-associated receptor kinase 2-like        NA          &lt;NA&gt;
4 light-regulated protein 1, chloroplastic        NA          &lt;NA&gt;
5             uncharacterized LOC108984807        NA          &lt;NA&gt;
6 triphosphate tunnel metalloenzyme 3-like        NA          &lt;NA&gt;
          Subcell_loc Parent_haplotype
1 extracellular space          J.regia
2     plasma membrane          J.regia
3     plasma membrane          J.regia
4       mitochondrion          J.regia
5                              J.regia
6             nucleus          J.regia</code></pre>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(implength_2Y<span class="sc">$</span>GeneID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 546</code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(implength_2Y, <span class="st">'DGEresults/Limma_results_table_length_2Y.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="section-3" class="level1">
<h1>*</h1>
</section>
<section id="nem-3y-and-length-3y-create-dgelist-object" class="level1">
<h1>NEM 3Y and length 3Y Create DGEList object</h1>
</section>
<section id="have-to-do-this-becuse-3y-data-are-missing-two-samples-so-cant-use-dat1" class="level1">
<h1>Have to do this becuse 3Y data are missing two samples, so can’t use ‘dat1’</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"C:/Users/hsaxe/OneDrive/Documents/ALAB/Transcriptome_data/Root/SCRI_ROOT_RNAseq_counts_combined_genomes.txt"</span>)</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">=</span> dat2 <span class="sc">%&gt;%</span> </span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">gsub</span>(<span class="st">"LOC"</span>, <span class="st">""</span>, GeneID)) <span class="sc">%&gt;%</span> </span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">matches</span>(<span class="st">'JMS|STJM'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a>dds2 <span class="ot">=</span> <span class="fu">DGEList</span>(dat2)</span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dds2<span class="sc">$</span>counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 67924    30</code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Calculate library normalization factors (does not do anything to data)</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>dds2 <span class="ot">=</span> <span class="fu">calcNormFactors</span>(dds2)</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="do">## These are the size factors (normalization factors) for each sample</span></span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a><span class="co"># dds$samples</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>metadata_S2 <span class="ot">=</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(RLN_3Y <span class="sc">!=</span> <span class="fu">is.na</span>(RLN_3Y)) <span class="sc">%&gt;%</span> </span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.numeric, scale) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="filter-by-max-cpm-this-is-used-for-dge-analysis-1" class="level2">
<h2 class="anchored" data-anchor-id="filter-by-max-cpm-this-is-used-for-dge-analysis-1">Filter by max CPM (This is used for DGE analysis)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">=</span> <span class="fu">expression_filter</span>(dds2, </span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">DGEList =</span> T,</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">FilterFUN =</span> max,</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">FilterThreshold =</span> <span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will negatively impact filtering</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Data has no missing values"
[1] "Removed 49942 features based on max threshold of 30, 17982 remaining"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="do">## CPM normalized counts of all data</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a><span class="co"># cpm = cpm(dds, prior.count = 2, log = F) </span></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a><span class="co"># cpm = cpm %&gt;% </span></span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   data.frame() %&gt;% </span></span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   rownames_to_column(var = 'GeneID') %&gt;%  </span></span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   rename_with(~ gsub('X', '', gsub('\\.', '-', .x)))</span></span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a><span class="co"># fwrite(cpm, 'DGEresults/cpm_plotting_data.csv')</span></span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a><span class="do">## CPM normalized counts of filtered data</span></span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>cpmd2 <span class="ot">=</span> <span class="fu">cpm</span>(d2)</span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a><span class="co"># fwrite(cpmd %&gt;% data.frame() %&gt;% rownames_to_column(var = 'GeneID'), 'DGEresults/DGE_CPM_data.csv')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="nem_3y-plot-pca-to-look-in-r" class="level2">
<h2 class="anchored" data-anchor-id="nem_3y-plot-pca-to-look-in-r">NEM_3Y Plot PCA to look in R</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>pca3 <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd2, metadata,</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>               <span class="co"># Using 'Group' here becuase 'Hybrid' is already in metadata. Will cause error if I used 'Hybrid'</span></span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'RLN_3Y'</span>,</span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'RLN_3Y'</span>,</span>
<span id="cb118-12"><a href="#cb118-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'2D'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>pca3<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>pca_scatter <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd, metadata,</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>               <span class="co"># Using 'Group' here becuase 'Hybrid' is already in metadata. Will cause error if I used 'Hybrid'</span></span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Hybrid, </span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'scatter'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>plot_dat <span class="ot">=</span> pca_scatter<span class="sc">$</span>plot_dat <span class="sc">%&gt;%</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hybrid, </span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>         RLN_3Y,</span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>         PC1,</span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>         PC2,</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a>         PC3, </span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a>         PC4, </span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a>         PC5) <span class="sc">%&gt;%</span> </span>
<span id="cb123-9"><a href="#cb123-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(Hybrid, RLN_3Y), <span class="at">names_to =</span> <span class="st">'PC'</span>)</span>
<span id="cb123-10"><a href="#cb123-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-11"><a href="#cb123-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_dat, <span class="fu">aes</span>(value, RLN_3Y))<span class="sc">+</span></span>
<span id="cb123-12"><a href="#cb123-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb123-13"><a href="#cb123-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'lm'</span>)<span class="sc">+</span></span>
<span id="cb123-14"><a href="#cb123-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>PC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 10 rows containing non-finite values (`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 10 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>PC5 looks correlated with NEM_3Y</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(pca_scatter<span class="sc">$</span>plot_dat<span class="sc">$</span>PC2, pca_scatter<span class="sc">$</span>plot_dat<span class="sc">$</span>RLN_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  pca_scatter$plot_dat$PC2 and pca_scatter$plot_dat$RLN_3Y
t = -1.2139, df = 3, p-value = 0.3116
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.9666998  0.6246129
sample estimates:
       cor 
-0.5739344 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># not actually correlated</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>plot_dat <span class="ot">=</span> pca_scatter<span class="sc">$</span>plot_dat <span class="sc">%&gt;%</span></span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hybrid, </span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>         length_3Y,</span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>         PC1,</span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>         PC2,</span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a>         PC3, </span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a>         PC4, </span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>         PC5) <span class="sc">%&gt;%</span> </span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(Hybrid, length_3Y), <span class="at">names_to =</span> <span class="st">'PC'</span>)</span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_dat, <span class="fu">aes</span>(value, length_3Y))<span class="sc">+</span></span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'lm'</span>)<span class="sc">+</span></span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>PC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 10 rows containing non-finite values (`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 10 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="pc2-looks-correlated-with-length_3y" class="level2">
<h2 class="anchored" data-anchor-id="pc2-looks-correlated-with-length_3y">PC2 looks correlated with length_3Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(pca_scatter<span class="sc">$</span>plot_dat<span class="sc">$</span>PC2, pca_scatter<span class="sc">$</span>plot_dat<span class="sc">$</span>length_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  pca_scatter$plot_dat$PC2 and pca_scatter$plot_dat$length_3Y
t = -3.6199, df = 3, p-value = 0.03625
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.99357916 -0.09695561
sample estimates:
       cor 
-0.9020593 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>c3 <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd2, metadata,</span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>             <span class="co"># Having to use 'Hybrid' here to get it into the plotting data because the function drops everything but the join_by_name when summarizing for scatter</span></span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">join_by_name =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_name =</span> Hybrid, </span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="st">'PC7'</span>,</span>
<span id="cb136-7"><a href="#cb136-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> <span class="st">'RLN_3Y'</span>,</span>
<span id="cb136-8"><a href="#cb136-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">scale =</span> T, </span>
<span id="cb136-9"><a href="#cb136-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">center =</span> T, </span>
<span id="cb136-10"><a href="#cb136-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb136-11"><a href="#cb136-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot_type =</span> <span class="st">'scatter'</span>,</span>
<span id="cb136-12"><a href="#cb136-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">summarise_for_scatter =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a>c3<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="nem_3y-fit-model-for-genes-associated-with-nem-score" class="level1">
<h1>NEM_3Y Fit model for genes associated with NEM score</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a>mm_NEM_3Y <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>RLN_3Y, </span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> metadata_S2)</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mm_NEM_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          (Intercept)     RLN_3Y
29JM-11-1           1 -0.9237004
29JM-11-2           1 -0.9237004
29JM-11-3           1 -0.9237004
29JM-11-4           1 -0.9237004
29JM-11-5           1 -0.9237004
29JM-11-6           1 -0.9237004</code></pre>
</div>
</div>
<section id="filtered-mean-variance-trend-nem" class="level2">
<h2 class="anchored" data-anchor-id="filtered-mean-variance-trend-nem">Filtered mean-variance trend NEM</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>y_NEM_3Y <span class="ot">&lt;-</span> <span class="fu">voom</span>(d2, </span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>                 mm_NEM_3Y,</span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">plot =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="fitting-linear-models-in-limma-with-random-effects-nem_3y" class="level2">
<h2 class="anchored" data-anchor-id="fitting-linear-models-in-limma-with-random-effects-nem_3y">Fitting linear models in limma with random effects NEM_3Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Need to tell limma where the within class correlation is coming from</span></span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a>dupcor_NEM_3Y <span class="ot">=</span> <span class="fu">duplicateCorrelation</span>(y_NEM_3Y,</span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a>                                      mm_NEM_3Y, </span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">block =</span> metadata_S2<span class="sc">$</span>Hybrid)</span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a><span class="do">## How correlated are the hybrid replicates on average?</span></span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a>consensus.corr._NEM_3Y <span class="ot">=</span> dupcor_NEM_3Y<span class="sc">$</span>consensus.correlation</span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a>consensus.corr._NEM_3Y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.5873101</code></pre>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lmFit fits a linear model using weighted least squares for each gene:</span></span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>fit_NEM_3Y <span class="ot">=</span> <span class="fu">lmFit</span>(y_NEM_3Y, </span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">design =</span> mm_NEM_3Y, </span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">block =</span> metadata_S2<span class="sc">$</span>Hybrid,</span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">correlation =</span> consensus.corr._NEM_3Y) </span>
<span id="cb145-6"><a href="#cb145-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-7"><a href="#cb145-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Ebayes</span></span>
<span id="cb145-8"><a href="#cb145-8" aria-hidden="true" tabindex="-1"></a>BlockFit_NEM_3Y <span class="ot">=</span> <span class="fu">eBayes</span>(fit_NEM_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="limma-results-_nem_3y" class="level1">
<h1>Limma results _NEM_3Y</h1>
<section id="result-summary" class="level2">
<h2 class="anchored" data-anchor-id="result-summary">Result summary</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>res_summaries_NEM_3Y <span class="ot">=</span> BlockFit_NEM_3Y <span class="sc">%&gt;%</span> <span class="fu">decideTests</span>() <span class="sc">%&gt;%</span> <span class="fu">summary</span>()</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a>res_summaries_NEM_3Y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       (Intercept) RLN_3Y
Down             3     13
NotSig         566  17962
Up           17413      7</code></pre>
</div>
</div>
</section>
<section id="table-of-results-_nem_3y" class="level2">
<h2 class="anchored" data-anchor-id="table-of-results-_nem_3y">Table of results _NEM_3Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>impNEM_3Y <span class="ot">=</span> <span class="fu">topTable</span>(BlockFit_NEM_3Y,</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">sort.by =</span> <span class="st">"logFC"</span>,</span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">p.value =</span> <span class="fl">0.05</span>,</span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">adjust.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">number =</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb148-7"><a href="#cb148-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">R =</span> <span class="fu">sqrt</span>(t<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>(t<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">40</span>)), <span class="at">AveExpr =</span> <span class="dv">2</span><span class="sc">^</span>AveExpr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Removing intercept from test coefficients</code></pre>
</div>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(impNEM_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 20  8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="do">## adding Hybrid as a blocking variable reduced DEGs by several thousand</span></span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(impNEM_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID     logFC   AveExpr         t      P.Value    adj.P.Val        B
1 109021176 -7.104981  1.377514 -7.129332 4.245672e-08 0.0002909522 4.632277
2 121268810 -5.803453  3.158092 -4.967693 2.166036e-05 0.0278211820 1.721509
3 121261538  4.059037  1.599117  4.881351 2.784137e-05 0.0333762331 1.048592
4 118348788  3.426551  2.769002  5.274343 8.864440e-06 0.0177111505 2.413891
5 121262079 -3.161700 10.156697 -5.039740 1.756279e-05 0.0242933856 2.837121
6 109011537  3.051927 13.394997  5.133816 1.335287e-05 0.0240111314 3.159639
          R
1 0.7480669
2 0.6176985
3 0.6109925
4 0.6404624
5 0.6231931
6 0.6302313</code></pre>
</div>
</div>
</section>
<section id="merge-annotation-with-results-_nem_3y" class="level2">
<h2 class="anchored" data-anchor-id="merge-annotation-with-results-_nem_3y">Merge annotation with results _NEM_3Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a>impNEM_3Y <span class="ot">=</span> impNEM_3Y <span class="sc">%&gt;%</span></span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(annotation_combined, <span class="at">by =</span> <span class="st">"GeneID"</span>)</span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(impNEM_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID     logFC  AveExpr         t      P.Value    adj.P.Val        B
1 109021176 -7.104981 1.377514 -7.129332 4.245672e-08 0.0002909522 4.632277
2 109021176 -7.104981 1.377514 -7.129332 4.245672e-08 0.0002909522 4.632277
3 109021176 -7.104981 1.377514 -7.129332 4.245672e-08 0.0002909522 4.632277
4 121268810 -5.803453 3.158092 -4.967693 2.166036e-05 0.0278211820 1.721509
5 121261538  4.059037 1.599117  4.881351 2.784137e-05 0.0333762331 1.048592
6 118348788  3.426551 2.769002  5.274343 8.864440e-06 0.0177111505 2.413891
          R feature class chromosome    start      end
1 0.7480669    mRNA               11 30069599 30078607
2 0.7480669    mRNA               11 30073566 30078607
3 0.7480669    mRNA               11 30074141 30078607
4 0.6176985    mRNA               1D  1147194  1149994
5 0.6109925    mRNA               4D 30605557 30611409
6 0.6404624    mRNA                6  1042315  1046158
                                                                 name Jr-GeneID
1                               E3 ubiquitin-protein ligase UPL1-like        NA
2                               E3 ubiquitin-protein ligase UPL1-like        NA
3                               E3 ubiquitin-protein ligase UPL1-like        NA
4                     probable receptor-like protein kinase At5g24010 109013888
5 endoplasmic reticulum-Golgi intermediate compartment protein 3-like 109003827
6                                                     laccase-14-like        NA
  Jr-chromosome         Subcell_loc Parent_haplotype
1          &lt;NA&gt;             nucleus          J.regia
2          &lt;NA&gt;             nucleus          J.regia
3          &lt;NA&gt;             nucleus          J.regia
4             7     plasma membrane     J.microcarpa
5             8 endomembrane system     J.microcarpa
6          &lt;NA&gt; extracellular space          J.regia</code></pre>
</div>
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(impNEM_3Y, <span class="st">'DGEresults/Limma_results_table_NEM_3Y.csv'</span>)</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a>impNEM_3Y <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'DGEresults/Limma_results_table_NEM_3Y.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pca-of-_nem_3y-degs" class="level2">
<h2 class="anchored" data-anchor-id="pca-of-_nem_3y-degs">PCA of _NEM_3Y DEGs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">=</span> impNEM_3Y <span class="sc">%&gt;%</span> </span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">as.character</span>(GeneID))</span>
<span id="cb157-4"><a href="#cb157-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-5"><a href="#cb157-5" aria-hidden="true" tabindex="-1"></a>PCA_NEM_3Y <span class="ot">=</span> cpmd2 <span class="sc">%&gt;%</span> </span>
<span id="cb157-6"><a href="#cb157-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb157-7"><a href="#cb157-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb157-8"><a href="#cb157-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(ids) <span class="sc">%&gt;%</span> </span>
<span id="cb157-9"><a href="#cb157-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(GeneID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a>pca_plot_NEM_3Y <span class="ot">=</span> <span class="fu">plot_pca</span>(PCA_NEM_3Y, metadata,</span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb159-5"><a href="#cb159-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb159-6"><a href="#cb159-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb159-7"><a href="#cb159-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb159-8"><a href="#cb159-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb159-9"><a href="#cb159-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'RLN_3Y'</span>,</span>
<span id="cb159-10"><a href="#cb159-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'RLN_3Y'</span>,</span>
<span id="cb159-11"><a href="#cb159-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'2D'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>pca_plot_NEM_3Y<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="read-in-_nem_3y-results" class="level2">
<h2 class="anchored" data-anchor-id="read-in-_nem_3y-results">Read in _NEM_3Y results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>impNEM_3Y <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'DGEresults/Limma_results_table_NEM_3Y.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>pca3 <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd, metadata,</span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb163-3"><a href="#cb163-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb163-4"><a href="#cb163-4" aria-hidden="true" tabindex="-1"></a>               <span class="co"># Using 'Group' here becuase 'Hybrid' is already in metadata. Will cause error if I used 'Hybrid'</span></span>
<span id="cb163-5"><a href="#cb163-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb163-6"><a href="#cb163-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb163-7"><a href="#cb163-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb163-8"><a href="#cb163-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb163-9"><a href="#cb163-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb163-10"><a href="#cb163-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'length_3Y'</span>,</span>
<span id="cb163-11"><a href="#cb163-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'length_3Y'</span>,</span>
<span id="cb163-12"><a href="#cb163-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'2D'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>pca3<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-60-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>c3 <span class="ot">=</span> <span class="fu">plot_pca</span>(cpmd, metadata,</span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a>             <span class="co"># Having to use 'Hybrid' here to get it into the plotting data because the function drops everything but the join_by_name when summarizing for scatter</span></span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">join_by_name =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb166-5"><a href="#cb166-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">plotting_factors_name =</span> Hybrid, </span>
<span id="cb166-6"><a href="#cb166-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="st">'PC7'</span>,</span>
<span id="cb166-7"><a href="#cb166-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> <span class="st">'length_3Y'</span>,</span>
<span id="cb166-8"><a href="#cb166-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">scale =</span> T, </span>
<span id="cb166-9"><a href="#cb166-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">center =</span> T, </span>
<span id="cb166-10"><a href="#cb166-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">'Hybrid'</span>,</span>
<span id="cb166-11"><a href="#cb166-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">plot_type =</span> <span class="st">'scatter'</span>,</span>
<span id="cb166-12"><a href="#cb166-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">summarise_for_scatter =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a>c3<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing non-finite values (`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing non-finite values (`stat_cor()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing non-finite values (`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_text()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="section-4" class="level1">
<h1>*</h1>
</section>
<section id="length_3y-fit-model-for-genes-associated-with-nem-score" class="level1">
<h1>Length_3Y Fit model for genes associated with NEM score</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a>mm_Length_3Y <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>length_3Y, </span>
<span id="cb175-2"><a href="#cb175-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> metadata_S2)</span>
<span id="cb175-3"><a href="#cb175-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-4"><a href="#cb175-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mm_Length_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          (Intercept) length_3Y
29JM-11-1           1 0.7477421
29JM-11-2           1 0.7477421
29JM-11-3           1 0.7477421
29JM-11-4           1 0.7477421
29JM-11-5           1 0.7477421
29JM-11-6           1 0.7477421</code></pre>
</div>
</div>
<section id="filtered-mean-variance-trend-nem-1" class="level2">
<h2 class="anchored" data-anchor-id="filtered-mean-variance-trend-nem-1">Filtered mean-variance trend NEM</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb177"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1" aria-hidden="true" tabindex="-1"></a>y_Length_3Y <span class="ot">&lt;-</span> <span class="fu">voom</span>(d2, </span>
<span id="cb177-2"><a href="#cb177-2" aria-hidden="true" tabindex="-1"></a>                 mm_Length_3Y,</span>
<span id="cb177-3"><a href="#cb177-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">plot =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-63-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="fitting-linear-models-in-limma-with-random-effects-length_3y" class="level2">
<h2 class="anchored" data-anchor-id="fitting-linear-models-in-limma-with-random-effects-length_3y">Fitting linear models in limma with random effects Length_3Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Need to tell limma where the within class correlation is coming from</span></span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a>dupcor_Length_3Y <span class="ot">=</span> <span class="fu">duplicateCorrelation</span>(y_Length_3Y,</span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a>                                      mm_Length_3Y, </span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">block =</span> metadata_S2<span class="sc">$</span>Hybrid)</span>
<span id="cb178-5"><a href="#cb178-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-6"><a href="#cb178-6" aria-hidden="true" tabindex="-1"></a><span class="do">## How correlated are the hybrid replicates on average?</span></span>
<span id="cb178-7"><a href="#cb178-7" aria-hidden="true" tabindex="-1"></a>consensus.corr._Length_3Y <span class="ot">=</span> dupcor_Length_3Y<span class="sc">$</span>consensus.correlation</span>
<span id="cb178-8"><a href="#cb178-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-9"><a href="#cb178-9" aria-hidden="true" tabindex="-1"></a>consensus.corr._Length_3Y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.4824957</code></pre>
</div>
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lmFit fits a linear model using weighted least squares for each gene:</span></span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a>fit_Length_3Y <span class="ot">=</span> <span class="fu">lmFit</span>(y_Length_3Y, </span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">design =</span> mm_Length_3Y, </span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">block =</span> metadata_S2<span class="sc">$</span>Hybrid,</span>
<span id="cb180-5"><a href="#cb180-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">correlation =</span> consensus.corr._Length_3Y) </span>
<span id="cb180-6"><a href="#cb180-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-7"><a href="#cb180-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Ebayes</span></span>
<span id="cb180-8"><a href="#cb180-8" aria-hidden="true" tabindex="-1"></a>BlockFit_Length_3Y <span class="ot">=</span> <span class="fu">eBayes</span>(fit_Length_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="limma-results-_length_3y" class="level1">
<h1>Limma results _Length_3Y</h1>
<section id="result-summary-1" class="level2">
<h2 class="anchored" data-anchor-id="result-summary-1">Result summary</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a>res_summaries_Length_3Y <span class="ot">=</span> BlockFit_Length_3Y <span class="sc">%&gt;%</span> <span class="fu">decideTests</span>() <span class="sc">%&gt;%</span> <span class="fu">summary</span>()</span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-3"><a href="#cb181-3" aria-hidden="true" tabindex="-1"></a>res_summaries_Length_3Y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       (Intercept) length_3Y
Down             8      3381
NotSig         514     12658
Up           17460      1943</code></pre>
</div>
</div>
</section>
<section id="table-of-results-_length_3y" class="level2">
<h2 class="anchored" data-anchor-id="table-of-results-_length_3y">Table of results _Length_3Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a>impLength_3Y <span class="ot">=</span> <span class="fu">topTable</span>(BlockFit_Length_3Y,</span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">sort.by =</span> <span class="st">"logFC"</span>,</span>
<span id="cb183-3"><a href="#cb183-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">p.value =</span> <span class="fl">0.05</span>,</span>
<span id="cb183-4"><a href="#cb183-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">adjust.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb183-5"><a href="#cb183-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">number =</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span></span>
<span id="cb183-6"><a href="#cb183-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb183-7"><a href="#cb183-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">R =</span> <span class="fu">sqrt</span>(t<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>(t<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">40</span>)), <span class="at">AveExpr =</span> <span class="dv">2</span><span class="sc">^</span>AveExpr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Removing intercept from test coefficients</code></pre>
</div>
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(impLength_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5324    8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="do">## adding Hybrid as a blocking variable reduced DEGs by several thousand</span></span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(impLength_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID    logFC  AveExpr        t      P.Value    adj.P.Val          B
1 121248876 5.301599 3.425589 5.264151 8.968987e-06 0.0008765235  2.4011184
2 109002793 5.021070 2.720068 4.816599 3.312576e-05 0.0014891684  1.3439539
3 108989061 4.777773 5.530849 5.244331 9.504258e-06 0.0009042623  2.6498829
4 121263736 4.719908 3.667753 4.295222 1.494307e-04 0.0030954565  0.4474998
5 121261538 4.713332 1.599117 3.263793 2.600144e-03 0.0158378735 -1.8349006
6 109019660 4.513595 1.289745 5.840641 1.663369e-06 0.0004235295  2.8636710
          R
1 0.6397316
2 0.6058753
3 0.6383054
4 0.5618202
5 0.4585882
6 0.6784425</code></pre>
</div>
</div>
</section>
<section id="merge-annotation-with-results-_length_3y" class="level2">
<h2 class="anchored" data-anchor-id="merge-annotation-with-results-_length_3y">Merge annotation with results _Length_3Y</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a>impLength_3Y <span class="ot">=</span> impLength_3Y <span class="sc">%&gt;%</span></span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(annotation_combined, <span class="at">by =</span> <span class="st">"GeneID"</span>)</span>
<span id="cb189-3"><a href="#cb189-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4"><a href="#cb189-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(impLength_3Y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     GeneID    logFC  AveExpr        t      P.Value    adj.P.Val          B
1 121248876 5.301599 3.425589 5.264151 8.968987e-06 0.0008765235  2.4011184
2 109002793 5.021070 2.720068 4.816599 3.312576e-05 0.0014891684  1.3439539
3 108989061 4.777773 5.530849 5.244331 9.504258e-06 0.0009042623  2.6498829
4 121263736 4.719908 3.667753 4.295222 1.494307e-04 0.0030954565  0.4474998
5 121261538 4.713332 1.599117 3.263793 2.600144e-03 0.0158378735 -1.8349006
6 109019660 4.513595 1.289745 5.840641 1.663369e-06 0.0004235295  2.8636710
          R feature class chromosome    start      end
1 0.6397316    mRNA               2D 20832957 20834478
2 0.6058753    mRNA                8  1118528  1119375
3 0.6383054    mRNA                1 21070915 21072478
4 0.5618202    mRNA               4S  1043357  1044200
5 0.4585882    mRNA               4D 30605557 30611409
6 0.6784425    mRNA               13  9506430  9511614
                                                                 name Jr-GeneID
1                    desmethylxanthohumol 6'-O-methyltransferase-like 108989061
2                             14 kDa proline-rich protein DC2.15-like        NA
3                    desmethylxanthohumol 6'-O-methyltransferase-like        NA
4                             14 kDa proline-rich protein DC2.15-like 109002793
5 endoplasmic reticulum-Golgi intermediate compartment protein 3-like 109003827
6                                  lipoxygenase 6, chloroplastic-like        NA
  Jr-chromosome         Subcell_loc Parent_haplotype
1             1           cytoplasm     J.microcarpa
2          &lt;NA&gt;     plasma membrane          J.regia
3          &lt;NA&gt;           cytoplasm          J.regia
4             8     plasma membrane     J.microcarpa
5             8 endomembrane system     J.microcarpa
6          &lt;NA&gt;             nucleus          J.regia</code></pre>
</div>
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(impLength_3Y, <span class="st">'DGEresults/Limma_results_table_Length_3Y.csv'</span>)</span>
<span id="cb191-2"><a href="#cb191-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-3"><a href="#cb191-3" aria-hidden="true" tabindex="-1"></a>impLength_3Y <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'DGEresults/Limma_results_table_Length_3Y.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pca-of-_length_3y-degs" class="level2">
<h2 class="anchored" data-anchor-id="pca-of-_length_3y-degs">PCA of _Length_3Y DEGs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">=</span> impLength_3Y <span class="sc">%&gt;%</span> </span>
<span id="cb192-2"><a href="#cb192-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb192-3"><a href="#cb192-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">as.character</span>(GeneID))</span>
<span id="cb192-4"><a href="#cb192-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-5"><a href="#cb192-5" aria-hidden="true" tabindex="-1"></a>PCA_Length_3Y <span class="ot">=</span> cpmd2 <span class="sc">%&gt;%</span> </span>
<span id="cb192-6"><a href="#cb192-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb192-7"><a href="#cb192-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb192-8"><a href="#cb192-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(ids) <span class="sc">%&gt;%</span> </span>
<span id="cb192-9"><a href="#cb192-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">'GeneID'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(GeneID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a>pca_plot_Length_3Y <span class="ot">=</span> <span class="fu">plot_pca</span>(PCA_Length_3Y, metadata,</span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">join_by_name =</span> <span class="st">'Sample'</span>,</span>
<span id="cb194-3"><a href="#cb194-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_in =</span> <span class="st">'col_names'</span>,</span>
<span id="cb194-4"><a href="#cb194-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">plotting_factors_name =</span> Group, </span>
<span id="cb194-5"><a href="#cb194-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">x =</span> <span class="st">'PC1'</span>,</span>
<span id="cb194-6"><a href="#cb194-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> <span class="st">'PC2'</span>,</span>
<span id="cb194-7"><a href="#cb194-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> T, </span>
<span id="cb194-8"><a href="#cb194-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">center =</span> T, </span>
<span id="cb194-9"><a href="#cb194-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">'length_3Y'</span>,</span>
<span id="cb194-10"><a href="#cb194-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">'length_3Y'</span>,</span>
<span id="cb194-11"><a href="#cb194-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot_type =</span> <span class="st">'2D'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Make sure no numeric identifiers are in data as this will drastically impact PCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb196"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb196-1"><a href="#cb196-1" aria-hidden="true" tabindex="-1"></a>pca_plot_Length_3Y<span class="sc">$</span>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-68-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="read-in-_length_3y-results" class="level2">
<h2 class="anchored" data-anchor-id="read-in-_length_3y-results">Read in _Length_3Y results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a>impLength_3Y <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">'DGEresults/Limma_results_table_Length_3Y.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="venn-diagram-positive-genes" class="level1">
<h1>Venn Diagram positive genes</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a>CG_pos <span class="ot">=</span> impCG <span class="sc">%&gt;%</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&gt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb198-5"><a href="#cb198-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb198-6"><a href="#cb198-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-7"><a href="#cb198-7" aria-hidden="true" tabindex="-1"></a>PHY_pos <span class="ot">=</span> impPHY <span class="sc">%&gt;%</span></span>
<span id="cb198-8"><a href="#cb198-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&gt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb198-9"><a href="#cb198-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb198-10"><a href="#cb198-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb198-11"><a href="#cb198-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-12"><a href="#cb198-12" aria-hidden="true" tabindex="-1"></a>NEM_pos <span class="ot">=</span> impNEM_3Y <span class="sc">%&gt;%</span></span>
<span id="cb198-13"><a href="#cb198-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&gt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb198-14"><a href="#cb198-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID)  <span class="sc">%&gt;%</span> </span>
<span id="cb198-15"><a href="#cb198-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb198-16"><a href="#cb198-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-17"><a href="#cb198-17" aria-hidden="true" tabindex="-1"></a>Length_pos <span class="ot">=</span> impLength_3Y <span class="sc">%&gt;%</span></span>
<span id="cb198-18"><a href="#cb198-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&gt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb198-19"><a href="#cb198-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID)  <span class="sc">%&gt;%</span> </span>
<span id="cb198-20"><a href="#cb198-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb198-21"><a href="#cb198-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-22"><a href="#cb198-22" aria-hidden="true" tabindex="-1"></a>venn_pos <span class="ot">=</span> <span class="fu">list</span>(<span class="at">CG_pos =</span> CG_pos, </span>
<span id="cb198-23"><a href="#cb198-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">PHY_pos =</span> PHY_pos,</span>
<span id="cb198-24"><a href="#cb198-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">NEM_pos =</span> NEM_pos,</span>
<span id="cb198-25"><a href="#cb198-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">Length_pos =</span> Length_pos)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="venn-diagram-negative-genes" class="level1">
<h1>Venn Diagram negative genes</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a>CG_neg <span class="ot">=</span> impCG <span class="sc">%&gt;%</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb199-6"><a href="#cb199-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-7"><a href="#cb199-7" aria-hidden="true" tabindex="-1"></a>PHY_neg <span class="ot">=</span> impPHY <span class="sc">%&gt;%</span></span>
<span id="cb199-8"><a href="#cb199-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb199-9"><a href="#cb199-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb199-10"><a href="#cb199-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb199-11"><a href="#cb199-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-12"><a href="#cb199-12" aria-hidden="true" tabindex="-1"></a>NEM_neg <span class="ot">=</span> impNEM_3Y <span class="sc">%&gt;%</span></span>
<span id="cb199-13"><a href="#cb199-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb199-14"><a href="#cb199-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID)  <span class="sc">%&gt;%</span> </span>
<span id="cb199-15"><a href="#cb199-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb199-16"><a href="#cb199-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-17"><a href="#cb199-17" aria-hidden="true" tabindex="-1"></a>Length_neg <span class="ot">=</span> impLength_3Y <span class="sc">%&gt;%</span></span>
<span id="cb199-18"><a href="#cb199-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&lt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb199-19"><a href="#cb199-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneID)  <span class="sc">%&gt;%</span> </span>
<span id="cb199-20"><a href="#cb199-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(GeneID)</span>
<span id="cb199-21"><a href="#cb199-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-22"><a href="#cb199-22" aria-hidden="true" tabindex="-1"></a>venn_neg <span class="ot">=</span> <span class="fu">list</span>(<span class="at">CG_neg =</span> CG_neg,</span>
<span id="cb199-23"><a href="#cb199-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">PHY_neg =</span> PHY_neg, </span>
<span id="cb199-24"><a href="#cb199-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">NEM_neg =</span> NEM_neg,</span>
<span id="cb199-25"><a href="#cb199-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">Length_neg =</span> Length_neg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-and-save-venn-diagram-figure-with-results-summaries" class="level1">
<h1>Plot and save venn diagram figure with results summaries</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggvenn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggvenn' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">=</span> <span class="fu">ggvenn</span>(venn_pos, <span class="at">text_size =</span> <span class="dv">6</span>)</span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a>b <span class="ot">=</span> <span class="fu">ggvenn</span>(venn_neg, <span class="at">text_size =</span> <span class="dv">6</span>)</span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-5"><a href="#cb203-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(a, b, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'A)'</span>, <span class="st">'B)'</span>), <span class="at">label.y =</span> <span class="fl">0.75</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-72-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_plot</span>(<span class="st">'DGEresults/Venn_Pos_Neg.png'</span>,</span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ggarrange</span>(a, b, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'A)'</span>, <span class="st">'B)'</span>), </span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">font.label =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">25</span>),</span>
<span id="cb204-4"><a href="#cb204-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label.y =</span> <span class="fl">0.75</span>, </span>
<span id="cb204-5"><a href="#cb204-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span>)),</span>
<span id="cb204-6"><a href="#cb204-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">width =</span> <span class="dv">45</span>, </span>
<span id="cb204-7"><a href="#cb204-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">height =</span> <span class="dv">23</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a>all_res_summaries <span class="ot">=</span> <span class="fu">cbind</span>(res_summaries_CG, </span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a>                          res_summaries_PHY,</span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>                          res_summaries_NEM_3Y,</span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a>                          res_summaries_Length_3Y) <span class="sc">%&gt;%</span> </span>
<span id="cb206-5"><a href="#cb206-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb206-6"><a href="#cb206-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">matches</span>(<span class="st">'Intercept'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb206-7"><a href="#cb206-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">'A. tumefaciens'</span>,</span>
<span id="cb206-8"><a href="#cb206-8" aria-hidden="true" tabindex="-1"></a>          <span class="st">'Phytophthora spp.'</span>,</span>
<span id="cb206-9"><a href="#cb206-9" aria-hidden="true" tabindex="-1"></a>          <span class="st">'P. vulnus'</span>,</span>
<span id="cb206-10"><a href="#cb206-10" aria-hidden="true" tabindex="-1"></a>          <span class="st">'three year tree length'</span>))</span>
<span id="cb206-11"><a href="#cb206-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-12"><a href="#cb206-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-13"><a href="#cb206-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-14"><a href="#cb206-14" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(all_res_summaries, <span class="at">FUN =</span> <span class="cf">function</span>(x){</span>
<span id="cb206-15"><a href="#cb206-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="fu">paste0</span>(x[<span class="dv">1</span>], <span class="st">' genes negatively associated with'</span>),</span>
<span id="cb206-16"><a href="#cb206-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">', '</span>,</span>
<span id="cb206-17"><a href="#cb206-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste0</span>(x[<span class="dv">3</span>], <span class="st">' genes positively associated with'</span>),</span>
<span id="cb206-18"><a href="#cb206-18" aria-hidden="true" tabindex="-1"></a>        <span class="st">', and '</span>,</span>
<span id="cb206-19"><a href="#cb206-19" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste0</span>(x[<span class="dv">2</span>], <span class="st">' genes not associated with '</span>, x[<span class="dv">4</span>]))</span>
<span id="cb206-20"><a href="#cb206-20" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$CG_Avg
[1] "3888 genes negatively associated with, 2021 genes positively associated with, and 13290 genes not associated with A. tumefaciens"

$PHY_Avg
[1] "1061 genes negatively associated with, 407 genes positively associated with, and 17731 genes not associated with Phytophthora spp."

$RLN_3Y
[1] "13 genes negatively associated with, 7 genes positively associated with, and 17962 genes not associated with P. vulnus"

$length_3Y
[1] "3381 genes negatively associated with, 1943 genes positively associated with, and 12658 genes not associated with three year tree length"</code></pre>
</div>
</div>
<p>####End of analysis</p>
</section>
<section id="extras" class="level1">
<h1>Extras&gt;&gt;&gt;&gt;&gt;&gt;</h1>
</section>
<section id="trait-data-have-some-collinearity" class="level1">
<h1>Trait data have some collinearity</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb208"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggcorrplot' was built under R version 4.2.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get matrix for correlation</span></span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a>cordat <span class="ot">=</span>  metadata_S <span class="sc">%&gt;%</span></span>
<span id="cb210-3"><a href="#cb210-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(CG_Avg, PHY_Avg, RLN_2Y) <span class="sc">%&gt;%</span></span>
<span id="cb210-4"><a href="#cb210-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale</span>() <span class="sc">%&gt;%</span></span>
<span id="cb210-5"><a href="#cb210-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb210-6"><a href="#cb210-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-7"><a href="#cb210-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate p-values</span></span>
<span id="cb210-8"><a href="#cb210-8" aria-hidden="true" tabindex="-1"></a>corpmat1 <span class="ot">=</span> <span class="fu">cor_pmat</span>(cordat, <span class="at">method =</span> <span class="st">'pearson'</span>)</span>
<span id="cb210-9"><a href="#cb210-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-10"><a href="#cb210-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate correlation coefficients</span></span>
<span id="cb210-11"><a href="#cb210-11" aria-hidden="true" tabindex="-1"></a>cormat1 <span class="ot">=</span><span class="fu">cor</span>(cordat, <span class="at">method =</span> <span class="st">'pearson'</span>)</span>
<span id="cb210-12"><a href="#cb210-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-13"><a href="#cb210-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot. All relationships are significant</span></span>
<span id="cb210-14"><a href="#cb210-14" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggcorrplot</span>(cormat1, <span class="at">type =</span> <span class="st">'lower'</span>, <span class="at">hc.order =</span> T, <span class="at">lab =</span> T, <span class="at">p.mat =</span> corpmat1)</span>
<span id="cb210-15"><a href="#cb210-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-16"><a href="#cb210-16" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SCRI_ROOT_RNAs_files/figure-html/unnamed-chunk-74-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_plot</span>(<span class="st">'DGEresults/Pheno_corrplot.png'</span>, p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>